
<!DOCTYPE html>


<html xmlns="http://www.w3.org/1999/xhtml">
  <head>
    <meta http-equiv="Content-Type" content="text/html; charset=utf-8" />
    
    <title>Model Reference and Vignettes &mdash; Zelig 5.0-1 documentation</title>
    
    <link rel="stylesheet" href="_static/basic.css" type="text/css" />
    <link rel="stylesheet" href="_static/pygments.css" type="text/css" />
    <link rel="stylesheet" href="static/my-styles.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-3.1.0/css/bootstrap.min.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-3.1.0/css/bootstrap-theme.min.css" type="text/css" />
    <link rel="stylesheet" href="_static/bootstrap-sphinx.css" type="text/css" />
    
    <script type="text/javascript">
      var DOCUMENTATION_OPTIONS = {
        URL_ROOT:    './',
        VERSION:     '5.0-1',
        COLLAPSE_INDEX: false,
        FILE_SUFFIX: '.html',
        HAS_SOURCE:  true
      };
    </script>
    <script type="text/javascript" src="_static/jquery.js"></script>
    <script type="text/javascript" src="_static/underscore.js"></script>
    <script type="text/javascript" src="_static/doctools.js"></script>
    <script type="text/javascript" src="_static/js/jquery-1.11.0.min.js"></script>
    <script type="text/javascript" src="_static/js/jquery-fix.js"></script>
    <script type="text/javascript" src="_static/bootstrap-3.1.0/js/bootstrap.min.js"></script>
    <script type="text/javascript" src="_static/bootstrap-sphinx.js"></script>
    <link rel="author" title="About these documents" href="about.html" />
    <link rel="top" title="Zelig 5.0-1 documentation" href="index.html" />
    <link rel="next" title="zelig-exp" href="zelig-exp.html" />
    <link rel="prev" title="Installation and Quickstart" href="installation_quickstart.html" />
<meta charset='utf-8'>
<meta http-equiv='X-UA-Compatible' content='IE=edge,chrome=1'>
<meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1'>
<meta name="apple-mobile-web-app-capable" content="yes">

  </head>
  <body>

<a href="https://github.com/IQSS/Zelig"><img style="position: absolute; top: 45px; right: 0; border: 0;" src="https://s3.amazonaws.com/github/ribbons/forkme_right_orange_ff7600.png"" alt="Fork me on GitHub"></a>
<div id="navbar" class="navbar-inverse navbar-default navbar-fixed-top">

    <div class="container">

      <div class="navbar-header">

        <!-- .btn-navbar is used as the toggle for collapsed navbar content -->
        <button type="button" class="navbar-toggle" data-toggle="collapse" data-target=".nav-collapse">
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
          <span class="icon-bar"></span>
        </button>

        <!-- <a class="logo"> <img height="32" width="32" src="http://datascience.iq.harvard.edu/files/styles/os_files_small/public/datascience/files/zelig-small_1.png?itok=catC420a" /></a> -->
        <a class="navbar-brand" href="index.html">Zelig</a>
        <span class="navbar-text navbar-version pull-left"><b>Version 5.0-1</b></span>
      </div>
        <div class="collapse navbar-collapse nav-collapse">
          <ul class="nav navbar-nav">
            <li class="divider-vertical"></li>
            
                <li><a href="http://roadmap.datascience.iq.harvard.edu/milestones/milestone-roadmap/Zelig/">Roadmap</a></li>
            
            
              <li class="dropdown globaltoc-container">
  <a role="button"
     id="dLabelGlobalToc"
     data-toggle="dropdown"
     data-target="#"
     href="index.html">Site <b class="caret"></b></a>
  <ul class="dropdown-menu globaltoc"
      role="menu"
      aria-labelledby="dLabelGlobalToc"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation_quickstart.html">Installation and Quickstart</a><ul>
<li class="toctree-l2"><a class="reference internal" href="installation_quickstart.html#installing-r-and-zelig">Installing R and Zelig</a></li>
<li class="toctree-l2"><a class="reference internal" href="installation_quickstart.html#quickstart-guide">Quickstart Guide</a></li>
</ul>
</li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Model Reference and Vignettes</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reference">Reference</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-exp">zelig-exp</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-gamma">zelig-gamma</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-logit">zelig-logit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-lognorm">zelig-lognorm</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-ls">zelig-ls</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-negbin">zelig-negbin</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-normal">zelig-normal</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poisson">zelig-poisson</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probit">zelig-probit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-relogit">zelig-relogit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-tobit">zelig-tobit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-weibull">zelig-weibull</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-factorbayes">zelig-factorbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-mlogitbayes">zelig-mlogitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-oprobitbayes">zelig-oprobitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poissonbayes">zelig-poissonbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probitbayes">zelig-probitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-tobitbayes">zelig-tobitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-gammagee">zelig-gammagee</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probitgee">zelig-probitgee</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poissongee">zelig-poissongee</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">Frequently Asked Questions</a><ul>
<li class="toctree-l2"><a class="reference internal" href="faq.html#why-cant-i-install-zelig">Why can’t I install Zelig?</a></li>
<li class="toctree-l2"><a class="reference internal" href="faq.html#why-cant-i-install-r">Why can’t I install R?</a></li>
<li class="toctree-l2"><a class="reference internal" href="faq.html#why-cant-i-load-data">Why can’t I load data?</a></li>
<li class="toctree-l2"><a class="reference internal" href="faq.html#r-is-neat-how-can-i-find-out-more">R is neat. How can I find out more?</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About Zelig</a><ul>
<li class="toctree-l2"><a class="reference internal" href="about.html#technical-vision">Technical Vision</a></li>
<li class="toctree-l2"><a class="reference internal" href="about.html#release-notes">Release Notes</a></li>
</ul>
</li>
</ul>
</ul>
</li>
              
                <li class="dropdown">
  <a role="button"
     id="dLabelLocalToc"
     data-toggle="dropdown"
     data-target="#"
     href="#">Page <b class="caret"></b></a>
  <ul class="dropdown-menu localtoc"
      role="menu"
      aria-labelledby="dLabelLocalToc"><ul>
<li><a class="reference internal" href="#">Model Reference and Vignettes</a><ul>
<li><a class="reference internal" href="#reference">Reference</a></li>
<li><a class="reference internal" href="#zelig-exp">zelig-exp</a><ul>
<li><a class="reference internal" href="#syntax">Syntax</a></li>
<li><a class="reference internal" href="#input-values">Input Values</a></li>
<li><a class="reference internal" href="#example">Example</a></li>
<li><a class="reference internal" href="#model">Model</a></li>
<li><a class="reference internal" href="#quantities-of-interest">Quantities of Interest</a></li>
<li><a class="reference internal" href="#output-values">Output Values</a></li>
<li><a class="reference internal" href="#see-also">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-gamma">zelig-gamma</a><ul>
<li><a class="reference internal" href="#id1">Syntax</a></li>
<li><a class="reference internal" href="#id2">Example</a></li>
<li><a class="reference internal" href="#id3">Model</a></li>
<li><a class="reference internal" href="#id4">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id5">Output Values</a></li>
<li><a class="reference internal" href="#id6">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-logit">zelig-logit</a><ul>
<li><a class="reference internal" href="#id7">Syntax</a></li>
<li><a class="reference internal" href="#examples">Examples</a><ul>
<li><a class="reference internal" href="#basic-example">Basic Example</a></li>
<li><a class="reference internal" href="#simulating-first-differences">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id8">Model</a></li>
<li><a class="reference internal" href="#id9">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id10">Output Values</a></li>
<li><a class="reference internal" href="#id11">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-lognorm">zelig-lognorm</a><ul>
<li><a class="reference internal" href="#id12">Syntax</a></li>
<li><a class="reference internal" href="#id13">Input Values</a></li>
<li><a class="reference internal" href="#id14">Example</a></li>
<li><a class="reference internal" href="#id15">Model</a></li>
<li><a class="reference internal" href="#id16">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id17">Output Values</a></li>
<li><a class="reference internal" href="#id18">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-ls">zelig-ls</a><ul>
<li><a class="reference internal" href="#id19">Syntax</a></li>
<li><a class="reference internal" href="#id20">Examples</a><ul>
<li><a class="reference internal" href="#basic-example-with-first-differences">Basic Example with First Differences</a></li>
<li><a class="reference internal" href="#using-dummy-variables">Using Dummy Variables</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id21">Model</a></li>
<li><a class="reference internal" href="#id22">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id23">Output Values</a></li>
<li><a class="reference internal" href="#id24">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-negbin">zelig-negbin</a><ul>
<li><a class="reference internal" href="#id25">Syntax</a></li>
<li><a class="reference internal" href="#id26">Example</a></li>
<li><a class="reference internal" href="#id27">Model</a></li>
<li><a class="reference internal" href="#id28">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id29">Output Values</a></li>
<li><a class="reference internal" href="#id30">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-normal">zelig-normal</a><ul>
<li><a class="reference internal" href="#id33">Syntax</a></li>
<li><a class="reference internal" href="#id34">Examples</a><ul>
<li><a class="reference internal" href="#id35">Basic Example with First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id36">Model</a></li>
<li><a class="reference internal" href="#id37">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id38">Output Values</a></li>
<li><a class="reference internal" href="#id39">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-poisson">zelig-poisson</a><ul>
<li><a class="reference internal" href="#id40">Syntax</a></li>
<li><a class="reference internal" href="#id41">Example</a></li>
<li><a class="reference internal" href="#id42">Model</a></li>
<li><a class="reference internal" href="#id43">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id44">Output Values</a></li>
<li><a class="reference internal" href="#id45">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-probit">zelig-probit</a><ul>
<li><a class="reference internal" href="#id46">Syntax</a></li>
<li><a class="reference internal" href="#id47">Example</a></li>
<li><a class="reference internal" href="#id48">Model</a></li>
<li><a class="reference internal" href="#id49">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id50">Output Values</a></li>
<li><a class="reference internal" href="#id51">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-relogit">zelig-relogit</a><ul>
<li><a class="reference internal" href="#id52">Syntax</a></li>
<li><a class="reference internal" href="#arguments">Arguments</a></li>
<li><a class="reference internal" href="#example-1-one-tau-with-prior-correction-and-bias-correction">Example 1: One Tau with Prior Correction and Bias Correction</a></li>
<li><a class="reference internal" href="#example-2-one-tau-with-weighting-robust-standard-errors-and-bias-correction">Example 2: One Tau with Weighting, Robust Standard Errors, and Bias Correction</a></li>
<li><a class="reference internal" href="#example-3-two-taus-with-bias-correction-and-prior-correction">Example 3: Two Taus with Bias Correction and Prior Correction</a></li>
<li><a class="reference internal" href="#id53">Model</a></li>
<li><a class="reference internal" href="#id54">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id55">Output Values</a></li>
<li><a class="reference internal" href="#differences-with-stata-version">Differences with Stata Version</a></li>
<li><a class="reference internal" href="#id56">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-tobit">zelig-tobit</a><ul>
<li><a class="reference internal" href="#id57">Syntax</a></li>
<li><a class="reference internal" href="#inputs">Inputs</a></li>
<li><a class="reference internal" href="#id58">Examples</a><ul>
<li><a class="reference internal" href="#id59">Basic Example</a></li>
<li><a class="reference internal" href="#id60">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id61">Model</a></li>
<li><a class="reference internal" href="#id62">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id63">Output Values</a></li>
<li><a class="reference internal" href="#id66">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-weibull">zelig-weibull</a><ul>
<li><a class="reference internal" href="#id67">Syntax</a></li>
<li><a class="reference internal" href="#id68">Input Values</a></li>
<li><a class="reference internal" href="#id69">Example</a></li>
<li><a class="reference internal" href="#id70">Model</a></li>
<li><a class="reference internal" href="#id73">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id74">Output Values</a></li>
<li><a class="reference internal" href="#id75">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-factorbayes">zelig-factorbayes</a><ul>
<li><a class="reference internal" href="#id76">Syntax</a></li>
<li><a class="reference internal" href="#id77">Inputs</a></li>
<li><a class="reference internal" href="#additional-inputs">Additional Inputs</a></li>
<li><a class="reference internal" href="#id78">Example</a></li>
<li><a class="reference internal" href="#id79">Model</a></li>
<li><a class="reference internal" href="#id80">Output Values</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-mlogitbayes">zelig-mlogitbayes</a><ul>
<li><a class="reference internal" href="#id81">Syntax</a></li>
<li><a class="reference internal" href="#id82">Additional Inputs</a></li>
<li><a class="reference internal" href="#id83">Examples</a><ul>
<li><a class="reference internal" href="#id84">Basic Example</a></li>
<li><a class="reference internal" href="#id85">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id86">Model</a></li>
<li><a class="reference internal" href="#id87">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id88">Output Values</a></li>
<li><a class="reference internal" href="#id89">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-oprobitbayes">zelig-oprobitbayes</a><ul>
<li><a class="reference internal" href="#id90">Syntax</a></li>
<li><a class="reference internal" href="#id91">Additional Inputs</a></li>
<li><a class="reference internal" href="#id92">Examples</a><ul>
<li><a class="reference internal" href="#id93">Basic Example</a></li>
<li><a class="reference internal" href="#id94">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id95">Model</a></li>
<li><a class="reference internal" href="#id96">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id97">Output Values</a></li>
<li><a class="reference internal" href="#id98">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-poissonbayes">zelig-poissonbayes</a><ul>
<li><a class="reference internal" href="#id99">Syntax</a></li>
<li><a class="reference internal" href="#id100">Additional Inputs</a></li>
<li><a class="reference internal" href="#id101">Examples</a><ul>
<li><a class="reference internal" href="#id102">Basic Example</a></li>
<li><a class="reference internal" href="#id103">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id104">Model</a></li>
<li><a class="reference internal" href="#id105">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id106">Output Values</a></li>
<li><a class="reference internal" href="#id107">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-probitbayes">zelig-probitbayes</a><ul>
<li><a class="reference internal" href="#id108">Syntax</a></li>
<li><a class="reference internal" href="#id109">Additional Inputs</a></li>
<li><a class="reference internal" href="#id110">Examples</a><ul>
<li><a class="reference internal" href="#id111">Basic Example</a></li>
<li><a class="reference internal" href="#id112">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id113">Model</a></li>
<li><a class="reference internal" href="#id114">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id115">Output Values</a></li>
<li><a class="reference internal" href="#id116">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-tobitbayes">zelig-tobitbayes</a><ul>
<li><a class="reference internal" href="#id117">Syntax</a></li>
<li><a class="reference internal" href="#id118">Inputs</a></li>
<li><a class="reference internal" href="#id119">Additional Inputs</a></li>
<li><a class="reference internal" href="#id120">Examples</a><ul>
<li><a class="reference internal" href="#id121">Basic Example</a></li>
<li><a class="reference internal" href="#id122">Simulating First Differences</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id123">Model</a></li>
<li><a class="reference internal" href="#id124">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id125">Output Values</a></li>
<li><a class="reference internal" href="#id126">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-gammagee">zelig-gammagee</a><ul>
<li><a class="reference internal" href="#id127">Syntax</a></li>
<li><a class="reference internal" href="#id128">Additional Inputs</a></li>
<li><a class="reference internal" href="#id129">Examples</a><ul>
<li><a class="reference internal" href="#example-with-exchangeable-dependence">Example with Exchangeable Dependence</a></li>
</ul>
</li>
<li><a class="reference internal" href="#the-model">The Model</a></li>
<li><a class="reference internal" href="#id130">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id131">Output Values</a></li>
<li><a class="reference internal" href="#id132">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-probitgee">zelig-probitgee</a><ul>
<li><a class="reference internal" href="#id133">Syntax</a></li>
<li><a class="reference internal" href="#id134">Additional Inputs</a></li>
<li><a class="reference internal" href="#id135">Examples</a><ul>
<li><a class="reference internal" href="#example-with-stationary-3-dependence">Example with Stationary 3 Dependence</a></li>
<li><a class="reference internal" href="#id136">Simulating First Differences</a></li>
<li><a class="reference internal" href="#example-with-fixed-correlation-structure">Example with Fixed Correlation Structure</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id137">The Model</a></li>
<li><a class="reference internal" href="#id138">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id139">Output Values</a></li>
<li><a class="reference internal" href="#id140">See also</a></li>
</ul>
</li>
<li><a class="reference internal" href="#zelig-poissongee">zelig-poissongee</a><ul>
<li><a class="reference internal" href="#id141">Syntax</a></li>
<li><a class="reference internal" href="#id142">Additional Inputs</a></li>
<li><a class="reference internal" href="#id143">Examples</a><ul>
<li><a class="reference internal" href="#id144">Example with Exchangeable Dependence</a></li>
</ul>
</li>
<li><a class="reference internal" href="#id145">The Model</a></li>
<li><a class="reference internal" href="#id146">Quantities of Interest</a></li>
<li><a class="reference internal" href="#id147">Output Values</a></li>
<li><a class="reference internal" href="#id148">See also</a></li>
</ul>
</li>
</ul>
</li>
</ul>
</ul>
</li>
              
            
            
              
                
  <li>
    <a href="installation_quickstart.html" title="Previous Chapter: Installation and Quickstart"><span class="glyphicon glyphicon-chevron-left visible-sm"></span><span class="hidden-sm">&laquo; Installation and...</span>
    </a>
  </li>
  <li>
    <a href="zelig-exp.html" title="Next Chapter: zelig-exp"><span class="glyphicon glyphicon-chevron-right visible-sm"></span><span class="hidden-sm">zelig-exp &raquo;</span>
    </a>
  </li>
              
            
            
            
            
          </ul>

          
            
<form class="navbar-form navbar-right" action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
          

        </div>
    </div>
  </div>

<div class="container">
  <div class="row">
      <div class="col-md-3">
        <div id="sidebar" class="bs-sidenav" role="complementary"><ul class="current">
<li class="toctree-l1"><a class="reference internal" href="installation_quickstart.html">Installation and Quickstart</a></li>
<li class="toctree-l1 current"><a class="current reference internal" href="">Model Reference and Vignettes</a><ul>
<li class="toctree-l2"><a class="reference internal" href="#reference">Reference</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-exp">zelig-exp</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-gamma">zelig-gamma</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-logit">zelig-logit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-lognorm">zelig-lognorm</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-ls">zelig-ls</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-negbin">zelig-negbin</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-normal">zelig-normal</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poisson">zelig-poisson</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probit">zelig-probit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-relogit">zelig-relogit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-tobit">zelig-tobit</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-weibull">zelig-weibull</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-factorbayes">zelig-factorbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-mlogitbayes">zelig-mlogitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-oprobitbayes">zelig-oprobitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poissonbayes">zelig-poissonbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probitbayes">zelig-probitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-tobitbayes">zelig-tobitbayes</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-gammagee">zelig-gammagee</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-probitgee">zelig-probitgee</a></li>
<li class="toctree-l2"><a class="reference internal" href="#zelig-poissongee">zelig-poissongee</a></li>
</ul>
</li>
<li class="toctree-l1"><a class="reference internal" href="faq.html">Frequently Asked Questions</a></li>
<li class="toctree-l1"><a class="reference internal" href="about.html">About Zelig</a></li>
</ul>

<form action="search.html" method="get">
 <div class="form-group">
  <input type="text" name="q" class="form-control" placeholder="Search" />
 </div>
  <input type="hidden" name="check_keywords" value="yes" />
  <input type="hidden" name="area" value="default" />
</form>
        </div>
      </div>
    <div class="col-md-9">
      
  <div class="section" id="model-reference-and-vignettes">
<span id="vignettes"></span><h1>Model Reference and Vignettes<a class="headerlink" href="#model-reference-and-vignettes" title="Permalink to this headline">¶</a></h1>
<p>This section includes technical information on the models currently implemented in Zelig (5.0-1). This includes a reference with a list of supported models as well as individual model vignettes with detailed information on the model, quantities of interest and syntax.</p>
<div class="section" id="reference">
<h2>Reference<a class="headerlink" href="#reference" title="Permalink to this headline">¶</a></h2>
<p>The following models are currently supported in Zelig 5.0-1:</p>
<ul class="simple">
<li><a class="reference internal" href="zelig-exp.html#zexp"><em>Exponential Regression</em></a>: <tt class="docutils literal"><span class="pre">zexp$new()</span></tt></li>
<li><a class="reference internal" href="zelig-gamma.html#zgamma"><em>Gamma Regression</em></a>: <tt class="docutils literal"><span class="pre">zgamma()</span></tt></li>
<li><a class="reference internal" href="zelig-logit.html#zlogit"><em>Logistic Regression</em></a>: <tt class="docutils literal"><span class="pre">zlogit$new()</span></tt></li>
<li><a class="reference internal" href="zelig-lognorm.html#zlognorm"><em>Log Normal Regression</em></a>: <tt class="docutils literal"><span class="pre">zlognorm$new()</span></tt></li>
<li><a class="reference internal" href="zelig-ls.html#zls"><em>Least Squares Regression</em></a>: <tt class="docutils literal"><span class="pre">zls$new()</span></tt></li>
<li><a class="reference internal" href="zelig-negbin.html#znegbin"><em>Negative Binomial Regression</em></a>: <tt class="docutils literal"><span class="pre">zbinom$new()</span></tt></li>
<li><a class="reference internal" href="zelig-normal.html#znorm"><em>Normal Regression</em></a>: <tt class="docutils literal"><span class="pre">znormal$new()</span></tt></li>
<li><a class="reference internal" href="zelig-poisson.html#zpoisson"><em>Poisson Regression</em></a>: <tt class="docutils literal"><span class="pre">zpoisson$new()</span></tt></li>
<li><a class="reference internal" href="zelig-probit.html#zprobit"><em>Probit Regression</em></a>: <tt class="docutils literal"><span class="pre">zprobit$new()</span></tt></li>
<li><a class="reference internal" href="zelig-relogit.html#zrelogit"><em>Rare Events Logistic Regression</em></a>: <tt class="docutils literal"><span class="pre">zrelogit$new()</span></tt></li>
<li><a class="reference internal" href="zelig-tobit.html#ztobit"><em>Tobit Regression</em></a>: <tt class="docutils literal"><span class="pre">ztobit$new()</span></tt></li>
<li><a class="reference internal" href="zelig-weibull.html#zweibull"><em>Weibull Regression for Duration Dependent Variables</em></a>: <tt class="docutils literal"><span class="pre">zweibull$new()</span></tt></li>
<li><a class="reference internal" href="zelig-factorbayes.html#zfactorbayes"><em>Bayesian Factor Analysis</em></a>: <tt class="docutils literal"><span class="pre">zfactorbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-mlogitbayes.html#zmlogitbayes"><em>Bayesian Multinomial Logistic Regression</em></a>: <tt class="docutils literal"><span class="pre">zmlogitbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-oprobitbayes.html#zoprobitbayes"><em>Bayesian Ordered Probit Regression</em></a>: <tt class="docutils literal"><span class="pre">zoprobitbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-poissonbayes.html#zpoissonbayes"><em>Bayesian Poisson Regression</em></a>: <tt class="docutils literal"><span class="pre">zpoissonbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-probitbayes.html#zprobitbayes"><em>Bayesian Probit Regression</em></a>: <tt class="docutils literal"><span class="pre">zprobitbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-tobitbayes.html#ztobitbayes"><em>Bayesian Tobit Regression</em></a>: <tt class="docutils literal"><span class="pre">ztobitbayes$new()</span></tt></li>
<li><a class="reference internal" href="zelig-gammagee.html#zgammagee"><em>Generalized Estimating Equation for Gamma Regression</em></a>: <tt class="docutils literal"><span class="pre">zgammagee$new()</span></tt></li>
<li><a class="reference internal" href="zelig-probitgee.html#zprobitgee"><em>Generalized Estimating Equation for Probit Regression</em></a>: <tt class="docutils literal"><span class="pre">zprobitgee$new()</span></tt></li>
<li><a class="reference internal" href="zelig-poissongee.html#zpoissongee"><em>Generalized Estimating Equation for Poisson Regression</em></a>: <tt class="docutils literal"><span class="pre">zpoissongee$new()</span></tt></li>
</ul>
<hr class="docutils" />
</div>
<div class="section" id="zelig-exp">
<span id="zexp"></span><h2>zelig-exp<a class="headerlink" href="#zelig-exp" title="Permalink to this headline">¶</a></h2>
<p>Exponential Regression for Duration Dependent Variables</p>
<p>Use the exponential duration regression model if you have a dependent
variable representing a duration (time until an event). The model
assumes a constant hazard rate for all events. The dependent variable
may be censored (for observations have not yet been completed when data
were collected).</p>
<div class="section" id="syntax">
<h3>Syntax<a class="headerlink" href="#syntax" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zexp<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;exp&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>Exponential models require that the dependent variable be in the form
Surv(Y, C), where Y and C are vectors of length <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>. For each
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> in 1, …, <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>, the value <img class="math" src="_images/math/410a65f64e2a609a2a7f544ed72f0982b719807a.png" alt="y_i"/> is the
duration (lifetime, for example), and the associated <img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is a
binary variable such that <img class="math" src="_images/math/45bd383d229a85e1c894b645ee5ce5cfad178bdf.png" alt="c_i = 1"/> if the duration is not
censored (<em>e.g.</em>, the subject dies during the study) or <img class="math" src="_images/math/839e310a40128c2d71f158db2ee1ff3a1a40b81b.png" alt="c_i = 0"/>
if the duration is censored (<em>e.g.</em>, the subject is still alive at the
end of the study and is know to live at least as long as <img class="math" src="_images/math/410a65f64e2a609a2a7f544ed72f0982b719807a.png" alt="y_i"/>).
If <img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is omitted, all Y are assumed to be completed; that is,
time defaults to 1 for all observations.</p>
</div>
<div class="section" id="input-values">
<h3>Input Values<a class="headerlink" href="#input-values" title="Permalink to this headline">¶</a></h3>
<p>In addition to the standard inputs, zelig() takes the following
additional options for exponential regression:</p>
<ul class="simple">
<li>robust: defaults to FALSE. If TRUE, zelig() computes robust standard
errors based on sandwich estimators (see and ) and the options
selected in cluster.</li>
<li>cluster: if robust = TRUE, you may select a variable to define groups
of correlated observations. Let x3 be a variable that consists of
either discrete numeric values, character strings, or factors that
define strata. Then</li>
</ul>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x1 <span class="o">+</span> x2<span class="p">,</span> robust <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> cluster <span class="o">=</span> <span class="s">&quot;x3&quot;</span><span class="p">,</span>
                        model <span class="o">=</span> <span class="s">&quot;exp&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
</pre></div>
</div>
<p>means that the observations can be correlated within the strata
defined by the variable x3, and that robust standard errors should be
calculated according to those clusters. If robust = TRUE but cluster
is not specified, zelig() assumes that each observation falls into
its own cluster.</p>
</div>
<div class="section" id="example">
<h3>Example<a class="headerlink" href="#example" title="Permalink to this headline">¶</a></h3>
<p>Attach the sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>coalition<span class="p">)</span>
</pre></div>
</div>
<p>Estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>duration<span class="p">,</span> ciep12<span class="p">)</span> <span class="o">~</span> fract <span class="o">+</span> numst2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;exp&quot;</span><span class="p">,</span> data <span class="o">=</span> coalition<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Olivia Lau, Kosuke Imai, Gary King. 2011.</span>
<span class="c">##   exp: Exponential Regression for Duration Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>View the regression output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## survival::survreg(formula = Surv(duration, ciep12) ~ fract +</span>
<span class="c">##     numst2, data = ., dist = &quot;exponential&quot;, model = FALSE)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">##  (Intercept)        fract       numst2</span>
<span class="c">##  5.535872596 -0.003908965  0.461179302</span>
<span class="c">##</span>
<span class="c">## Scale fixed at 1</span>
<span class="c">##</span>
<span class="c">## Loglik(model)= -1077.4   Loglik(intercept only)= -1100.7</span>
<span class="c">##  Chisq= 46.66 on 2 degrees of freedom, p= 7.4e-11</span>
<span class="c">## n= 314</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the baseline values (with the ruling coalition in the minority) and
the alternative values (with the ruling coalition in the majority) for
X:</p>
<div class="highlight-r"><div class="highlight"><pre>x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate expected values and first differences:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.low<span class="p">,</span> x1 <span class="o">=</span> x.high<span class="p">)</span>
</pre></div>
</div>
<p>Summarize quantities of interest and produce some plots:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd     50%     2.5%    97.5%</span>
<span class="c">## 1 15.34602 1.523781 15.2551 12.63501 18.76127</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd     50%      2.5%    97.5%</span>
<span class="c">## [1,] 15.77413 15.24358 11.4823 0.3324345 53.15109</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 24.30661 1.967885 24.23044 20.61469 28.22129</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%      2.5%    97.5%</span>
<span class="c">## [1,] 23.82064 25.94461 15.01297 0.5677649 89.84064</span>
<span class="c">## fd</span>
<span class="c">##       mean       sd      50%     2.5%   97.5%</span>
<span class="c">## 1 8.960586 2.443173 8.937186 4.049825 13.8958</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-exp" src="_images/Zelig-exp-1.png" />
<p class="caption">Zelig-exp</p>
</div>
</div>
<div class="section" id="model">
<h3>Model<a class="headerlink" href="#model" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> be the survival time for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>. This
variable might be censored for some observations at a fixed time
<img class="math" src="_images/math/86ef86313546967dcb442753dcc6f50484dc244b.png" alt="y_c"/> such that the fully observed dependent variable,
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/>, is defined as</p>
<div class="math">
<p><img src="_images/math/080eb82277537c413153c5a513d4df9309471caf.png" alt="Y_i = \left\{ \begin{array}{ll}
      Y_i^* &amp; \textrm{if }Y_i^* \leq y_c \\
      y_c &amp; \textrm{if }Y_i^* &gt; y_c \\
    \end{array} \right."/></p>
</div><ul>
<li><p class="first">The <em>stochastic component</em> is described by the distribution of the
partially observed variable <img class="math" src="_images/math/e4b82f5d0d0b135c68864771e045dc913778a223.png" alt="Y^*"/>. We assume <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/>
follows the exponential distribution whose density function is given
by</p>
<div class="math">
<p><img src="_images/math/67e1264b7f56ebc4a97823f66b518116f44c057a.png" alt="f(y_i^*\mid \lambda_i) = \frac{1}{\lambda_i} \exp\left(-\frac{y_i^*}{\lambda_i}\right)"/></p>
</div><p>for <img class="math" src="_images/math/aee03fa2ac369970345e3f13ba80325460b429e3.png" alt="y_i^*\ge 0"/> and <img class="math" src="_images/math/7d72a5d0f43da45d3d7a997a9971becc5f8832c8.png" alt="\lambda_i&gt;0"/>. The mean of this
distribution is <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/>.</p>
<p>In addition, survival models like the exponential have three
additional properties. The hazard function <img class="math" src="_images/math/8f59c002cacf4a159c9d1b7ccfa141d30d5dc65c.png" alt="h(t)"/> measures the
probability of not surviving past time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> given survival up to
<img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. In general, the hazard function is equal to
<img class="math" src="_images/math/89ae40bbb08f26a2d7a02d4d5af4526f1a20c029.png" alt="f(t)/S(t)"/> where the survival function
<img class="math" src="_images/math/cc2e13874b1a2ecf7f79657e6c3a29960829dcc3.png" alt="S(t) = 1 - \int_{0}^t f(s) ds"/> represents the fraction still
surviving at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. The cumulative hazard function
<img class="math" src="_images/math/30319a828ec4029a092aefb503bd693e5776e59b.png" alt="H(t)"/> describes the probability of dying before time
<img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. In general,
<img class="math" src="_images/math/17b2212e68f15f537abbafa6fa716bb8369f7372.png" alt="H(t)= \int_{0}^{t} h(s) ds = -\log S(t)"/>. In the case of the
exponential model,</p>
<div class="math">
<p><img src="_images/math/01d104152a44c6ad669dfd89c7f40028ab4a15e8.png" alt="\begin{aligned}
h(t) &amp;=&amp; \frac{1}{\lambda_i} \\
S(t) &amp;=&amp; \exp\left( -\frac{t}{\lambda_i} \right) \\
H(t) &amp;=&amp; \frac{t}{\lambda_i}\end{aligned}"/></p>
</div><p>For the exponential model, the hazard function <img class="math" src="_images/math/8f59c002cacf4a159c9d1b7ccfa141d30d5dc65c.png" alt="h(t)"/> is
constant over time. The Weibull model and lognormal models allow the
hazard function to vary as a function of elapsed time (see and
respectively).</p>
</li>
<li><p class="first">The <em>systematic component</em> <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/> is modeled as</p>
<div class="math">
<p><img src="_images/math/2ef3e09dc6e0d4710d1d8455992bbb32d072fbb7.png" alt="\lambda_i = \exp(x_i \beta),"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of explanatory variables, and
<img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="quantities-of-interest">
<h3>Quantities of Interest<a class="headerlink" href="#quantities-of-interest" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) for the exponential model are simulations
of the expected duration given <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> and draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/>
from its posterior,</p>
<div class="math">
<p><img src="_images/math/ad77ed06fe3262275ae476c64b6707e4aee6e543.png" alt="E(Y) = \lambda_i = \exp(x_i \beta)."/></p>
</div></li>
<li><p class="first">The predicted values (qi$pr) are draws from the exponential
distribution with rate equal to the expected value.</p>
</li>
<li><p class="first">The first difference (or difference in expected values, qi$ev.diff),
is</p>
<div class="math">
<p><img src="_images/math/7631b945385f93d17fe8b3a5bf6e15c8143b2b49.png" alt="\textrm{FD} \; = \; E(Y \mid x_1) - E(Y \mid x),"/></p>
</div><p>where <img class="math" src="_images/math/188c175aac0a8a9c22499336711b5d7256407254.png" alt="x"/> and <img class="math" src="_images/math/b31b1ea8e4fe3f618dedfb0e2d8c69a3df7a391d.png" alt="x_1"/> are different vectors of values for
the explanatory variables.</p>
</li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/c086d795134e74f3fe435647603c4583f4d1dac3.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) - E[Y_i(t_i=0)]
  \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations is due to two
factors: uncertainty in the imputation process for censored
<img class="math" src="_images/math/9c2fbff8ba260029bc65603e6a06944c5746ad65.png" alt="y_i^*"/> and uncertainty in simulating <img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>,
the counterfactual expected value of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in
the treatment group, under the assumption that everything stays the
same except that the treatment indicator is switched to
<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/df8a385ab7b219b2fed30372b1381df5163a900e.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
  \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations is due to two
factors: uncertainty in the imputation process for censored
<img class="math" src="_images/math/9c2fbff8ba260029bc65603e6a06944c5746ad65.png" alt="y_i^*"/> and uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="output-values">
<h3>Output Values<a class="headerlink" href="#output-values" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(Surv(Y,</span> <span class="pre">C)</span> <span class="pre">~</span> <span class="pre">X,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">exp,</span> <span class="pre">data)</span></tt>, then you may
examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the coefficients by using z.out$coefficients, and
a default summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="see-also">
<h3>See also<a class="headerlink" href="#see-also" title="Permalink to this headline">¶</a></h3>
<p>The exponential function is part of the survival library by Terry
Therneau, ported to R by Thomas Lumley. Advanced users may wish to refer
to <tt class="docutils literal"><span class="pre">help(survfit)</span></tt> in the survival library.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-gamma">
<span id="zgamma"></span><h2>zelig-gamma<a class="headerlink" href="#zelig-gamma" title="Permalink to this headline">¶</a></h2>
<p>Gamma Regression for Continuous, Positive Dependent Variables</p>
<p>Use the gamma regression model if you have a positive-valued dependent
variable such as the number of years a parliamentary cabinet endures, or
the seconds you can stay airborne while jumping. The gamma distribution
assumes that all waiting times are complete by the end of the study
(censoring is not allowed).</p>
<div class="section" id="id1">
<h3>Syntax<a class="headerlink" href="#id1" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zgamma<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">,</span> x1 <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id2">
<h3>Example<a class="headerlink" href="#id2" title="Permalink to this headline">¶</a></h3>
<p>Attach the sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>coalition<span class="p">)</span>
</pre></div>
</div>
<p>Estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>duration <span class="o">~</span> fract <span class="o">+</span> numst2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma&quot;</span><span class="p">,</span> data <span class="o">=</span> coalition<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2007.</span>
<span class="c">##   gamma: Gamma Regression for Continuous, Positive Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>View the regression output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  stats::glm(formula = duration ~ fract + numst2, family = Gamma(&quot;inverse&quot;),</span>
<span class="c">##     data = .)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)        fract       numst2</span>
<span class="c">##  -0.0129597    0.0001149   -0.0173875</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 313 Total (i.e. Null);  311 Residual</span>
<span class="c">## Null Deviance:       300.7</span>
<span class="c">## Residual Deviance: 272.2         AIC: 2428</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the baseline values (with the ruling coalition in the minority) and
the alternative values (with the ruling coalition in the majority) for
X:</p>
<div class="highlight-r"><div class="highlight"><pre>x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate expected values (qi$ev) and first differences (qi$fd):</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.low<span class="p">,</span> x1 <span class="o">=</span> x.high<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##         mean       sd      50%     2.5%   97.5%</span>
<span class="c">## [1,] 14.4723 1.049237 14.44492 12.65866 16.6945</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%      2.5%    97.5%</span>
<span class="c">## [1,] 14.75237 13.72097 10.66599 0.7613514 51.32591</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean       sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 19.22903 1.168735 19.13605 17.23567 21.78014</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%     2.5%  97.5%</span>
<span class="c">## [1,] 19.45355 15.85049 14.89541 1.237274 59.244</span>
<span class="c">## fd</span>
<span class="c">##          mean       sd      50%    2.5%    97.5%</span>
<span class="c">## [1,] 4.756728 1.596003 4.739692 1.67932 8.005434</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-gamma" src="_images/Zelig-gamma-1.png" />
<p class="caption">Zelig-gamma</p>
</div>
</div>
<div class="section" id="id3">
<h3>Model<a class="headerlink" href="#id3" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The Gamma distribution with scale parameter <img class="math" src="_images/math/ad59b6e24a4a00ac621801f8d7513d68be654ab5.png" alt="\alpha"/> has a
<em>stochastic component</em>:</p>
<div class="math">
<p><img src="_images/math/05d44853b67b61aea8416dec47c4bb271c39feee.png" alt="\begin{aligned}
Y &amp;\sim&amp; \textrm{Gamma}(y_i \mid \lambda_i, \alpha) \\
f(y)  &amp;=&amp; \frac{1}{\alpha^{\lambda_i} \, \Gamma \lambda_i} \, y_i^{\lambda_i
  - 1} \exp -\left\{ \frac{y_i}{\alpha} \right\}\end{aligned}"/></p>
</div><div class="line-block">
<div class="line">for <img class="math" src="_images/math/d4a2907c3d8ac12950c8e344f35236de63f31929.png" alt="\alpha, \lambda_i, y_i &gt; 0"/>.</div>
</div>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/d89022e0c4ea97317e8099dc27fd004602d759b8.png" alt="\lambda_i = \frac{1}{x_i \beta}"/></p>
</div></li>
</ul>
</div>
<div class="section" id="id4">
<h3>Quantities of Interest<a class="headerlink" href="#id4" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) are simulations of the mean of the
stochastic component given draws of <img class="math" src="_images/math/ad59b6e24a4a00ac621801f8d7513d68be654ab5.png" alt="\alpha"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/>
from their posteriors:</p>
<div class="math">
<p><img src="_images/math/c59d39819f033b5ae240c27b97121ccceb2b3a82.png" alt="E(Y) = \alpha \lambda_i."/></p>
</div></li>
<li><p class="first">The predicted values (qi$pr) are draws from the gamma distribution
for each given set of parameters <img class="math" src="_images/math/d625e4d40861475147286b5a40866be5ac03ff8c.png" alt="(\alpha, \lambda_i)"/>.</p>
</li>
<li><p class="first">If x1 is specified, sim() also returns the differences in the
expected values (qi$fd),</p>
<div class="math">
<p><img src="_images/math/15f1a5b5e19ffd4d3b304de7da740b451173e4f9.png" alt="E(Y \mid x_1) -
  E(Y \mid x)"/></p>
</div><p>.</p>
</li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id5">
<h3>Output Values<a class="headerlink" href="#id5" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">gamma,</span> <span class="pre">data)</span></tt>, then you may examine the
available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the
coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id6">
<h3>See also<a class="headerlink" href="#id6" title="Permalink to this headline">¶</a></h3>
<p>The gamma model is part of the stats package. Advanced users may
wish to refer to <tt class="docutils literal"><span class="pre">help(glm)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-logit">
<span id="zlogit"></span><h2>zelig-logit<a class="headerlink" href="#zelig-logit" title="Permalink to this headline">¶</a></h2>
<p>Logistic Regression for Dichotomous Dependent Variables</p>
<p>Logistic regression specifies a dichotomous dependent variable as a
function of a set of explanatory variables.</p>
<div class="section" id="id7">
<h3>Syntax<a class="headerlink" href="#id7" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zlogit<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;logit&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">,</span> x1 <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="examples">
<h3>Examples<a class="headerlink" href="#examples" title="Permalink to this headline">¶</a></h3>
<div class="section" id="basic-example">
<h4>Basic Example<a class="headerlink" href="#basic-example" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample turnout dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>turnout<span class="p">)</span>
</pre></div>
</div>
<p>Estimating parameter values for the logistic regression:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out1 <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> age <span class="o">+</span> race<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;logit&quot;</span><span class="p">,</span> data <span class="o">=</span> turnout<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2007.</span>
<span class="c">##   logit: Logistic Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out1 <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> age <span class="o">=</span> <span class="m">36</span><span class="p">,</span> race <span class="o">=</span> <span class="s">&quot;white&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##         mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.74836 0.01146077 0.7485047 0.7262389 0.7702287</span>
<span class="c">## pv</span>
<span class="c">##         0    1</span>
<span class="c">## [1,] 0.23 0.77</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-logit-1" src="_images/Zelig-logit-1-1.png" />
<p class="caption">Zelig-logit-1</p>
</div>
</div>
<div class="section" id="simulating-first-differences">
<h4>Simulating First Differences<a class="headerlink" href="#simulating-first-differences" title="Permalink to this headline">¶</a></h4>
<p>Estimating the risk difference (and risk ratio) between low education
(25th percentile) and high education (75th percentile) while all the
other variables held at their default values.</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2 <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> race <span class="o">+</span> educate<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;logit&quot;</span><span class="p">,</span> data <span class="o">=</span> turnout<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2007.</span>
<span class="c">##   logit: Logistic Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.75</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.25</span><span class="p">))</span>
s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out2<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##           mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.8226229 0.01060798 0.8228707 0.8019673 0.8432182</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.172 0.828</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.709421 0.01340055 0.7103348 0.6810109 0.7344127</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.308 0.692</span>
<span class="c">## fd</span>
<span class="c">##            mean         sd        50%       2.5%       97.5%</span>
<span class="c">## [1,] -0.1132019 0.01160019 -0.1135177 -0.1363117 -0.08977958</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-logit-2" src="_images/Zelig-logit-2-1.png" />
<p class="caption">Zelig-logit-2</p>
</div>
</div>
</div>
<div class="section" id="id8">
<h3>Model<a class="headerlink" href="#id8" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> be the binary dependent variable for observation
<img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> which takes the value of either 0 or 1.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/beb30650790678aa2d5315baf665f9effc8ac3ad.png" alt="\begin{aligned}
Y_i &amp;\sim&amp; \textrm{Bernoulli}(y_i \mid \pi_i) \\
    &amp;=&amp; \pi_i^{y_i} (1-\pi_i)^{1-y_i}\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/204ad227887156eca0ea3882997e3c5d0703174b.png" alt="\pi_i=\Pr(Y_i=1)"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by:</p>
<div class="math">
<p><img src="_images/math/59cf11e927045c99a310cdaa15d32bc7cfacb160.png" alt="\pi_i \; = \; \frac{1}{1 + \exp(-x_i \beta)}."/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id9">
<h3>Quantities of Interest<a class="headerlink" href="#id9" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) for the logit model are simulations of
the predicted probability of a success:</p>
<div class="math">
<p><img src="_images/math/bd411b566ef1222a52b2c10e6eca8f7f994040d4.png" alt="E(Y) =
  \pi_i= \frac{1}{1 + \exp(-x_i \beta)},"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution.</p>
</li>
<li><p class="first">The predicted values (qi$pr) are draws from the Binomial distribution
with mean equal to the simulated expected value <img class="math" src="_images/math/916754f7e7de4247fa27c52f610948b4055311fe.png" alt="\pi_i"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) for the logit model is defined as</p>
<div class="math">
<p><img src="_images/math/8f669920018c72d7f23a4ba843925381c4703f17.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1) - \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">The risk ratio (qi$rr) is defined as</p>
<div class="math">
<p><img src="_images/math/8540cd3e413420e1eceaa11af2b8fc70ef717f37.png" alt="\textrm{RR} = \Pr(Y = 1 \mid x_1) \ / \ \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/77d8002d68a309fc60d4b182880a2dbe1cde6c81.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)}\right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id10">
<h3>Output Values<a class="headerlink" href="#id10" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">logit,</span> <span class="pre">data)</span></tt>, then you may examine the
available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the
coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id11">
<h3>See also<a class="headerlink" href="#id11" title="Permalink to this headline">¶</a></h3>
<p>The logit model is part of the stats package. Advanced users may
wish to refer to <tt class="docutils literal"><span class="pre">help(glm)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-lognorm">
<span id="zlognorm"></span><h2>zelig-lognorm<a class="headerlink" href="#zelig-lognorm" title="Permalink to this headline">¶</a></h2>
<p>Log-Normal Regression for Duration Dependent Variables</p>
<p>The log-normal model describes an event’s duration, the dependent
variable, as a function of a set of explanatory variables. The
log-normal model may take time censored dependent variables, and allows
the hazard rate to increase and decrease.</p>
<div class="section" id="id12">
<h3>Syntax<a class="headerlink" href="#id12" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zlognorm<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zlognorm<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;lognorm&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>Log-normal models require that the dependent variable be in the form
Surv(Y, C), where Y and C are vectors of length <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>. For each
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> in 1, …, <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>, the value <img class="math" src="_images/math/410a65f64e2a609a2a7f544ed72f0982b719807a.png" alt="y_i"/> is the
duration (lifetime, for example) of each subject, and the associated
<img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is a binary variable such that <img class="math" src="_images/math/45bd383d229a85e1c894b645ee5ce5cfad178bdf.png" alt="c_i = 1"/> if the
duration is not censored (<em>e.g.</em>, the subject dies during the study) or
<img class="math" src="_images/math/839e310a40128c2d71f158db2ee1ff3a1a40b81b.png" alt="c_i = 0"/> if the duration is censored (<em>e.g.</em>, the subject is
still alive at the end of the study). If <img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is omitted, all Y
are assumed to be completed; that is, time defaults to 1 for all
observations.</p>
</div>
<div class="section" id="id13">
<h3>Input Values<a class="headerlink" href="#id13" title="Permalink to this headline">¶</a></h3>
<p>In addition to the standard inputs, zelig() takes the following
additional options for lognormal regression:</p>
<ul class="simple">
<li>robust: defaults to FALSE. If TRUE, zelig() computes robust standard
errors based on sandwich estimators (see and ) based on the options
in cluster.</li>
<li>cluster: if robust = TRUE, you may select a variable to define groups
of correlated observations. Let x3 be a variable that consists of
either discrete numeric values, character strings, or factors that
define strata. Then</li>
</ul>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x1 <span class="o">+</span> x2<span class="p">,</span> robust <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> cluster <span class="o">=</span> <span class="s">&quot;x3&quot;</span><span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;exp&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
</pre></div>
</div>
<p>means that the observations can be correlated within the strata
defined by the variable x3, and that robust standard errors should be
calculated according to those clusters. If robust = TRUE but cluster
is not specified, zelig() assumes that each observation falls into
its own cluster.</p>
</div>
<div class="section" id="id14">
<h3>Example<a class="headerlink" href="#id14" title="Permalink to this headline">¶</a></h3>
<p>Attach the sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>coalition<span class="p">)</span>
</pre></div>
</div>
<p>Estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>duration<span class="p">,</span> ciep12<span class="p">)</span> <span class="o">~</span> fract <span class="o">+</span> numst2<span class="p">,</span> model <span class="o">=</span><span class="s">&quot;lognorm&quot;</span><span class="p">,</span>  data <span class="o">=</span> coalition<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Matthew Owen, Olivia Lau, Kosuke Imai, Gary King. 2007.</span>
<span class="c">##   lognorm: Log-Normal Regression for Duration Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>View the regression output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## survival::survreg(formula = Surv(duration, ciep12) ~ fract +</span>
<span class="c">##     numst2, data = ., dist = &quot;lognormal&quot;, model = FALSE)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)       fract      numst2</span>
<span class="c">##  5.36666977 -0.00443755  0.55983251</span>
<span class="c">##</span>
<span class="c">## Scale= 1.20008</span>
<span class="c">##</span>
<span class="c">## Loglik(model)= -1077.9   Loglik(intercept only)= -1101.2</span>
<span class="c">##  Chisq= 46.58 on 2 degrees of freedom, p= 7.7e-11</span>
<span class="c">## n= 314</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the baseline values (with the ruling coalition in the minority) and
the alternative values (with the ruling coalition in the majority) for
X:</p>
<div class="highlight-r"><div class="highlight"><pre>x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2<span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate expected values (qi$ev) and first differences (qi$fd):</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.low<span class="p">,</span> x1 <span class="o">=</span> x.high<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd      50%    2.5%    97.5%</span>
<span class="c">## 1 18.31304 2.360751 18.03655 14.3681 23.58271</span>
<span class="c">## pv</span>
<span class="c">##       mean       sd      50%    2.5%    97.5%</span>
<span class="c">## 1 18.31304 2.360751 18.03655 14.3681 23.58271</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 31.97293 3.632501 31.66406 25.67784 39.67214</span>
<span class="c">## pv</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 31.97293 3.632501 31.66406 25.67784 39.67214</span>
<span class="c">## fd</span>
<span class="c">##       mean       sd     50%     2.5%    97.5%</span>
<span class="c">## 1 13.65989 3.629384 13.4982 6.797792 21.28857</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-lognorm" src="_images/Zelig-lognorm-1.png" />
<p class="caption">Zelig-lognorm</p>
</div>
</div>
<div class="section" id="id15">
<h3>Model<a class="headerlink" href="#id15" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> be the survival time for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> with
the density function <img class="math" src="_images/math/c52f3efaf426aaf75ff6fea4bb823873865b1303.png" alt="f(y)"/> and the corresponding distribution
function <img class="math" src="_images/math/39c3fab3abd2a55f823cd0336db679ce1b36eead.png" alt="F(t)=\int_{0}^t f(y) dy"/>. This variable might be
censored for some observations at a fixed time <img class="math" src="_images/math/86ef86313546967dcb442753dcc6f50484dc244b.png" alt="y_c"/> such that the
fully observed dependent variable, <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/>, is defined as</p>
<div class="math">
<p><img src="_images/math/080eb82277537c413153c5a513d4df9309471caf.png" alt="Y_i = \left\{ \begin{array}{ll}
      Y_i^* &amp; \textrm{if }Y_i^* \leq y_c \\
      y_c &amp; \textrm{if }Y_i^* &gt; y_c \\
    \end{array} \right."/></p>
</div><ul>
<li><p class="first">The <em>stochastic component</em> is described by the distribution of the
partially observed variable, <img class="math" src="_images/math/e4b82f5d0d0b135c68864771e045dc913778a223.png" alt="Y^*"/>. For the lognormal model,
there are two equivalent representations:</p>
<div class="math">
<p><img src="_images/math/2d2f0134ef7cd6c7d89866a45b1771b6a1fde79b.png" alt="\begin{aligned}
    Y_i^* \; \sim \; \textrm{LogNormal}(\mu_i, \sigma^2) &amp; \textrm{ or
} &amp; \log(Y_i^*) \; \sim \; \textrm{Normal}(\mu_i, \sigma^2)\end{aligned}"/></p>
</div><p>where the parameters <img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> and <img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/> are the mean
and variance of the Normal distribution. (Note that the output from
zelig() parameterizes scale:math:` = sigma`.)</p>
<p>In addition, survival models like the lognormal have three additional
properties. The hazard function <img class="math" src="_images/math/8f59c002cacf4a159c9d1b7ccfa141d30d5dc65c.png" alt="h(t)"/> measures the probability
of not surviving past time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> given survival up to <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>.
In general, the hazard function is equal to <img class="math" src="_images/math/89ae40bbb08f26a2d7a02d4d5af4526f1a20c029.png" alt="f(t)/S(t)"/> where
the survival function <img class="math" src="_images/math/967eb1670359684fab112e11f3a002670dbbff45.png" alt="S(t) =1 - \int_{0}^t f(s) ds"/> represents the fraction still surviving at
time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. The cumulative hazard function <img class="math" src="_images/math/30319a828ec4029a092aefb503bd693e5776e59b.png" alt="H(t)"/> describes
the probability of dying before time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. In general,
<img class="math" src="_images/math/a951cf862eec77b95758fc88815c3addfff6ba11.png" alt="H(t)=
\int_{0}^{t} h(s) ds = -\log S(t)"/>. In the case of the lognormal
model,</p>
<div class="math">
<p><img src="_images/math/deea3408797487ffda9af0b8e8b0ea0b87db7ef5.png" alt="\begin{aligned}
h(t) &amp;=&amp; \frac{1}{\sqrt{2 \pi} \, \sigma t \, S(t)}
\exp\left\{-\frac{1}{2 \sigma^2} (\log \lambda t)^2\right\} \\
S(t) &amp;=&amp; 1 - \Phi\left(\frac{1}{\sigma} \log \lambda t\right) \\
H(t) &amp;=&amp; -\log \left\{ 1 - \Phi\left(\frac{1}{\sigma} \log \lambda t\right) \right\}\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/262565daee431ce18fed8cbe6ff35db082bb6a9b.png" alt="\Phi(\cdot)"/> is the cumulative density function for the
Normal distribution.</p>
</li>
<li><p class="first">The <em>systematic component</em> is described as:</p>
<div class="math">
<p><img src="_images/math/41364463bbef7e4f118743532ee95c3ebf4ba7a3.png" alt="\mu_i = x_i \beta ."/></p>
</div></li>
</ul>
</div>
<div class="section" id="id16">
<h3>Quantities of Interest<a class="headerlink" href="#id16" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) for the lognormal model are simulations
of the expected duration:</p>
<div class="math">
<p><img src="_images/math/31ea238b8fad5a97c190036baed367c1644d71d9.png" alt="E(Y) =  \exp\left(\mu_i + \frac{1}{2}\sigma^2 \right),"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and <img class="math" src="_images/math/2298cf1485084afe72757a9c8483af49a138d81f.png" alt="\sigma"/> from their sampling
distributions.</p>
</li>
<li><p class="first">The predicted value is a draw from the log-normal distribution given
simulations of the parameters <img class="math" src="_images/math/212bf4406b9299a69a57ea55bc3cf9ccdb9489ad.png" alt="(\lambda_i, \sigma)"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) is</p>
<div class="math">
<p><img src="_images/math/47e62bb3ada6780d6fd2b247bf221d015d2207d2.png" alt="\textrm{FD} = E(Y \mid x_1) - E(Y \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/a42073a06e5f64918f546d89584110fe35c0e21d.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \{ Y_i(t_i=1) - E[Y_i(t_i=0)] \},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations is due to two
factors: uncertainty in the imputation process for censored
<img class="math" src="_images/math/9c2fbff8ba260029bc65603e6a06944c5746ad65.png" alt="y_i^*"/> and uncertainty in simulating <img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>,
the counterfactual expected value of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in
the treatment group, under the assumption that everything stays the
same except that the treatment indicator is switched to
<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/db43e079e29b09aa583b792c735eafc894037a2a.png" alt="\frac{1}{\sum_{i=1}^n t_i} \sum_{i:t_i=1}^n \{  Y_i(t_i=1) -
\widehat{Y_i(t_i=0)} \},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations are due to two
factors: uncertainty in the imputation process for censored
<img class="math" src="_images/math/9c2fbff8ba260029bc65603e6a06944c5746ad65.png" alt="y_i^*"/> and uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id17">
<h3>Output Values<a class="headerlink" href="#id17" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(Surv(Y,</span> <span class="pre">C)</span> <span class="pre">~</span> <span class="pre">X,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">lognorm,</span> <span class="pre">data)</span></tt>, then you may
examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the coefficients by using z.out$coefficients, and
a default summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id18">
<h3>See also<a class="headerlink" href="#id18" title="Permalink to this headline">¶</a></h3>
<p>The exponential function is part of the survival library by by Terry
Therneau, ported to R by Thomas Lumley. Advanced users may wish to refer
to <tt class="docutils literal"><span class="pre">help(survfit)</span></tt> in the survival library.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-ls">
<span id="zls"></span><h2>zelig-ls<a class="headerlink" href="#zelig-ls" title="Permalink to this headline">¶</a></h2>
<p>Least Squares Regression for Continuous Dependent Variables</p>
<p>Use least squares regression analysis to estimate the best linear
predictor for the specified dependent variables.</p>
<div class="section" id="id19">
<h3>Syntax<a class="headerlink" href="#id19" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zls<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;ls&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id20">
<h3>Examples<a class="headerlink" href="#id20" title="Permalink to this headline">¶</a></h3>
<div class="section" id="basic-example-with-first-differences">
<h4>Basic Example with First Differences<a class="headerlink" href="#basic-example-with-first-differences" title="Permalink to this headline">¶</a></h4>
<p>Attach sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>macro<span class="p">)</span>
</pre></div>
</div>
<p>Estimate model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out1 <span class="o">&lt;-</span> zelig<span class="p">(</span>unem <span class="o">~</span> gdp <span class="o">+</span> capmob <span class="o">+</span> trade<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;ls&quot;</span><span class="p">,</span> data <span class="o">=</span> macro<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, and Olivia Lau. 2007.</span>
<span class="c">##   ls: Least Squares Regression for Continuous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Summarize regression coefficients:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## stats::lm(formula = unem ~ gdp + capmob + trade, data = .)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)          gdp       capmob        trade</span>
<span class="c">##     6.18129     -0.32360      1.42194      0.01985</span>
<span class="c">##</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set explanatory variables to their default (mean/mode) values, with
high (80th percentile) and low (20th percentile) values for the trade
variable:</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> trade <span class="o">=</span> quantile<span class="p">(</span>macro<span class="o">$</span>trade<span class="p">,</span> <span class="m">0.8</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> trade <span class="o">=</span> quantile<span class="p">(</span>macro<span class="o">$</span>trade<span class="p">,</span> <span class="m">0.2</span><span class="p">))</span>
</pre></div>
</div>
<p>Generate first differences for the effect of high versus low trade on GDP:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean        sd      50%     2.5%    97.5%</span>
<span class="c">## 1 5.430943 0.1889071 5.432185 5.064686 5.813209</span>
<span class="c">## pv</span>
<span class="c">##       mean        sd      50%     2.5%    97.5%</span>
<span class="c">## 1 5.430943 0.1889071 5.432185 5.064686 5.813209</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 4.602064 0.182975 4.595302 4.258268 4.962241</span>
<span class="c">## pv</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 4.602064 0.182975 4.595302 4.258268 4.962241</span>
<span class="c">## fd</span>
<span class="c">##         mean        sd        50%     2.5%     97.5%</span>
<span class="c">## 1 -0.8288789 0.2358199 -0.8343236 -1.30502 -0.382212</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="" src="_images/unnamed-chunk-10-1.png" />
</div>
</div>
<div class="section" id="using-dummy-variables">
<h4>Using Dummy Variables<a class="headerlink" href="#using-dummy-variables" title="Permalink to this headline">¶</a></h4>
<p>Estimate a model with fixed effects for each country (see for help
with dummy variables). Note that you do not need to create dummy
variables, as the program will automatically parse the unique values
in the selected variable into discrete levels.</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2 <span class="o">&lt;-</span> zelig<span class="p">(</span>unem <span class="o">~</span> gdp <span class="o">+</span> trade <span class="o">+</span> capmob <span class="o">+</span> as.factor<span class="p">(</span>country<span class="p">),</span> model <span class="o">=</span> <span class="s">&quot;ls&quot;</span><span class="p">,</span> data <span class="o">=</span> macro<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, and Olivia Lau. 2007.</span>
<span class="c">##   ls: Least Squares Regression for Continuous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Set values for the explanatory variables, using the default mean/mode
values, with country set to the United States and Japan,
respectively:</p>
<div class="highlight-r"><div class="highlight"><pre>x.US <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">,</span> country <span class="o">=</span> <span class="s">&quot;United States&quot;</span><span class="p">)</span>
x.Japan <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">,</span> country <span class="o">=</span> <span class="s">&quot;Japan&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out2<span class="p">,</span> x <span class="o">=</span> x.US<span class="p">,</span> x1 <span class="o">=</span> x.Japan<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="" src="_images/unnamed-chunk-14-1.png" />
</div>
</div>
</div>
<div class="section" id="id21">
<h3>Model<a class="headerlink" href="#id21" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The <em>stochastic component</em> is described by a density with mean
<img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> and the common variance <img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/></p>
<div class="math">
<p><img src="_images/math/34d94a6f5d061597bbcec0b62d41637f1106c37d.png" alt="Y_i \; \sim \; f(y_i \mid \mu_i, \sigma^2)."/></p>
</div></li>
<li><p class="first">The <em>systematic component</em> models the conditional mean as</p>
<div class="math">
<p><img src="_images/math/e9096bd766e29271a33d6f06c16ab7464b39dd16.png" alt="\mu_i =  x_i \beta"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of covariates, and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is
the vector of coefficients.</p>
<p>The least squares estimator is the best linear predictor of a
dependent variable given <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/>, and minimizes the sum of
squared residuals, <img class="math" src="_images/math/26c72a7cbda08b0d6d18fbed4a88be6d712a98bc.png" alt="\sum_{i=1}^n (Y_i-x_i \beta)^2"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id22">
<h3>Quantities of Interest<a class="headerlink" href="#id22" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected value (qi$ev) is the mean of simulations from the
stochastic component,</p>
<div class="math">
<p><img src="_images/math/6742e8f4c9da4bc790debe293a4448142af6d624.png" alt="E(Y) = x_i \beta,"/></p>
</div><p>given a draw of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution.</p>
</li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id23">
<h3>Output Values<a class="headerlink" href="#id23" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">ls,</span> <span class="pre">data)</span></tt>, then you may examine the
available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the
coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements available through
the $ operator are listed below.</p>
<ul>
<li><p class="first">From the zelig() output object z.out, you may extract:</p>
<ul class="simple">
<li>coefficients: parameter estimates for the explanatory variables.</li>
<li>residuals: the working residuals in the final iteration of the
IWLS fit.</li>
<li>fitted.values: fitted values.</li>
<li>df.residual: the residual degrees of freedom.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
</ul>
</li>
<li><p class="first">From summary(z.out), you may extract:</p>
<ul>
<li><p class="first">coefficients: the parameter estimates with their associated
standard errors, <img class="math" src="_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/>-values, and <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>-statistics.</p>
<div class="math">
<p><img src="_images/math/e58a382501917d62fae84d66164159e3f1b6c2ee.png" alt="\hat{\beta} \; = \; \left(\sum_{i=1}^n x_i' x_i\right)^{-1} \sum x_i y_i"/></p>
</div></li>
<li><p class="first">sigma: the square root of the estimate variance of the random
error <img class="math" src="_images/math/630e3a780577ea7921e81ee2ac5237dd1802ec8d.png" alt="e"/>:</p>
<div class="math">
<p><img src="_images/math/ffc81e7e9c9aac1ae3fdf35621486649779a8ed9.png" alt="\hat{\sigma} \; = \; \frac{\sum (Y_i-x_i\hat{\beta})^2}{n-k}"/></p>
</div></li>
<li><p class="first">r.squared: the fraction of the variance explained by the model.</p>
<div class="math">
<p><img src="_images/math/00412d048f95c80e8dd72b1e864c6bda22059802.png" alt="R^2 \; = \; 1 - \frac{\sum (Y_i-x_i\hat{\beta})^2}{\sum (y_i -
         \bar{y})^2}"/></p>
</div></li>
<li><p class="first">adj.r.squared: the above <img class="math" src="_images/math/4f58a3b2897df3c0b9d53a442ae896516ecdeaa0.png" alt="R^2"/> statistic, penalizing for an
increased number of explanatory variables.</p>
</li>
<li><p class="first">cov.unscaled: a <img class="math" src="_images/math/3908ba50237c94909083edd0295cd1c9345b0a8c.png" alt="k \times k"/> matrix of unscaled covariances.</p>
</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id24">
<h3>See also<a class="headerlink" href="#id24" title="Permalink to this headline">¶</a></h3>
<p>The least squares regression is part of the stats package by William N.
Venables and Brian D. Ripley .In addition, advanced users may wish to
refer to <tt class="docutils literal"><span class="pre">help(lm)</span></tt> and <tt class="docutils literal"><span class="pre">help(lm.fit)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-negbin">
<span id="znegbin"></span><h2>zelig-negbin<a class="headerlink" href="#zelig-negbin" title="Permalink to this headline">¶</a></h2>
<p>Negative Binomial Regression for Event Count Dependent Variables</p>
<p>Use the negative binomial regression if you have a count of events for
each observation of your dependent variable. The negative binomial model
is frequently used to estimate over-dispersed event count models.</p>
<div class="section" id="id25">
<h3>Syntax<a class="headerlink" href="#id25" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> znegbin<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;negbin&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id26">
<h3>Example<a class="headerlink" href="#id26" title="Permalink to this headline">¶</a></h3>
<p>Load sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>sanction<span class="p">)</span>
</pre></div>
</div>
<p>Estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>num <span class="o">~</span> target <span class="o">+</span> coop<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;negbin&quot;</span><span class="p">,</span> data <span class="o">=</span> sanction<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2008.</span>
<span class="c">##   negbinom: Negative Binomial Regression for Event Count Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  MASS::glm.nb(formula = num ~ target + coop, data = ., init.theta = 1.841603403,</span>
<span class="c">##     link = log)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)       target         coop</span>
<span class="c">##      -1.564        0.151        1.286</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 77 Total (i.e. Null);  75 Residual</span>
<span class="c">## Null Deviance:       237.1</span>
<span class="c">## Residual Deviance: 56.55         AIC: 360.2</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set values for the explanatory variables to their default mean values:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulate fitted values:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd    50%    2.5%    97.5%</span>
<span class="c">## [1,] 2.982169 0.3553061 2.9551 2.34326 3.777098</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">##     0     1     2     3     4     5     6     7     8     9    10    11</span>
<span class="c">## 0.162 0.199 0.200 0.118 0.108 0.068 0.048 0.038 0.020 0.011 0.010 0.006</span>
<span class="c">##    12    14    15    16    18</span>
<span class="c">## 0.008 0.001 0.001 0.001 0.001</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-negbin" src="_images/Zelig-negbin-1.png" />
<p class="caption">Zelig-negbin</p>
</div>
</div>
<div class="section" id="id27">
<h3>Model<a class="headerlink" href="#id27" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> be the number of independent events that occur during a
fixed time period. This variable can take any non-negative integer
value.</p>
<ul>
<li><p class="first">The negative binomial distribution is derived by letting the mean of
the Poisson distribution vary according to a fixed parameter
<img class="math" src="_images/math/1a534f73d7999c523b01512801d3bda18adee3b7.png" alt="\zeta"/> given by the Gamma distribution. The <em>stochastic
component</em> is given by</p>
<div class="math">
<p><img src="_images/math/bd08836a98d9156598663d31c4c76c26581aab11.png" alt="\begin{aligned}
     Y_i \mid \zeta_i &amp; \sim &amp; \textrm{Poisson}(\zeta_i \mu_i),\\
     \zeta_i &amp; \sim &amp; \frac{1}{\theta}\textrm{Gamma}(\theta).
   \end{aligned}"/></p>
</div><p>The marginal distribution of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> is then the negative
binomial with mean <img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> and variance
<img class="math" src="_images/math/4b5943fd01fbda26432b4db9936abed9fb52cc64.png" alt="\mu_i + \mu_i^2/\theta"/>:</p>
<div class="math">
<p><img src="_images/math/bfcce2bd73058487e73f11f41d0a02f9837e2ec3.png" alt="\begin{aligned}
   Y_i &amp; \sim &amp; \textrm{NegBin}(\mu_i, \theta), \\
       &amp; = &amp; \frac{\Gamma (\theta + y_i)}{y! \, \Gamma(\theta)}
             \frac{\mu_i^{y_i} \, \theta^{\theta}}{(\mu_i + \theta)^{\theta + y_i}},
   \end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/a9cfbeb8ebee1f365919e147a79e242dcb67ee5d.png" alt="\theta"/> is the systematic parameter of the Gamma
distribution modeling <img class="math" src="_images/math/b2c74aae8f9d0cb0072e974b47723f2e5521b401.png" alt="\zeta_i"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/81474e43896f32f2f86670eec250ff10b8653b09.png" alt="\mu_i = \exp(x_i \beta)"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id28">
<h3>Quantities of Interest<a class="headerlink" href="#id28" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) are simulations of the mean of the
stochastic component. Thus,</p>
<div class="math">
<p><img src="_images/math/ed68ddf7d6e8d78ed9fb9a9bb55febcf8f71d49e.png" alt="E(Y) = \mu_i = \exp(x_i
  \beta),"/></p>
</div><p>given simulations of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/>.</p>
</li>
<li><p class="first">The predicted value (qi$pr) drawn from the distribution defined by
the set of parameters <img class="math" src="_images/math/9054fec1d747aca9cfcaa9d679834cf2fd26b899.png" alt="(\mu_i, \theta)"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) is</p>
<div class="math">
<p><img src="_images/math/82a49121147fab5307d7df11a6b69165b3f6794c.png" alt="\textrm{FD} \; = \; E(Y | x_1) - E(Y \mid x)"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id29">
<h3>Output Values<a class="headerlink" href="#id29" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">negbin,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id30">
<h3>See also<a class="headerlink" href="#id30" title="Permalink to this headline">¶</a></h3>
<p>The negative binomial model is part of the MASS package by William N.
Venable and Brian D. Ripley . Advanced users may wish to refer to
<a href="#id31"><span class="problematic" id="id32">``</span></a>help(glm.nb)`.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-normal">
<span id="znorm"></span><h2>zelig-normal<a class="headerlink" href="#zelig-normal" title="Permalink to this headline">¶</a></h2>
<p>Normal Regression for Continuous Dependent Variables</p>
<p>The Normal regression model is a close variant of the more standard
least squares regression model (see ). Both models specify a continuous
dependent variable as a linear function of a set of explanatory
variables. The Normal model reports maximum likelihood (rather than
least squares) estimates. The two models differ only in their estimate
for the stochastic parameter <img class="math" src="_images/math/2298cf1485084afe72757a9c8483af49a138d81f.png" alt="\sigma"/>.</p>
<div class="section" id="id33">
<h3>Syntax<a class="headerlink" href="#id33" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> znormal<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;normal&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id34">
<h3>Examples<a class="headerlink" href="#id34" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id35">
<h4>Basic Example with First Differences<a class="headerlink" href="#id35" title="Permalink to this headline">¶</a></h4>
<p>Attach sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>macro<span class="p">)</span>
</pre></div>
</div>
<p>Estimate model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out1 <span class="o">&lt;-</span> zelig<span class="p">(</span>unem <span class="o">~</span> gdp <span class="o">+</span> capmob <span class="o">+</span> trade<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;normal&quot;</span><span class="p">,</span> data <span class="o">=</span> macro<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2008.</span>
<span class="c">##   normal: Normal Regression for Continuous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Summarize of regression coefficients:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  stats::glm(formula = unem ~ gdp + capmob + trade, family = gaussian(&quot;identity&quot;),</span>
<span class="c">##     data = .)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)          gdp       capmob        trade</span>
<span class="c">##     6.18129     -0.32360      1.42194      0.01985</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 349 Total (i.e. Null);  346 Residual</span>
<span class="c">## Null Deviance:       3665</span>
<span class="c">## Residual Deviance: 2610  AIC: 1706</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set explanatory variables to their default (mean/mode) values, with
high (80th percentile) and low (20th percentile) values for trade:</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> trade <span class="o">=</span> quantile<span class="p">(</span>macro<span class="o">$</span>trade<span class="p">,</span> <span class="m">0.8</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> trade <span class="o">=</span> quantile<span class="p">(</span>macro<span class="o">$</span>trade<span class="p">,</span> <span class="m">0.2</span><span class="p">))</span>
</pre></div>
</div>
<p>Generate first differences for the effect of high versus low trade on GDP:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd      50%     2.5%   97.5%</span>
<span class="c">## [1,] 5.432055 0.1905829 5.431937 5.063223 5.81511</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%       2.5%    97.5%</span>
<span class="c">## [1,] 5.436278 2.772421 5.395798 0.07670153 10.99464</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd      50%    2.5%    97.5%</span>
<span class="c">## [1,] 4.603524 0.1883068 4.610879 4.23593 4.957381</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%      2.5%    97.5%</span>
<span class="c">## [1,] 4.589726 2.877556 4.539204 -1.117499 10.72641</span>
<span class="c">## fd</span>
<span class="c">##            mean        sd        50%      2.5%      97.5%</span>
<span class="c">## [1,] -0.8285311 0.2382877 -0.8352265 -1.278203 -0.3666807</span>
</pre></div>
</div>
<p>A visual summary of quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-normal" src="_images/Zelig-normal-1.png" />
<p class="caption">Zelig-normal</p>
</div>
</div>
</div>
<div class="section" id="id36">
<h3>Model<a class="headerlink" href="#id36" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> be the continuous dependent variable for observation
<img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is described by a univariate normal model
with a vector of means <img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> and scalar variance
<img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/>:</p>
<div class="math">
<p><img src="_images/math/92cca0a95579bbd947f5e2566bdf78a2027f3f9b.png" alt="Y_i \; \sim \; \textrm{Normal}(\mu_i, \sigma^2)."/></p>
</div></li>
<li><p class="first">The <em>systematic component</em> is</p>
<div class="math">
<p><img src="_images/math/e2582c2521dff40fd1d1a9e277d397dccf4919cb.png" alt="\mu_i \;= \; x_i \beta,"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id37">
<h3>Quantities of Interest<a class="headerlink" href="#id37" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected value (qi$ev) is the mean of simulations from the the
stochastic component,</p>
<div class="math">
<p><img src="_images/math/4c7424be232769f740891d7a55f7aef33f47572f.png" alt="E(Y) = \mu_i = x_i \beta,"/></p>
</div><p>given a draw of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its posterior.</p>
</li>
<li><p class="first">The predicted value (qi$pr) is drawn from the distribution defined by
the set of parameters <img class="math" src="_images/math/26ce59f07ba5807f8bb61216cbf3188c6cb18c80.png" alt="(\mu_i, \sigma)"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) is:</p>
<div class="math">
<p><img src="_images/math/5eb5fe9153bef31d3b9734bb97e106810e590906.png" alt="\textrm{FD}\; = \;E(Y \mid x_1) -  E(Y \mid x)"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id38">
<h3>Output Values<a class="headerlink" href="#id38" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">normal,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id39">
<h3>See also<a class="headerlink" href="#id39" title="Permalink to this headline">¶</a></h3>
<p>The normal model is part of the stats package by . Advanced users may
wish to refer to <tt class="docutils literal"><span class="pre">help(glm)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-poisson">
<span id="zpoisson"></span><h2>zelig-poisson<a class="headerlink" href="#zelig-poisson" title="Permalink to this headline">¶</a></h2>
<p>Poisson Regression for Event Count Dependent Variables</p>
<p>Use the Poisson regression model if the observations of your dependent
variable represents the number of independent events that occur during a
fixed period of time (see the negative binomial model, , for
over-dispersed event counts.) For a Bayesian implementation of this
model, see .</p>
<div class="section" id="id40">
<h3>Syntax<a class="headerlink" href="#id40" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zpoisson<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id41">
<h3>Example<a class="headerlink" href="#id41" title="Permalink to this headline">¶</a></h3>
<p>Load sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>sanction<span class="p">)</span>
</pre></div>
</div>
<p>Estimate Poisson model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>num <span class="o">~</span> target <span class="o">+</span> coop<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson&quot;</span><span class="p">,</span> data <span class="o">=</span> sanction<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2007.</span>
<span class="c">##   poisson: Poisson Regression for Event Count Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  stats::glm(formula = num ~ target + coop, family = poisson(&quot;log&quot;),</span>
<span class="c">##     data = .)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)       target         coop</span>
<span class="c">##    -0.96772     -0.02102      1.21082</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 77 Total (i.e. Null);  75 Residual</span>
<span class="c">## Null Deviance:       1584</span>
<span class="c">## Residual Deviance: 720.8         AIC: 944.3</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set values for the explanatory variables to their default mean values:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulate fitted values:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##         mean        sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 3.24873 0.2314021 3.234492 2.832226 3.753931</span>
<span class="c">## pv</span>
<span class="c">##       mean       sd 50% 2.5% 97.5%</span>
<span class="c">## [1,] 3.281 1.838319   3    0     7</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-poisson" src="_images/Zelig-poisson-1.png" />
<p class="caption">Zelig-poisson</p>
</div>
</div>
<div class="section" id="id42">
<h3>Model<a class="headerlink" href="#id42" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> be the number of independent events that occur during a
fixed time period. This variable can take any non-negative integer.</p>
<ul>
<li><p class="first">The Poisson distribution has <em>stochastic component</em></p>
<div class="math">
<p><img src="_images/math/5540ca84e8f5169523335a479866020c89dcb8e1.png" alt="Y_i \; \sim \; \textrm{Poisson}(\lambda_i),"/></p>
</div><p>where <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/> is the mean and variance parameter.</p>
</li>
<li><p class="first">The <em>systematic component</em> is</p>
<div class="math">
<p><img src="_images/math/2340a2662ddcb518d4617bd57234f976b60fc74d.png" alt="\lambda_i \; = \; \exp(x_i \beta),"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of explanatory variables, and
<img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id43">
<h3>Quantities of Interest<a class="headerlink" href="#id43" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected value (qi$ev) is the mean of simulations from the
stochastic component,</p>
<div class="math">
<p><img src="_images/math/df22f703cd62d9dfefbd0f0a3b9972af7ca4e507.png" alt="E(Y) = \lambda_i =  \exp(x_i
  \beta),"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution.</p>
</li>
<li><p class="first">The predicted value (qi$pr) is a random draw from the poisson
distribution defined by mean <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/>.</p>
</li>
<li><p class="first">The first difference in the expected values (qi$fd) is given by:</p>
<div class="math">
<p><img src="_images/math/82a49121147fab5307d7df11a6b69165b3f6794c.png" alt="\textrm{FD} \; = \; E(Y | x_1) - E(Y \mid x)"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id44">
<h3>Output Values<a class="headerlink" href="#id44" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">poisson,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id45">
<h3>See also<a class="headerlink" href="#id45" title="Permalink to this headline">¶</a></h3>
<p>The poisson model is part of the stats package by . Advanced users may
wish to refer to <tt class="docutils literal"><span class="pre">help(glm)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-probit">
<span id="zprobit"></span><h2>zelig-probit<a class="headerlink" href="#zelig-probit" title="Permalink to this headline">¶</a></h2>
<p>Probit Regression for Dichotomous Dependent Variables</p>
<p>Use probit regression to model binary dependent variables specified as a
function of a set of explanatory variables.</p>
<div class="section" id="id46">
<h3>Syntax<a class="headerlink" href="#id46" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zprobit<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">,</span> x1 <span class="o">=</span> <span class="kc">NULL</span><span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id47">
<h3>Example<a class="headerlink" href="#id47" title="Permalink to this headline">¶</a></h3>
<p>Attach the sample turnout dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>turnout<span class="p">)</span>
</pre></div>
</div>
<p>Estimate parameter values for the probit regression:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> race <span class="o">+</span> educate<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit&quot;</span><span class="p">,</span> data <span class="o">=</span> turnout<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2007.</span>
<span class="c">##   probit: Probit Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  stats::glm(formula = vote ~ race + educate, family = binomial(&quot;probit&quot;),</span>
<span class="c">##     data = .)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)    racewhite      educate</span>
<span class="c">##    -0.72595      0.29908      0.09712</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 1999 Total (i.e. Null);  1997 Residual</span>
<span class="c">## Null Deviance:       2267</span>
<span class="c">## Residual Deviance: 2136  AIC: 2142</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set values for the explanatory variables to their default values.</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest from the posterior distribution.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-probit" src="_images/Zelig-probit-1.png" />
<p class="caption">Zelig-probit</p>
</div>
</div>
<div class="section" id="id48">
<h3>Model<a class="headerlink" href="#id48" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> be the observed binary dependent variable for
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> which takes the value of either 0 or 1.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/e6ab2177225a79f988ab416b0353405999e15789.png" alt="Y_i \; \sim \; \textrm{Bernoulli}(\pi_i),"/></p>
</div><p>where <img class="math" src="_images/math/204ad227887156eca0ea3882997e3c5d0703174b.png" alt="\pi_i=\Pr(Y_i=1)"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is</p>
<div class="math">
<p><img src="_images/math/a8bd29a343890834661f63d7510931e96651bc8d.png" alt="\pi_i \; = \; \Phi (x_i \beta)"/></p>
</div><p>where <img class="math" src="_images/math/e4f08b55c6f99b472e0a295b96ff6487d545dfd0.png" alt="\Phi(\mu)"/> is the cumulative distribution function of
the Normal distribution with mean 0 and unit variance.</p>
</li>
</ul>
</div>
<div class="section" id="id49">
<h3>Quantities of Interest<a class="headerlink" href="#id49" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected value (qi$ev) is a simulation of predicted probability
of success</p>
<div class="math">
<p><img src="_images/math/28b543f2a46c190d6f10c93b07f717ac6d15a5b4.png" alt="E(Y) = \pi_i = \Phi(x_i
  \beta),"/></p>
</div><p>given a draw of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution.</p>
</li>
<li><p class="first">The predicted value (qi$pr) is a draw from a Bernoulli distribution
with mean <img class="math" src="_images/math/916754f7e7de4247fa27c52f610948b4055311fe.png" alt="\pi_i"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) in expected values is defined as</p>
<div class="math">
<p><img src="_images/math/8f669920018c72d7f23a4ba843925381c4703f17.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1) - \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">The risk ratio (qi$rr) is defined as</p>
<div class="math">
<p><img src="_images/math/6a3b874e942e65340c5fd207b718ea58c330392c.png" alt="\textrm{RR} = \Pr(Y = 1 \mid x_1) / \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id50">
<h3>Output Values<a class="headerlink" href="#id50" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">probit,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="id51">
<h3>See also<a class="headerlink" href="#id51" title="Permalink to this headline">¶</a></h3>
<p>The probit model is part of the stats package by . Advanced users may
wish to refer to <tt class="docutils literal"><span class="pre">help(glm)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-relogit">
<span id="zrelogit"></span><h2>zelig-relogit<a class="headerlink" href="#zelig-relogit" title="Permalink to this headline">¶</a></h2>
<p>Rare Events Logistic Regression for Dichotomous Dependent Variables</p>
<p>The relogit procedure estimates the same model as standard logistic
regression (appropriate when you have a dichotomous dependent variable
and a set of explanatory variables; see ), but the estimates are
corrected for the bias that occurs when the sample is small or the
observed events are rare (i.e., if the dependent variable has many more
1s than 0s or the reverse). The relogit procedure also optionally uses
prior correction for case-control sampling designs.</p>
<div class="section" id="id52">
<h3>Syntax<a class="headerlink" href="#id52" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zrelogit<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> tau <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span>
                       case.control <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;prior&quot;</span><span class="p">,</span> <span class="s">&quot;weighting&quot;</span><span class="p">),</span>
                       bias.correct <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> robust <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
                       data <span class="o">=</span> mydata<span class="p">,</span> <span class="kc">...</span><span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;relogit&quot;</span><span class="p">,</span> tau <span class="o">=</span> <span class="kc">NULL</span><span class="p">,</span>
                       case.control <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;prior&quot;</span><span class="p">,</span> <span class="s">&quot;weighting&quot;</span><span class="p">),</span>
                       bias.correct <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> robust <span class="o">=</span> <span class="kc">FALSE</span><span class="p">,</span>
                       data <span class="o">=</span> mydata<span class="p">,</span> <span class="kc">...</span><span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="arguments">
<h3>Arguments<a class="headerlink" href="#arguments" title="Permalink to this headline">¶</a></h3>
<p>The relogit procedure supports four optional arguments in addition to
the standard arguments for zelig(). You may additionally use:</p>
<ul class="simple">
<li>tau: a vector containing either one or two values for <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/>,
the true population fraction of ones. Use, for example, tau = c(0.05,
0.1) to specify that the lower bound on tau is 0.05 and the upper
bound is 0.1. If left unspecified, only finite-sample bias correction
is performed, not case-control correction.</li>
<li>case.control: if tau is specified, choose a method to correct for
case-control sampling design: “prior” (default) or “weighting”.</li>
<li>bias.correct: a logical value of TRUE (default) or FALSE indicating
whether the intercept should be corrected for finite sample (rare
events) bias.</li>
</ul>
<p>Note that if tau = NULL, bias.correct = FALSE, the
relogit procedure performs a standard logistic regression without any
correction.</p>
</div>
<div class="section" id="example-1-one-tau-with-prior-correction-and-bias-correction">
<h3>Example 1: One Tau with Prior Correction and Bias Correction<a class="headerlink" href="#example-1-one-tau-with-prior-correction-and-bias-correction" title="Permalink to this headline">¶</a></h3>
<p>Due to memory and space considerations, the data used here are a sample
drawn from the full data set used in King and Zeng, 2001, The proportion
of militarized interstate conflicts to the absence of disputes is
<img class="math" src="_images/math/0476daae6e85154c40577131b46dfc345e403b50.png" alt="\tau = 1,042 / 303,772
\approx 0.00343"/>. To estimate the model,</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>mid<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>z.out1 <span class="o">&lt;-</span> zelig<span class="p">(</span>conflict <span class="o">~</span> major <span class="o">+</span> contig <span class="o">+</span> power <span class="o">+</span> maxdem <span class="o">+</span> mindem <span class="o">+</span> years<span class="p">,</span> data <span class="o">=</span> mid<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;relogit&quot;</span><span class="p">,</span> tau <span class="o">=</span> <span class="m">1042</span><span class="o">/</span><span class="m">303772</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, and Olivia Lau. 2014.</span>
<span class="c">##   relogit: Rare Events Logistic Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Summarize the model output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:  relogit(formula = cbind(conflict, 1 - conflict) ~ major + contig +</span>
<span class="c">##     power + maxdem + mindem + years, data = ., tau = 0.00343020423212146,</span>
<span class="c">##     bias.correct = TRUE, case.control = &quot;prior&quot;)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)        major       contig        power       maxdem</span>
<span class="c">##    -7.50836      2.43196      4.10797      1.05358      0.04804</span>
<span class="c">##      mindem        years</span>
<span class="c">##    -0.06413     -0.06293</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 3125 Total (i.e. Null);  3119 Residual</span>
<span class="c">## Null Deviance:       3979</span>
<span class="c">## Residual Deviance: 1869  AIC: 1883</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the explanatory variables to their means:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out1 <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.out1<span class="p">)</span>
summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##             mean           sd        50%        2.5%       97.5%</span>
<span class="c">## [1,] 0.002396371 0.0001496194 0.00238697 0.002133949 0.002720633</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.997 0.003</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-relogit" src="_images/Zelig-relogit-1.png" />
<p class="caption">Zelig-relogit</p>
</div>
</div>
<div class="section" id="example-2-one-tau-with-weighting-robust-standard-errors-and-bias-correction">
<h3>Example 2: One Tau with Weighting, Robust Standard Errors, and Bias Correction<a class="headerlink" href="#example-2-one-tau-with-weighting-robust-standard-errors-and-bias-correction" title="Permalink to this headline">¶</a></h3>
<p>Suppose that we wish to perform case control correction using weighting
(rather than the default prior correction). To estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2 <span class="o">&lt;-</span> zelig<span class="p">(</span>conflict <span class="o">~</span> major <span class="o">+</span> contig <span class="o">+</span> power <span class="o">+</span> maxdem <span class="o">+</span> mindem <span class="o">+</span> years<span class="p">,</span> data <span class="o">=</span> mid<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;relogit&quot;</span><span class="p">,</span> tau <span class="o">=</span> <span class="m">1042</span><span class="o">/</span><span class="m">303772</span><span class="p">,</span> case.control <span class="o">=</span> <span class="s">&quot;weighting&quot;</span><span class="p">,</span> robust <span class="o">=</span> <span class="kc">TRUE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Error in glm.control(robust = TRUE): unused argument (robust = TRUE)</span>
</pre></div>
</div>
<p>Summarize the model output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## geepack::geeglm(formula = vote ~ race + educate, family = binomial(&quot;probit&quot;),</span>
<span class="c">##     data = ., id = c(1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L, 1L,</span>
<span class="c">##     2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 2L, 3L, 3L, 3L, 3L, 3L,</span>
<span class="c">##     3L, 3L, 3L, 3L, 3L, 4L, 4L, 4L, 4L, 4L, 4L, 4L, 4L, 4L, 4L,</span>
<span class="c">##     5L, 5L, 5L, 5L, 5L, 5L, 5L, 5L, 5L, 5L, 6L, 6L, 6L, 6L, 6L,</span>
<span class="c">##     6L, 6L, 6L, 6L, 6L, 7L, 7L, 7L, 7L, 7L, 7L, 7L, 7L, 7L, 7L,</span>
<span class="c">##     8L, 8L, 8L, 8L, 8L, 8L, 8L, 8L, 8L, 8L, 9L, 9L, 9L, 9L, 9L,</span>
<span class="c">##     9L, 9L, 9L, 9L, 9L, 10L, 10L, 10L, 10L, 10L, 10L, 10L, 10L,</span>
<span class="c">##     10L, 10L, 11L, 11L, 11L, 11L, 11L, 11L, 11L, 11L, 11L, 11L,</span>
<span class="c">##     12L, 12L, 12L, 12L, 12L, 12L, 12L, 12L, 12L, 12L, 13L, 13L,</span>
<span class="c">##     13L, 13L, 13L, 13L, 13L, 13L, 13L, 13L, 14L, 14L, 14L, 14L,</span>
<span class="c">##     14L, 14L, 14L, 14L, 14L, 14L, 15L, 15L, 15L, 15L, 15L, 15L,</span>
<span class="c">##     15L, 15L, 15L, 15L, 16L, 16L, 16L, 16L, 16L, 16L, 16L, 16L,</span>
<span class="c">##     16L, 16L, 17L, 17L, 17L, 17L, 17L, 17L, 17L, 17L, 17L, 17L,</span>
<span class="c">##     18L, 18L, 18L, 18L, 18L, 18L, 18L, 18L, 18L, 18L, 19L, 19L,</span>
<span class="c">##     19L, 19L, 19L, 19L, 19L, 19L, 19L, 19L, 20L, 20L, 20L, 20L,</span>
<span class="c">##     20L, 20L, 20L, 20L, 20L, 20L, 21L, 21L, 21L, 21L, 21L, 21L,</span>
<span class="c">##     21L, 21L, 21L, 21L, 22L, 22L, 22L, 22L, 22L, 22L, 22L, 22L,</span>
<span class="c">##     22L, 22L, 23L, 23L, 23L, 23L, 23L, 23L, 23L, 23L, 23L, 23L,</span>
<span class="c">##     24L, 24L, 24L, 24L, 24L, 24L, 24L, 24L, 24L, 24L, 25L, 25L,</span>
<span class="c">##     25L, 25L, 25L, 25L, 25L, 25L, 25L, 25L, 26L, 26L, 26L, 26L,</span>
<span class="c">##     26L, 26L, 26L, 26L, 26L, 26L, 27L, 27L, 27L, 27L, 27L, 27L,</span>
<span class="c">##     27L, 27L, 27L, 27L, 28L, 28L, 28L, 28L, 28L, 28L, 28L, 28L,</span>
<span class="c">##     28L, 28L, 29L, 29L, 29L, 29L, 29L, 29L, 29L, 29L, 29L, 29L,</span>
<span class="c">##     30L, 30L, 30L, 30L, 30L, 30L, 30L, 30L, 30L, 30L, 31L, 31L,</span>
<span class="c">##     31L, 31L, 31L, 31L, 31L, 31L, 31L, 31L, 32L, 32L, 32L, 32L,</span>
<span class="c">##     32L, 32L, 32L, 32L, 32L, 32L, 33L, 33L, 33L, 33L, 33L, 33L,</span>
<span class="c">##     33L, 33L, 33L, 33L, 34L, 34L, 34L, 34L, 34L, 34L, 34L, 34L,</span>
<span class="c">##     34L, 34L, 35L, 35L, 35L, 35L, 35L, 35L, 35L, 35L, 35L, 35L,</span>
<span class="c">##     36L, 36L, 36L, 36L, 36L, 36L, 36L, 36L, 36L, 36L, 37L, 37L,</span>
<span class="c">##     37L, 37L, 37L, 37L, 37L, 37L, 37L, 37L, 38L, 38L, 38L, 38L,</span>
<span class="c">##     38L, 38L, 38L, 38L, 38L, 38L, 39L, 39L, 39L, 39L, 39L, 39L,</span>
<span class="c">##     39L, 39L, 39L, 39L, 40L, 40L, 40L, 40L, 40L, 40L, 40L, 40L,</span>
<span class="c">##     40L, 40L, 41L, 41L, 41L, 41L, 41L, 41L, 41L, 41L, 41L, 41L,</span>
<span class="c">##     42L, 42L, 42L, 42L, 42L, 42L, 42L, 42L, 42L, 42L, 43L, 43L,</span>
<span class="c">##     43L, 43L, 43L, 43L, 43L, 43L, 43L, 43L, 44L, 44L, 44L, 44L,</span>
<span class="c">##     44L, 44L, 44L, 44L, 44L, 44L, 45L, 45L, 45L, 45L, 45L, 45L,</span>
<span class="c">##     45L, 45L, 45L, 45L, 46L, 46L, 46L, 46L, 46L, 46L, 46L, 46L,</span>
<span class="c">##     46L, 46L, 47L, 47L, 47L, 47L, 47L, 47L, 47L, 47L, 47L, 47L,</span>
<span class="c">##     48L, 48L, 48L, 48L, 48L, 48L, 48L, 48L, 48L, 48L, 49L, 49L,</span>
<span class="c">##     49L, 49L, 49L, 49L, 49L, 49L, 49L, 49L, 50L, 50L, 50L, 50L,</span>
<span class="c">##     50L, 50L, 50L, 50L, 50L, 50L, 51L, 51L, 51L, 51L, 51L, 51L,</span>
<span class="c">##     51L, 51L, 51L, 51L, 52L, 52L, 52L, 52L, 52L, 52L, 52L, 52L,</span>
<span class="c">##     52L, 52L, 53L, 53L, 53L, 53L, 53L, 53L, 53L, 53L, 53L, 53L,</span>
<span class="c">##     54L, 54L, 54L, 54L, 54L, 54L, 54L, 54L, 54L, 54L, 55L, 55L,</span>
<span class="c">##     55L, 55L, 55L, 55L, 55L, 55L, 55L, 55L, 56L, 56L, 56L, 56L,</span>
<span class="c">##     56L, 56L, 56L, 56L, 56L, 56L, 57L, 57L, 57L, 57L, 57L, 57L,</span>
<span class="c">##     57L, 57L, 57L, 57L, 58L, 58L, 58L, 58L, 58L, 58L, 58L, 58L,</span>
<span class="c">##     58L, 58L, 59L, 59L, 59L, 59L, 59L, 59L, 59L, 59L, 59L, 59L,</span>
<span class="c">##     60L, 60L, 60L, 60L, 60L, 60L, 60L, 60L, 60L, 60L, 61L, 61L,</span>
<span class="c">##     61L, 61L, 61L, 61L, 61L, 61L, 61L, 61L, 62L, 62L, 62L, 62L,</span>
<span class="c">##     62L, 62L, 62L, 62L, 62L, 62L, 63L, 63L, 63L, 63L, 63L, 63L,</span>
<span class="c">##     63L, 63L, 63L, 63L, 64L, 64L, 64L, 64L, 64L, 64L, 64L, 64L,</span>
<span class="c">##     64L, 64L, 65L, 65L, 65L, 65L, 65L, 65L, 65L, 65L, 65L, 65L,</span>
<span class="c">##     66L, 66L, 66L, 66L, 66L, 66L, 66L, 66L, 66L, 66L, 67L, 67L,</span>
<span class="c">##     67L, 67L, 67L, 67L, 67L, 67L, 67L, 67L, 68L, 68L, 68L, 68L,</span>
<span class="c">##     68L, 68L, 68L, 68L, 68L, 68L, 69L, 69L, 69L, 69L, 69L, 69L,</span>
<span class="c">##     69L, 69L, 69L, 69L, 70L, 70L, 70L, 70L, 70L, 70L, 70L, 70L,</span>
<span class="c">##     70L, 70L, 71L, 71L, 71L, 71L, 71L, 71L, 71L, 71L, 71L, 71L,</span>
<span class="c">##     72L, 72L, 72L, 72L, 72L, 72L, 72L, 72L, 72L, 72L, 73L, 73L,</span>
<span class="c">##     73L, 73L, 73L, 73L, 73L, 73L, 73L, 73L, 74L, 74L, 74L, 74L,</span>
<span class="c">##     74L, 74L, 74L, 74L, 74L, 74L, 75L, 75L, 75L, 75L, 75L, 75L,</span>
<span class="c">##     75L, 75L, 75L, 75L, 76L, 76L, 76L, 76L, 76L, 76L, 76L, 76L,</span>
<span class="c">##     76L, 76L, 77L, 77L, 77L, 77L, 77L, 77L, 77L, 77L, 77L, 77L,</span>
<span class="c">##     78L, 78L, 78L, 78L, 78L, 78L, 78L, 78L, 78L, 78L, 79L, 79L,</span>
<span class="c">##     79L, 79L, 79L, 79L, 79L, 79L, 79L, 79L, 80L, 80L, 80L, 80L,</span>
<span class="c">##     80L, 80L, 80L, 80L, 80L, 80L, 81L, 81L, 81L, 81L, 81L, 81L,</span>
<span class="c">##     81L, 81L, 81L, 81L, 82L, 82L, 82L, 82L, 82L, 82L, 82L, 82L,</span>
<span class="c">##     82L, 82L, 83L, 83L, 83L, 83L, 83L, 83L, 83L, 83L, 83L, 83L,</span>
<span class="c">##     84L, 84L, 84L, 84L, 84L, 84L, 84L, 84L, 84L, 84L, 85L, 85L,</span>
<span class="c">##     85L, 85L, 85L, 85L, 85L, 85L, 85L, 85L, 86L, 86L, 86L, 86L,</span>
<span class="c">##     86L, 86L, 86L, 86L, 86L, 86L, 87L, 87L, 87L, 87L, 87L, 87L,</span>
<span class="c">##     87L, 87L, 87L, 87L, 88L, 88L, 88L, 88L, 88L, 88L, 88L, 88L,</span>
<span class="c">##     88L, 88L, 89L, 89L, 89L, 89L, 89L, 89L, 89L, 89L, 89L, 89L,</span>
<span class="c">##     90L, 90L, 90L, 90L, 90L, 90L, 90L, 90L, 90L, 90L, 91L, 91L,</span>
<span class="c">##     91L, 91L, 91L, 91L, 91L, 91L, 91L, 91L, 92L, 92L, 92L, 92L,</span>
<span class="c">##     92L, 92L, 92L, 92L, 92L, 92L, 93L, 93L, 93L, 93L, 93L, 93L,</span>
<span class="c">##     93L, 93L, 93L, 93L, 94L, 94L, 94L, 94L, 94L, 94L, 94L, 94L,</span>
<span class="c">##     94L, 94L, 95L, 95L, 95L, 95L, 95L, 95L, 95L, 95L, 95L, 95L,</span>
<span class="c">##     96L, 96L, 96L, 96L, 96L, 96L, 96L, 96L, 96L, 96L, 97L, 97L,</span>
<span class="c">##     97L, 97L, 97L, 97L, 97L, 97L, 97L, 97L, 98L, 98L, 98L, 98L,</span>
<span class="c">##     98L, 98L, 98L, 98L, 98L, 98L, 99L, 99L, 99L, 99L, 99L, 99L,</span>
<span class="c">##     99L, 99L, 99L, 99L, 100L, 100L, 100L, 100L, 100L, 100L, 100L,</span>
<span class="c">##     100L, 100L, 100L, 101L, 101L, 101L, 101L, 101L, 101L, 101L,</span>
<span class="c">##     101L, 101L, 101L, 102L, 102L, 102L, 102L, 102L, 102L, 102L,</span>
<span class="c">##     102L, 102L, 102L, 103L, 103L, 103L, 103L, 103L, 103L, 103L,</span>
<span class="c">##     103L, 103L, 103L, 104L, 104L, 104L, 104L, 104L, 104L, 104L,</span>
<span class="c">##     104L, 104L, 104L, 105L, 105L, 105L, 105L, 105L, 105L, 105L,</span>
<span class="c">##     105L, 105L, 105L, 106L, 106L, 106L, 106L, 106L, 106L, 106L,</span>
<span class="c">##     106L, 106L, 106L, 107L, 107L, 107L, 107L, 107L, 107L, 107L,</span>
<span class="c">##     107L, 107L, 107L, 108L, 108L, 108L, 108L, 108L, 108L, 108L,</span>
<span class="c">##     108L, 108L, 108L, 109L, 109L, 109L, 109L, 109L, 109L, 109L,</span>
<span class="c">##     109L, 109L, 109L, 110L, 110L, 110L, 110L, 110L, 110L, 110L,</span>
<span class="c">##     110L, 110L, 110L, 111L, 111L, 111L, 111L, 111L, 111L, 111L,</span>
<span class="c">##     111L, 111L, 111L, 112L, 112L, 112L, 112L, 112L, 112L, 112L,</span>
<span class="c">##     112L, 112L, 112L, 113L, 113L, 113L, 113L, 113L, 113L, 113L,</span>
<span class="c">##     113L, 113L, 113L, 114L, 114L, 114L, 114L, 114L, 114L, 114L,</span>
<span class="c">##     114L, 114L, 114L, 115L, 115L, 115L, 115L, 115L, 115L, 115L,</span>
<span class="c">##     115L, 115L, 115L, 116L, 116L, 116L, 116L, 116L, 116L, 116L,</span>
<span class="c">##     116L, 116L, 116L, 117L, 117L, 117L, 117L, 117L, 117L, 117L,</span>
<span class="c">##     117L, 117L, 117L, 118L, 118L, 118L, 118L, 118L, 118L, 118L,</span>
<span class="c">##     118L, 118L, 118L, 119L, 119L, 119L, 119L, 119L, 119L, 119L,</span>
<span class="c">##     119L, 119L, 119L, 120L, 120L, 120L, 120L, 120L, 120L, 120L,</span>
<span class="c">##     120L, 120L, 120L, 121L, 121L, 121L, 121L, 121L, 121L, 121L,</span>
<span class="c">##     121L, 121L, 121L, 122L, 122L, 122L, 122L, 122L, 122L, 122L,</span>
<span class="c">##     122L, 122L, 122L, 123L, 123L, 123L, 123L, 123L, 123L, 123L,</span>
<span class="c">##     123L, 123L, 123L, 124L, 124L, 124L, 124L, 124L, 124L, 124L,</span>
<span class="c">##     124L, 124L, 124L, 125L, 125L, 125L, 125L, 125L, 125L, 125L,</span>
<span class="c">##     125L, 125L, 125L, 126L, 126L, 126L, 126L, 126L, 126L, 126L,</span>
<span class="c">##     126L, 126L, 126L, 127L, 127L, 127L, 127L, 127L, 127L, 127L,</span>
<span class="c">##     127L, 127L, 127L, 128L, 128L, 128L, 128L, 128L, 128L, 128L,</span>
<span class="c">##     128L, 128L, 128L, 129L, 129L, 129L, 129L, 129L, 129L, 129L,</span>
<span class="c">##     129L, 129L, 129L, 130L, 130L, 130L, 130L, 130L, 130L, 130L,</span>
<span class="c">##     130L, 130L, 130L, 131L, 131L, 131L, 131L, 131L, 131L, 131L,</span>
<span class="c">##     131L, 131L, 131L, 132L, 132L, 132L, 132L, 132L, 132L, 132L,</span>
<span class="c">##     132L, 132L, 132L, 133L, 133L, 133L, 133L, 133L, 133L, 133L,</span>
<span class="c">##     133L, 133L, 133L, 134L, 134L, 134L, 134L, 134L, 134L, 134L,</span>
<span class="c">##     134L, 134L, 134L, 135L, 135L, 135L, 135L, 135L, 135L, 135L,</span>
<span class="c">##     135L, 135L, 135L, 136L, 136L, 136L, 136L, 136L, 136L, 136L,</span>
<span class="c">##     136L, 136L, 136L, 137L, 137L, 137L, 137L, 137L, 137L, 137L,</span>
<span class="c">##     137L, 137L, 137L, 138L, 138L, 138L, 138L, 138L, 138L, 138L,</span>
<span class="c">##     138L, 138L, 138L, 139L, 139L, 139L, 139L, 139L, 139L, 139L,</span>
<span class="c">##     139L, 139L, 139L, 140L, 140L, 140L, 140L, 140L, 140L, 140L,</span>
<span class="c">##     140L, 140L, 140L, 141L, 141L, 141L, 141L, 141L, 141L, 141L,</span>
<span class="c">##     141L, 141L, 141L, 142L, 142L, 142L, 142L, 142L, 142L, 142L,</span>
<span class="c">##     142L, 142L, 142L, 143L, 143L, 143L, 143L, 143L, 143L, 143L,</span>
<span class="c">##     143L, 143L, 143L, 144L, 144L, 144L, 144L, 144L, 144L, 144L,</span>
<span class="c">##     144L, 144L, 144L, 145L, 145L, 145L, 145L, 145L, 145L, 145L,</span>
<span class="c">##     145L, 145L, 145L, 146L, 146L, 146L, 146L, 146L, 146L, 146L,</span>
<span class="c">##     146L, 146L, 146L, 147L, 147L, 147L, 147L, 147L, 147L, 147L,</span>
<span class="c">##     147L, 147L, 147L, 148L, 148L, 148L, 148L, 148L, 148L, 148L,</span>
<span class="c">##     148L, 148L, 148L, 149L, 149L, 149L, 149L, 149L, 149L, 149L,</span>
<span class="c">##     149L, 149L, 149L, 150L, 150L, 150L, 150L, 150L, 150L, 150L,</span>
<span class="c">##     150L, 150L, 150L, 151L, 151L, 151L, 151L, 151L, 151L, 151L,</span>
<span class="c">##     151L, 151L, 151L, 152L, 152L, 152L, 152L, 152L, 152L, 152L,</span>
<span class="c">##     152L, 152L, 152L, 153L, 153L, 153L, 153L, 153L, 153L, 153L,</span>
<span class="c">##     153L, 153L, 153L, 154L, 154L, 154L, 154L, 154L, 154L, 154L,</span>
<span class="c">##     154L, 154L, 154L, 155L, 155L, 155L, 155L, 155L, 155L, 155L,</span>
<span class="c">##     155L, 155L, 155L, 156L, 156L, 156L, 156L, 156L, 156L, 156L,</span>
<span class="c">##     156L, 156L, 156L, 157L, 157L, 157L, 157L, 157L, 157L, 157L,</span>
<span class="c">##     157L, 157L, 157L, 158L, 158L, 158L, 158L, 158L, 158L, 158L,</span>
<span class="c">##     158L, 158L, 158L, 159L, 159L, 159L, 159L, 159L, 159L, 159L,</span>
<span class="c">##     159L, 159L, 159L, 160L, 160L, 160L, 160L, 160L, 160L, 160L,</span>
<span class="c">##     160L, 160L, 160L, 161L, 161L, 161L, 161L, 161L, 161L, 161L,</span>
<span class="c">##     161L, 161L, 161L, 162L, 162L, 162L, 162L, 162L, 162L, 162L,</span>
<span class="c">##     162L, 162L, 162L, 163L, 163L, 163L, 163L, 163L, 163L, 163L,</span>
<span class="c">##     163L, 163L, 163L, 164L, 164L, 164L, 164L, 164L, 164L, 164L,</span>
<span class="c">##     164L, 164L, 164L, 165L, 165L, 165L, 165L, 165L, 165L, 165L,</span>
<span class="c">##     165L, 165L, 165L, 166L, 166L, 166L, 166L, 166L, 166L, 166L,</span>
<span class="c">##     166L, 166L, 166L, 167L, 167L, 167L, 167L, 167L, 167L, 167L,</span>
<span class="c">##     167L, 167L, 167L, 168L, 168L, 168L, 168L, 168L, 168L, 168L,</span>
<span class="c">##     168L, 168L, 168L, 169L, 169L, 169L, 169L, 169L, 169L, 169L,</span>
<span class="c">##     169L, 169L, 169L, 170L, 170L, 170L, 170L, 170L, 170L, 170L,</span>
<span class="c">##     170L, 170L, 170L, 171L, 171L, 171L, 171L, 171L, 171L, 171L,</span>
<span class="c">##     171L, 171L, 171L, 172L, 172L, 172L, 172L, 172L, 172L, 172L,</span>
<span class="c">##     172L, 172L, 172L, 173L, 173L, 173L, 173L, 173L, 173L, 173L,</span>
<span class="c">##     173L, 173L, 173L, 174L, 174L, 174L, 174L, 174L, 174L, 174L,</span>
<span class="c">##     174L, 174L, 174L, 175L, 175L, 175L, 175L, 175L, 175L, 175L,</span>
<span class="c">##     175L, 175L, 175L, 176L, 176L, 176L, 176L, 176L, 176L, 176L,</span>
<span class="c">##     176L, 176L, 176L, 177L, 177L, 177L, 177L, 177L, 177L, 177L,</span>
<span class="c">##     177L, 177L, 177L, 178L, 178L, 178L, 178L, 178L, 178L, 178L,</span>
<span class="c">##     178L, 178L, 178L, 179L, 179L, 179L, 179L, 179L, 179L, 179L,</span>
<span class="c">##     179L, 179L, 179L, 180L, 180L, 180L, 180L, 180L, 180L, 180L,</span>
<span class="c">##     180L, 180L, 180L, 181L, 181L, 181L, 181L, 181L, 181L, 181L,</span>
<span class="c">##     181L, 181L, 181L, 182L, 182L, 182L, 182L, 182L, 182L, 182L,</span>
<span class="c">##     182L, 182L, 182L, 183L, 183L, 183L, 183L, 183L, 183L, 183L,</span>
<span class="c">##     183L, 183L, 183L, 184L, 184L, 184L, 184L, 184L, 184L, 184L,</span>
<span class="c">##     184L, 184L, 184L, 185L, 185L, 185L, 185L, 185L, 185L, 185L,</span>
<span class="c">##     185L, 185L, 185L, 186L, 186L, 186L, 186L, 186L, 186L, 186L,</span>
<span class="c">##     186L, 186L, 186L, 187L, 187L, 187L, 187L, 187L, 187L, 187L,</span>
<span class="c">##     187L, 187L, 187L, 188L, 188L, 188L, 188L, 188L, 188L, 188L,</span>
<span class="c">##     188L, 188L, 188L, 189L, 189L, 189L, 189L, 189L, 189L, 189L,</span>
<span class="c">##     189L, 189L, 189L, 190L, 190L, 190L, 190L, 190L, 190L, 190L,</span>
<span class="c">##     190L, 190L, 190L, 191L, 191L, 191L, 191L, 191L, 191L, 191L,</span>
<span class="c">##     191L, 191L, 191L, 192L, 192L, 192L, 192L, 192L, 192L, 192L,</span>
<span class="c">##     192L, 192L, 192L, 193L, 193L, 193L, 193L, 193L, 193L, 193L,</span>
<span class="c">##     193L, 193L, 193L, 194L, 194L, 194L, 194L, 194L, 194L, 194L,</span>
<span class="c">##     194L, 194L, 194L, 195L, 195L, 195L, 195L, 195L, 195L, 195L,</span>
<span class="c">##     195L, 195L, 195L, 196L, 196L, 196L, 196L, 196L, 196L, 196L,</span>
<span class="c">##     196L, 196L, 196L, 197L, 197L, 197L, 197L, 197L, 197L, 197L,</span>
<span class="c">##     197L, 197L, 197L, 198L, 198L, 198L, 198L, 198L, 198L, 198L,</span>
<span class="c">##     198L, 198L, 198L, 199L, 199L, 199L, 199L, 199L, 199L, 199L,</span>
<span class="c">##     199L, 199L, 199L, 200L, 200L, 200L, 200L, 200L, 200L, 200L,</span>
<span class="c">##     200L, 200L, 200L), zcor = c(0.5, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 1, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 1, 0.5, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 0.5, 1, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 0.5,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 0.5, 1,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 0.5, 1, 1, 0.5, 0.5, 0.5, 1, 1, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 1, 0.5, 0.5, 0.5, 0.5, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5, 0.5, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 0.5, 0.5, 1, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 0.5, 1, 0.5, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     0.5, 1, 1, 0.5, 0.5, 1, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1,</span>
<span class="c">##     1, 0.5, 1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     1, 1, 0.5, 1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 0.5, 1, 1, 0.5, 1, 1, 1, 0.5, 1, 0.5, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 0.5, 0.5, 0.5, 1, 0.5, 1, 1, 0.5, 1, 1, 0.5, 1, 0.5, 1,</span>
<span class="c">##     0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 1, 0.5,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 0.5, 1, 1, 1, 1, 1, 1, 0.5, 0.5, 0.5, 0.5,</span>
<span class="c">##     0.5, 0.5, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1, 1,</span>
<span class="c">##     1, 1, 1, 1, 1, 1), corstr = &quot;fixed&quot;)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)   racewhite     educate</span>
<span class="c">##   -24.55278   -19.89639  -296.26380</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 2000 Total (i.e. Null);  1997 Residual</span>
<span class="c">##</span>
<span class="c">## Scale Link:                   identity</span>
<span class="c">## Estimated Scale Parameters:  [1] 3.359685e+15</span>
<span class="c">##</span>
<span class="c">## Correlation:  Structure = fixed    Link = identity</span>
<span class="c">## Estimated Correlation Parameters:</span>
<span class="c">## alpha:1</span>
<span class="c">##       1</span>
<span class="c">##</span>
<span class="c">## Number of clusters:   200   Maximum cluster size: 10</span>
<span class="c">##</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the explanatory variables to their means:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out2 <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out2<span class="p">,</span> x <span class="o">=</span> x.out2<span class="p">)</span>
summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##              mean sd          50%         2.5%        97.5%</span>
<span class="c">## [1,] 2.220446e-16  0 2.220446e-16 2.220446e-16 2.220446e-16</span>
<span class="c">## pv</span>
<span class="c">##      [,1]</span>
<span class="c">## [1,]    1</span>
</pre></div>
</div>
</div>
<div class="section" id="example-3-two-taus-with-bias-correction-and-prior-correction">
<h3>Example 3: Two Taus with Bias Correction and Prior Correction<a class="headerlink" href="#example-3-two-taus-with-bias-correction-and-prior-correction" title="Permalink to this headline">¶</a></h3>
<p>Suppose that we did not know that <img class="math" src="_images/math/3ea6d30492110055362e739a5b4776f93c6f1814.png" alt="\tau \approx 0.00343"/>, but only
that it was somewhere between <img class="math" src="_images/math/501e197d9ac2263e0c31dbecb1a332a39f7f0c84.png" alt="(0.002, 0.005)"/>. To estimate a
model with a range of feasible estimates for <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> (using the
default prior correction method for case control correction):</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2 <span class="o">&lt;-</span> zelig<span class="p">(</span>conflict <span class="o">~</span> major <span class="o">+</span> contig <span class="o">+</span> power <span class="o">+</span> maxdem <span class="o">+</span> mindem <span class="o">+</span> years<span class="p">,</span> data <span class="o">=</span> mid<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;relogit&quot;</span><span class="p">,</span> tau <span class="o">=</span> c<span class="p">(</span><span class="m">0.002</span><span class="p">,</span> <span class="m">0.005</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, and Olivia Lau. 2014.</span>
<span class="c">##   relogit: Rare Events Logistic Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Summarize the model output:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">## $lower.estimate</span>
<span class="c">##</span>
<span class="c">## Call:  (function (formula, data = sys.parent(), tau = NULL, bias.correct = TRUE,</span>
<span class="c">##     case.control = &quot;prior&quot;, ...)</span>
<span class="c">## {</span>
<span class="c">##     mf &lt;- match.call()</span>
<span class="c">##     mf$tau &lt;- mf$bias.correct &lt;- mf$case.control &lt;- NULL</span>
<span class="c">##     if (!is.null(tau)) {</span>
<span class="c">##         tau &lt;- unique(tau)</span>
<span class="c">##         if (length(case.control) &gt; 1)</span>
<span class="c">##             stop(&quot;You can only choose one option for case control correction.&quot;)</span>
<span class="c">##         ck1 &lt;- grep(&quot;p&quot;, case.control)</span>
<span class="c">##         ck2 &lt;- grep(&quot;w&quot;, case.control)</span>
<span class="c">##         if (length(ck1) == 0 &amp; length(ck2) == 0)</span>
<span class="c">##             stop(&quot;choose either case.control = \&quot;prior\&quot; &quot;, &quot;or case.control = \&quot;weighting\&quot;&quot;)</span>
<span class="c">##         if (length(ck2) == 0)</span>
<span class="c">##             weighting &lt;- FALSE</span>
<span class="c">##         else weighting &lt;- TRUE</span>
<span class="c">##     }</span>
<span class="c">##     else weighting &lt;- FALSE</span>
<span class="c">##     if (length(tau) &gt; 2)</span>
<span class="c">##         stop(&quot;tau must be a vector of length less than or equal to 2&quot;)</span>
<span class="c">##     else if (length(tau) == 2) {</span>
<span class="c">##         mf[[1]] &lt;- relogit</span>
<span class="c">##         res &lt;- list()</span>
<span class="c">##         mf$tau &lt;- min(tau)</span>
<span class="c">##         res$lower.estimate &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         mf$tau &lt;- max(tau)</span>
<span class="c">##         res$upper.estimate &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         res$formula &lt;- formula</span>
<span class="c">##         class(res) &lt;- c(&quot;Relogit2&quot;, &quot;Relogit&quot;)</span>
<span class="c">##         return(res)</span>
<span class="c">##     }</span>
<span class="c">##     else {</span>
<span class="c">##         mf[[1]] &lt;- glm</span>
<span class="c">##         mf$family &lt;- binomial(link = &quot;logit&quot;)</span>
<span class="c">##         y2 &lt;- model.response(model.frame(mf$formula, data))</span>
<span class="c">##         if (is.matrix(y2))</span>
<span class="c">##             y &lt;- y2[, 1]</span>
<span class="c">##         else y &lt;- y2</span>
<span class="c">##         ybar &lt;- mean(y)</span>
<span class="c">##         if (weighting) {</span>
<span class="c">##             w1 &lt;- tau/ybar</span>
<span class="c">##             w0 &lt;- (1 - tau)/(1 - ybar)</span>
<span class="c">##             wi &lt;- w1 * y + w0 * (1 - y)</span>
<span class="c">##             mf$weights &lt;- wi</span>
<span class="c">##         }</span>
<span class="c">##         res &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         res$call &lt;- match.call(expand.dots = TRUE)</span>
<span class="c">##         res$tau &lt;- tau</span>
<span class="c">##         X &lt;- model.matrix(res)</span>
<span class="c">##         if (bias.correct) {</span>
<span class="c">##             pihat &lt;- fitted(res)</span>
<span class="c">##             if (is.null(tau))</span>
<span class="c">##                 wi &lt;- rep(1, length(y))</span>
<span class="c">##             else if (weighting)</span>
<span class="c">##                 res$weighting &lt;- TRUE</span>
<span class="c">##             else {</span>
<span class="c">##                 w1 &lt;- tau/ybar</span>
<span class="c">##                 w0 &lt;- (1 - tau)/(1 - ybar)</span>
<span class="c">##                 wi &lt;- w1 * y + w0 * (1 - y)</span>
<span class="c">##                 res$weighting &lt;- FALSE</span>
<span class="c">##             }</span>
<span class="c">##             W &lt;- pihat * (1 - pihat) * wi</span>
<span class="c">##             Qdiag &lt;- lm.influence(lm(y ~ X - 1, weights = W))$hat/W</span>
<span class="c">##             if (is.null(tau))</span>
<span class="c">##                 xi &lt;- 0.5 * Qdiag * (2 * pihat - 1)</span>
<span class="c">##             else xi &lt;- 0.5 * Qdiag * ((1 + w0) * pihat - w0)</span>
<span class="c">##             res$coefficients &lt;- res$coefficients - lm(xi ~ X -</span>
<span class="c">##                 1, weights = W)$coefficients</span>
<span class="c">##             res$bias.correct &lt;- TRUE</span>
<span class="c">##         }</span>
<span class="c">##         else res$bias.correct &lt;- FALSE</span>
<span class="c">##         if (!is.null(tau) &amp; !weighting) {</span>
<span class="c">##             if (tau &lt;= 0 || tau &gt;= 1)</span>
<span class="c">##                 stop(&quot;\ntau needs to be between 0 and 1.\n&quot;)</span>
<span class="c">##             res$coefficients[&quot;(Intercept)&quot;] &lt;- res$coefficients[&quot;(Intercept)&quot;] -</span>
<span class="c">##                 log(((1 - tau)/tau) * (ybar/(1 - ybar)))</span>
<span class="c">##             res$prior.correct &lt;- TRUE</span>
<span class="c">##             res$weighting &lt;- FALSE</span>
<span class="c">##         }</span>
<span class="c">##         else res$prior.correct &lt;- FALSE</span>
<span class="c">##         if (is.null(res$weighting))</span>
<span class="c">##             res$weighting &lt;- FALSE</span>
<span class="c">##         res$linear.predictors &lt;- t(res$coefficients) %*% t(X)</span>
<span class="c">##         res$fitted.values &lt;- 1/(1 + exp(-res$linear.predictors))</span>
<span class="c">##         res$zelig &lt;- &quot;Relogit&quot;</span>
<span class="c">##         class(res) &lt;- c(&quot;Relogit&quot;, &quot;glm&quot;)</span>
<span class="c">##         return(res)</span>
<span class="c">##     }</span>
<span class="c">## })(formula = cbind(conflict, 1 - conflict) ~ major + contig +</span>
<span class="c">##     power + maxdem + mindem + years, data = ., tau = 0.002)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)        major       contig        power       maxdem</span>
<span class="c">##    -8.04923      2.43196      4.10791      1.05357      0.04804</span>
<span class="c">##      mindem        years</span>
<span class="c">##    -0.06412     -0.06293</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 3125 Total (i.e. Null);  3119 Residual</span>
<span class="c">## Null Deviance:       3979</span>
<span class="c">## Residual Deviance: 1869  AIC: 1883</span>
<span class="c">##</span>
<span class="c">## $upper.estimate</span>
<span class="c">##</span>
<span class="c">## Call:  (function (formula, data = sys.parent(), tau = NULL, bias.correct = TRUE,</span>
<span class="c">##     case.control = &quot;prior&quot;, ...)</span>
<span class="c">## {</span>
<span class="c">##     mf &lt;- match.call()</span>
<span class="c">##     mf$tau &lt;- mf$bias.correct &lt;- mf$case.control &lt;- NULL</span>
<span class="c">##     if (!is.null(tau)) {</span>
<span class="c">##         tau &lt;- unique(tau)</span>
<span class="c">##         if (length(case.control) &gt; 1)</span>
<span class="c">##             stop(&quot;You can only choose one option for case control correction.&quot;)</span>
<span class="c">##         ck1 &lt;- grep(&quot;p&quot;, case.control)</span>
<span class="c">##         ck2 &lt;- grep(&quot;w&quot;, case.control)</span>
<span class="c">##         if (length(ck1) == 0 &amp; length(ck2) == 0)</span>
<span class="c">##             stop(&quot;choose either case.control = \&quot;prior\&quot; &quot;, &quot;or case.control = \&quot;weighting\&quot;&quot;)</span>
<span class="c">##         if (length(ck2) == 0)</span>
<span class="c">##             weighting &lt;- FALSE</span>
<span class="c">##         else weighting &lt;- TRUE</span>
<span class="c">##     }</span>
<span class="c">##     else weighting &lt;- FALSE</span>
<span class="c">##     if (length(tau) &gt; 2)</span>
<span class="c">##         stop(&quot;tau must be a vector of length less than or equal to 2&quot;)</span>
<span class="c">##     else if (length(tau) == 2) {</span>
<span class="c">##         mf[[1]] &lt;- relogit</span>
<span class="c">##         res &lt;- list()</span>
<span class="c">##         mf$tau &lt;- min(tau)</span>
<span class="c">##         res$lower.estimate &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         mf$tau &lt;- max(tau)</span>
<span class="c">##         res$upper.estimate &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         res$formula &lt;- formula</span>
<span class="c">##         class(res) &lt;- c(&quot;Relogit2&quot;, &quot;Relogit&quot;)</span>
<span class="c">##         return(res)</span>
<span class="c">##     }</span>
<span class="c">##     else {</span>
<span class="c">##         mf[[1]] &lt;- glm</span>
<span class="c">##         mf$family &lt;- binomial(link = &quot;logit&quot;)</span>
<span class="c">##         y2 &lt;- model.response(model.frame(mf$formula, data))</span>
<span class="c">##         if (is.matrix(y2))</span>
<span class="c">##             y &lt;- y2[, 1]</span>
<span class="c">##         else y &lt;- y2</span>
<span class="c">##         ybar &lt;- mean(y)</span>
<span class="c">##         if (weighting) {</span>
<span class="c">##             w1 &lt;- tau/ybar</span>
<span class="c">##             w0 &lt;- (1 - tau)/(1 - ybar)</span>
<span class="c">##             wi &lt;- w1 * y + w0 * (1 - y)</span>
<span class="c">##             mf$weights &lt;- wi</span>
<span class="c">##         }</span>
<span class="c">##         res &lt;- eval(as.call(mf), parent.frame())</span>
<span class="c">##         res$call &lt;- match.call(expand.dots = TRUE)</span>
<span class="c">##         res$tau &lt;- tau</span>
<span class="c">##         X &lt;- model.matrix(res)</span>
<span class="c">##         if (bias.correct) {</span>
<span class="c">##             pihat &lt;- fitted(res)</span>
<span class="c">##             if (is.null(tau))</span>
<span class="c">##                 wi &lt;- rep(1, length(y))</span>
<span class="c">##             else if (weighting)</span>
<span class="c">##                 res$weighting &lt;- TRUE</span>
<span class="c">##             else {</span>
<span class="c">##                 w1 &lt;- tau/ybar</span>
<span class="c">##                 w0 &lt;- (1 - tau)/(1 - ybar)</span>
<span class="c">##                 wi &lt;- w1 * y + w0 * (1 - y)</span>
<span class="c">##                 res$weighting &lt;- FALSE</span>
<span class="c">##             }</span>
<span class="c">##             W &lt;- pihat * (1 - pihat) * wi</span>
<span class="c">##             Qdiag &lt;- lm.influence(lm(y ~ X - 1, weights = W))$hat/W</span>
<span class="c">##             if (is.null(tau))</span>
<span class="c">##                 xi &lt;- 0.5 * Qdiag * (2 * pihat - 1)</span>
<span class="c">##             else xi &lt;- 0.5 * Qdiag * ((1 + w0) * pihat - w0)</span>
<span class="c">##             res$coefficients &lt;- res$coefficients - lm(xi ~ X -</span>
<span class="c">##                 1, weights = W)$coefficients</span>
<span class="c">##             res$bias.correct &lt;- TRUE</span>
<span class="c">##         }</span>
<span class="c">##         else res$bias.correct &lt;- FALSE</span>
<span class="c">##         if (!is.null(tau) &amp; !weighting) {</span>
<span class="c">##             if (tau &lt;= 0 || tau &gt;= 1)</span>
<span class="c">##                 stop(&quot;\ntau needs to be between 0 and 1.\n&quot;)</span>
<span class="c">##             res$coefficients[&quot;(Intercept)&quot;] &lt;- res$coefficients[&quot;(Intercept)&quot;] -</span>
<span class="c">##                 log(((1 - tau)/tau) * (ybar/(1 - ybar)))</span>
<span class="c">##             res$prior.correct &lt;- TRUE</span>
<span class="c">##             res$weighting &lt;- FALSE</span>
<span class="c">##         }</span>
<span class="c">##         else res$prior.correct &lt;- FALSE</span>
<span class="c">##         if (is.null(res$weighting))</span>
<span class="c">##             res$weighting &lt;- FALSE</span>
<span class="c">##         res$linear.predictors &lt;- t(res$coefficients) %*% t(X)</span>
<span class="c">##         res$fitted.values &lt;- 1/(1 + exp(-res$linear.predictors))</span>
<span class="c">##         res$zelig &lt;- &quot;Relogit&quot;</span>
<span class="c">##         class(res) &lt;- c(&quot;Relogit&quot;, &quot;glm&quot;)</span>
<span class="c">##         return(res)</span>
<span class="c">##     }</span>
<span class="c">## })(formula = cbind(conflict, 1 - conflict) ~ major + contig +</span>
<span class="c">##     power + maxdem + mindem + years, data = ., tau = 0.005)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)        major       contig        power       maxdem</span>
<span class="c">##    -7.13001      2.43197      4.10805      1.05358      0.04804</span>
<span class="c">##      mindem        years</span>
<span class="c">##    -0.06413     -0.06294</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 3125 Total (i.e. Null);  3119 Residual</span>
<span class="c">## Null Deviance:       3979</span>
<span class="c">## Residual Deviance: 1869  AIC: 1883</span>
<span class="c">##</span>
<span class="c">## $formula</span>
<span class="c">## cbind(conflict, 1 - conflict) ~ major + contig + power + maxdem +</span>
<span class="c">##     mindem + years</span>
<span class="c">## &lt;environment: 0x7ff335918588&gt;</span>
<span class="c">##</span>
<span class="c">## attr(,&quot;class&quot;)</span>
<span class="c">## [1] &quot;Relogit2&quot; &quot;Relogit&quot;</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the explanatory variables to their means:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out2 <span class="o">&lt;-</span> setx<span class="p">(</span>z.out2<span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out2<span class="p">,</span> x <span class="o">=</span> x.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Error in UseMethod(&quot;vcov&quot;): no applicable method for &#39;vcov&#39; applied to an object of class &quot;c(&#39;Relogit2&#39;, &#39;Relogit&#39;)&quot;</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##              mean sd          50%         2.5%        97.5%</span>
<span class="c">## [1,] 2.220446e-16  0 2.220446e-16 2.220446e-16 2.220446e-16</span>
<span class="c">## pv</span>
<span class="c">##      [,1]</span>
<span class="c">## [1,]    1</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="" src="_images/unnamed-chunk-18-1.png" />
</div>
<p>The cost of giving a range of values for <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> is that point
estimates are not available for quantities of interest. Instead,
quantities are presented as confidence intervals with significance less
than or equal to a specified level (e.g., at least 95% of the
simulations are contained in the nominal 95% confidence interval).</p>
</div>
<div class="section" id="id53">
<h3>Model<a class="headerlink" href="#id53" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">Like the standard logistic regression, the <em>stochastic component</em> for
the rare events logistic regression is:</p>
<div class="math">
<p><img src="_images/math/e6ab2177225a79f988ab416b0353405999e15789.png" alt="Y_i \; \sim \; \textrm{Bernoulli}(\pi_i),"/></p>
</div><p>where <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> is the binary dependent variable, and takes a value
of either 0 or 1.</p>
</li>
<li><p class="first">The <em>systematic component</em> is:</p>
<div class="math">
<p><img src="_images/math/59cf11e927045c99a310cdaa15d32bc7cfacb160.png" alt="\pi_i \; = \; \frac{1}{1 + \exp(-x_i \beta)}."/></p>
</div></li>
<li><p class="first">If the sample is generated via a case-control (or choice-based)
design, such as when drawing all events (or “cases”) and a sample
from the non-events (or “controls”) and going backwards to collect
the explanatory variables, you must correct for selecting on the
dependent variable. While the slope coefficients are approximately
unbiased, the constant term may be significantly biased. Zelig has
two methods for case control correction:</p>
<ol class="arabic">
<li><p class="first">The “prior correction” method adjusts the intercept term. Let
<img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> be the true population fraction of events,
<img class="math" src="_images/math/eff822074d94619ea86dcccd9f1e410360a711c3.png" alt="\bar{y}"/> the fraction of events in the sample, and
<img class="math" src="_images/math/7f7c57d3dd82a0c7d00c82f510f14400a1cc4435.png" alt="\hat{\beta_0}"/> the uncorrected intercept term. The
corrected intercept <img class="math" src="_images/math/90acdaf850915294d7ef3bdc540005ee22197903.png" alt="\beta_0"/> is:</p>
<div class="math">
<p><img src="_images/math/b3f4429515a74871c61649fe2b350bfe49da7123.png" alt="\beta =  \hat{\beta_0} - \ln \left[ \bigg( \frac{1 - \tau}{\tau}
  \bigg) \bigg( \frac{\bar{y}}{1 - \bar{y}} \bigg) \right]."/></p>
</div></li>
<li><p class="first">The “weighting” method performs a weighted logistic regression to
correct for a case-control sampling design. Let the 1 subscript
denote observations for which the dependent variable is observed
as a 1, and the 0 subscript denote observations for which the
dependent variable is observed as a 0. Then the vector of weights
<img class="math" src="_images/math/b4e1bc8e2dfd2c26158273aef285b08a816f960a.png" alt="w_i"/></p>
<div class="math">
<p><img src="_images/math/50c4c6190b58f44612c516cb9e9ac3e6420f5b9c.png" alt="\begin{aligned}
w_1 &amp;=&amp; \frac{\tau}{\bar{y}} \\
w_0 &amp;=&amp; \frac{(1 - \tau)}{(1 - \bar{y})} \\
w_i &amp;=&amp; w_1 Y_i + w_0 (1 - Y_i)\end{aligned}"/></p>
</div></li>
</ol>
<p>If <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> is unknown, you may alternatively specify an upper
and lower bound for the possible range of <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/>. In this case,
the relogit procedure uses “robust Bayesian” methods to generate a
confidence interval (rather than a point estimate) for each quantity
of interest. The nominal coverage of the confidence interval is at
least as great as the actual coverage.</p>
</li>
<li><p class="first">By default, estimates of the the coefficients <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> are
bias-corrected to account for finite sample or rare events bias. In
addition, quantities of interest, such as predicted probabilities,
are also corrected of rare-events bias. If <img class="math" src="_images/math/d8af2dda6c66ca3f8936a4c3f429636ab9950ddc.png" alt="\widehat{\beta}"/>
are the uncorrected logit coefficients and
bias(<img class="math" src="_images/math/d8af2dda6c66ca3f8936a4c3f429636ab9950ddc.png" alt="\widehat{\beta}"/>) is the bias term, the corrected
coefficients <img class="math" src="_images/math/4024b5e1e832a4e71c716cc1f7383be57826b0b0.png" alt="\tilde{\beta}"/> are</p>
<div class="math">
<p><img src="_images/math/321de0f40e0675fea03bff2a20ea559cb463d6ea.png" alt="\widehat{\beta} - \textrm{bias}(\widehat{\beta}) = \tilde{\beta}"/></p>
</div><p>The bias term is</p>
<div class="math">
<p><img src="_images/math/b0553d525251be7ec5507573c00d1e00eee0e435.png" alt="\textrm{bias}(\widehat{\beta}) = (X'WX)^{-1} X'W \xi"/></p>
</div><p>where</p>
<div class="math">
<p><img src="_images/math/ffeae8f68e041844fe15a30663badb58587cb67d.png" alt="\begin{aligned}
\xi_i &amp;=&amp; 0.5 Q_{ii} \Big( (1 + w-1)\widehat{\pi}_i - w_1 \Big) \\
Q &amp;=&amp; X(X'WX)^{-1} X' \\
W = \textrm{diag}\{\widehat{\pi}_i (1 - \widehat{\pi}_i) w_i\}\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/b4e1bc8e2dfd2c26158273aef285b08a816f960a.png" alt="w_i"/> and <img class="math" src="_images/math/ee71d9a60992887aa4b7a1e010886633661a7ada.png" alt="w_1"/> are given in the “weighting”
section above.</p>
</li>
</ul>
</div>
<div class="section" id="id54">
<h3>Quantities of Interest<a class="headerlink" href="#id54" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">For either one or no <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/>:</p>
<ul>
<li><p class="first">The expected values (qi$ev) for the rare events logit are
simulations of the predicted probability</p>
<div class="math">
<p><img src="_images/math/fd2856bc7fec152ef785c927c85393b0f70596ac.png" alt="E(Y) = \pi_i =
    \frac{1}{1 + \exp(-x_i \beta)},"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its posterior.</p>
</li>
<li><p class="first">The predicted value (qi$pr) is a draw from a binomial distribution
with mean equal to the simulated <img class="math" src="_images/math/916754f7e7de4247fa27c52f610948b4055311fe.png" alt="\pi_i"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) is defined as</p>
<div class="math">
<p><img src="_images/math/9b4dec5fa18c43963e1e8ce022f3dce869dc2403.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1, \tau) - \Pr(Y = 1 \mid x, \tau)."/></p>
</div></li>
<li><p class="first">The risk ratio (qi$rr) is defined as</p>
<div class="math">
<p><img src="_images/math/840169782526b263d8745f139e906c4cbb6be812.png" alt="\textrm{RR} = \Pr(Y = 1 \mid x_1, \tau) \ / \ \Pr(Y = 1 \mid x, \tau)."/></p>
</div></li>
</ul>
</li>
<li><p class="first">For a range of <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> defined by <img class="math" src="_images/math/9c8d376f26cfc87ab8a0956d91e826558ef973d6.png" alt="[\tau_1, \tau_2]"/>, each
of the quantities of interest are <img class="math" src="_images/math/4f7a2e44ef1042deffae3520085171846386e918.png" alt="n \times 2"/> matrices, which
report the lower and upper bounds, respectively, for a confidence
interval with nominal coverage at least as great as the actual
coverage. At worst, these bounds are conservative estimates for the
likely range for each quantity of interest. Please refer to for the
specific method of calculating bounded quantities of interest.</p>
</li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the counterfactual predicted value of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id55">
<h3>Output Values<a class="headerlink" href="#id55" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">relogit,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>.</p>
</div>
<div class="section" id="differences-with-stata-version">
<h3>Differences with Stata Version<a class="headerlink" href="#differences-with-stata-version" title="Permalink to this headline">¶</a></h3>
<p>The Stata version of ReLogit and the R implementation differ slightly in
their coefficient estimates due to differences in the matrix inversion
routines implemented in R and Stata. Zelig uses orthogonal-triangular
decomposition (through lm.influence()) to compute the bias term, which
is more numerically stable than standard matrix calculations.</p>
</div>
<div class="section" id="id56">
<h3>See also<a class="headerlink" href="#id56" title="Permalink to this headline">¶</a></h3>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-tobit">
<span id="ztobit"></span><h2>zelig-tobit<a class="headerlink" href="#zelig-tobit" title="Permalink to this headline">¶</a></h2>
<p>Linear Regression for a Left-Censored Dependent Variable</p>
<p>Tobit regression estimates a linear regression model for a left-censored
dependent variable, where the dependent variable is censored from below.
While the classical tobit model has values censored at 0, you may select
another censoring point. For other linear regression models with fully
observed dependent variables, see Bayesian regression (), maximum
likelihood normal regression (), or least squares ().</p>
<div class="section" id="id57">
<h3>Syntax<a class="headerlink" href="#id57" title="Permalink to this headline">¶</a></h3>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> ztobit<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> below <span class="o">=</span> <span class="m">0</span><span class="p">,</span> above <span class="o">=</span> <span class="kc">Inf</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> below <span class="o">=</span> <span class="m">0</span><span class="p">,</span> above <span class="o">=</span> <span class="kc">Inf</span><span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;tobit&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="inputs">
<h3>Inputs<a class="headerlink" href="#inputs" title="Permalink to this headline">¶</a></h3>
<p>zelig() accepts the following arguments to specify how the dependent
variable is censored.</p>
<ul>
<li><p class="first"><tt class="docutils literal"><span class="pre">below</span></tt>: (defaults to 0) The point at which the dependent variable
is censored from below. If any values in the dependent variable are
observed to be less than the censoring point, it is assumed that that
particular observation is censored from below at the observed value.
(See for a Bayesian implementation that supports both left and right
censoring.)</p>
</li>
<li><p class="first">robust: defaults to FALSE. If TRUE, zelig() computes robust standard
errors based on sandwich estimators (see and ) and the options
selected in cluster.</p>
</li>
<li><p class="first">cluster: if robust = TRUE, you may select a variable to define groups
of correlated observations. Let x3 be a variable that consists of
either discrete numeric values, character strings, or factors that
define strata. Then</p>
<div class="highlight-python"><div class="highlight"><pre>&gt; z.out &lt;- zelig(y ~ x1 + x2, robust = TRUE, cluster = &quot;x3&quot;,
                 model = &quot;tobit&quot;, data = mydata)
</pre></div>
</div>
<p>means that the observations can be correlated within the strata
defined by the variable x3, and that robust standard errors should be
calculated according to those clusters. If robust = TRUE but cluster
is not specified, zelig() assumes that each observation falls into
its own cluster.</p>
</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(survreg)</span></tt> for more information.</p>
</div>
<div class="section" id="id58">
<h3>Examples<a class="headerlink" href="#id58" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id59">
<h4>Basic Example<a class="headerlink" href="#id59" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>tobin<span class="p">)</span>
</pre></div>
</div>
<p>Estimating linear regression using <tt class="docutils literal"><span class="pre">tobit</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>durable <span class="o">~</span> age <span class="o">+</span> quant<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;tobit&quot;</span><span class="p">,</span> data <span class="o">=</span> tobin<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Kosuke Imai, Gary King, Olivia Lau. 2011.</span>
<span class="c">##   tobit: Linear regression for Left-Censored Dependent Variable</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution given <tt class="docutils literal"><span class="pre">x.out</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean        sd      50%      2.5%    97.5%</span>
<span class="c">## 1 1.545981 0.6461613 1.461303 0.5197283 2.983462</span>
<span class="c">## pv</span>
<span class="c">##          mean      sd      50% 2.5%    97.5%</span>
<span class="c">## [1,] 3.003191 3.90936 1.321498    0 13.24722</span>
</pre></div>
</div>
</div>
<div class="section" id="id60">
<h4>Simulating First Differences<a class="headerlink" href="#id60" title="Permalink to this headline">¶</a></h4>
<p>Set explanatory variables to their default(mean/mode) values, with
high (80th percentile) and low (20th percentile) liquidity ratio
(<tt class="docutils literal"><span class="pre">quant</span></tt>):</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> quant <span class="o">=</span> quantile<span class="p">(</span>tobin<span class="o">$</span>quant<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.8</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> quant <span class="o">=</span> quantile<span class="p">(</span>tobin<span class="o">$</span>quant<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.2</span><span class="p">))</span>
</pre></div>
</div>
<p>Estimating the first difference for the effect of high versus low
liquidity ratio on duration(<tt class="docutils literal"><span class="pre">durable</span></tt>):</p>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean        sd      50%      2.5%    97.5%</span>
<span class="c">## 1 1.227702 0.7762866 1.068474 0.1532736 3.145453</span>
<span class="c">## pv</span>
<span class="c">##          mean      sd      50% 2.5%   97.5%</span>
<span class="c">## [1,] 2.875352 3.95269 1.043619    0 12.9054</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean        sd      50%      2.5%    97.5%</span>
<span class="c">## 1 2.017149 0.9548178 1.884671 0.5887645 4.321557</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50% 2.5%    97.5%</span>
<span class="c">## [1,] 3.468774 4.348168 2.061813    0 14.72022</span>
<span class="c">## fd</span>
<span class="c">##        mean       sd       50%      2.5%    97.5%</span>
<span class="c">## 1 0.7894464 1.155566 0.6871567 -1.267174 3.044403</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-tobit" src="_images/Zelig-tobit-1.png" />
<p class="caption">Zelig-tobit</p>
</div>
</div>
</div>
<div class="section" id="id61">
<h3>Model<a class="headerlink" href="#id61" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">Let <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> be a latent dependent variable which is distributed
with <em>stochastic</em> component</p>
<div class="math">
<p><img src="_images/math/f04da308df2549fd3f3ddac6d6294d0fe68c7536.png" alt="\begin{aligned}
Y_i^* &amp; \sim &amp; \textrm{Normal}(\mu_i, \sigma^2) \\\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> is a vector means and <img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/> is a
scalar variance parameter. <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> is not directly observed,
however. Rather we observed <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> which is defined as:</p>
<div class="math">
<p><img src="_images/math/c02c4c1f67cbb07aff7eef9e34f1779d12f26ae4.png" alt="Y_i = \left\{
\begin{array}{lcl}
Y_i^*  &amp;\textrm{if} &amp; c &lt;Y_i^* \\
c    &amp;\textrm{if} &amp; c \ge Y_i^*
\end{array}\right."/></p>
</div><p>where <img class="math" src="_images/math/65868d23a5bfe5b3b2d819386b19c14fa36af134.png" alt="c"/> is the lower bound below which <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> is
censored.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/bf760b1377dc29b5986cdb3a1ea4bc3174dcb094.png" alt="\begin{aligned}
\mu_{i} &amp;=&amp; x_{i} \beta,\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id62">
<h3>Quantities of Interest<a class="headerlink" href="#id62" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the tobit regression model are
the same as the expected value of <img class="math" src="_images/math/e5acb8ce977a9706090eb708ea2ae8ab001207b8.png" alt="Y*"/>:</p>
<div class="math">
<p><img src="_images/math/9ae22946494ae2cba96f0b452f7f93e671f63d93.png" alt="E(Y^* | X) = \mu_{i} = x_{i} \beta"/></p>
</div></li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) for the tobit regression model is
defined as</p>
<div class="math">
<p><img src="_images/math/3552382628dce9e0b3aa47fe65f61406a39686f1.png" alt="\begin{aligned}
\text{FD}=E(Y^* \mid x_{1}) - E(Y^* \mid x).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/fd39d2da6c33f4ce8a5109998fdcc63ae68141dc.png" alt="\begin{aligned}
\frac{1}{\sum t_{i}}\sum_{i:t_{i}=1}[E[Y^*_{i}(t_{i}=1)]-E[Y^*_{i}(t_{i}=0)]],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
</ul>
</div>
<div class="section" id="id63">
<h3>Output Values<a class="headerlink" href="#id63" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;tobit&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may examine the available information in <a href="#id64"><span class="problematic" id="id65">``</span></a>z.out`.</p>
</div>
<div class="section" id="id66">
<h3>See also<a class="headerlink" href="#id66" title="Permalink to this headline">¶</a></h3>
<p>The tobit function is part of the survival library by Terry Therneau,
ported to R by Thomas Lumley. Advanced users may wish to refer to
<tt class="docutils literal"><span class="pre">help(survfit)</span></tt> in the survival library.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-weibull">
<span id="zweibull"></span><h2>zelig-weibull<a class="headerlink" href="#zelig-weibull" title="Permalink to this headline">¶</a></h2>
<p>Weibull Regression for Duration Dependent Variables</p>
<p>Choose the Weibull regression model if the values in your dependent
variable are duration observations. The Weibull model relaxes the
exponential model’s assumption of constant hazard, and allows the
hazard rate to increase or decrease monotonically with respect to
elapsed time.</p>
<div class="section" id="id67">
<h3>Syntax<a class="headerlink" href="#id67" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zweibull<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>Y<span class="p">,</span> C<span class="p">)</span> <span class="o">~</span> X<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;weibull&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>Weibull models require that the dependent variable be in the form
Surv(Y, C), where Y and C are vectors of length <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>. For each
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> in 1, …, <img class="math" src="_images/math/413f8a8e40062a9090d9d50b88bc7b551b314c26.png" alt="n"/>, the value <img class="math" src="_images/math/410a65f64e2a609a2a7f544ed72f0982b719807a.png" alt="y_i"/> is the
duration (lifetime, for example), and the associated <img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is a
binary variable such that <img class="math" src="_images/math/45bd383d229a85e1c894b645ee5ce5cfad178bdf.png" alt="c_i = 1"/> if the duration is not
censored (<em>e.g.</em>, the subject dies during the study) or <img class="math" src="_images/math/839e310a40128c2d71f158db2ee1ff3a1a40b81b.png" alt="c_i = 0"/>
if the duration is censored (<em>e.g.</em>, the subject is still alive at the
end of the study). If <img class="math" src="_images/math/496867ca68c428fa123f26f0f73997632e61e116.png" alt="c_i"/> is omitted, all Y are assumed to be
completed; that is, time defaults to 1 for all observations.</p>
</div>
<div class="section" id="id68">
<h3>Input Values<a class="headerlink" href="#id68" title="Permalink to this headline">¶</a></h3>
<p>In addition to the standard inputs, zelig() takes the following
additional options for weibull regression:</p>
<ul class="simple">
<li>robust: defaults to FALSE. If TRUE, zelig() computes robust standard
errors based on sandwich estimators (see and ) based on the options
in cluster.</li>
<li>cluster: if robust = TRUE, you may select a variable to define groups
of correlated observations. Let x3 be a variable that consists of
either discrete numeric values, character strings, or factors that
define strata. Then</li>
</ul>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x1 <span class="o">+</span> x2<span class="p">,</span> robust <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> cluster <span class="o">=</span> <span class="s">&quot;x3&quot;</span><span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;exp&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
</pre></div>
</div>
<p>means that the observations can be correlated within the strata defined by the variable x3, and that robust standard errors should be calculated according to those clusters. If <cite>robust = TRUE</cite> but cluster is not specified, zelig() assumes that each observation falls into its own cluster.</p>
</div>
<div class="section" id="id69">
<h3>Example<a class="headerlink" href="#id69" title="Permalink to this headline">¶</a></h3>
<p>Attach the sample data:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>coalition<span class="p">)</span>
</pre></div>
</div>
<p>Estimate the model:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Surv<span class="p">(</span>duration<span class="p">,</span> ciep12<span class="p">)</span> <span class="o">~</span> fract <span class="o">+</span> numst2<span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;weibull&quot;</span><span class="p">,</span> data <span class="o">=</span> coalition<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Olivia Lau, Kosuke Imai, Gary King. 2007.</span>
<span class="c">##   weibull: Weibull Regression for Duration Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>View the regression output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## survival::survreg(formula = Surv(duration, ciep12) ~ fract +</span>
<span class="c">##     numst2, data = ., dist = &quot;weibull&quot;, model = FALSE)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">##  (Intercept)        fract       numst2</span>
<span class="c">##  5.498500132 -0.003838969  0.452319329</span>
<span class="c">##</span>
<span class="c">## Scale= 0.9566212</span>
<span class="c">##</span>
<span class="c">## Loglik(model)= -1077   Loglik(intercept only)= -1100.6</span>
<span class="c">##  Chisq= 47.18 on 2 degrees of freedom, p= 5.7e-11</span>
<span class="c">## n= 314</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set the baseline values (with the ruling coalition in the minority) and
the alternative values (with the ruling coalition in the majority) for
X:</p>
<div class="highlight-r"><div class="highlight"><pre>x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate expected values and first differences:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.low<span class="p">,</span> x1 <span class="o">=</span> x.high<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##      mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 58.6205 9.625224 57.78639 43.05993 79.42536</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%        2.5%    97.5%</span>
<span class="c">## [1,] 52.35899 140.9593 6.222801 0.002344735 382.6828</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##       mean       sd      50%     2.5%   97.5%</span>
<span class="c">## 1 92.29395 15.55911 90.39034 65.83954 126.616</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%         2.5%    97.5%</span>
<span class="c">## [1,] 83.19999 305.7794 8.686118 0.0009874311 666.0367</span>
<span class="c">## fd</span>
<span class="c">##       mean       sd      50%     2.5%    97.5%</span>
<span class="c">## 1 33.67345 10.78773 32.90091 15.41528 57.22683</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-weibull" src="_images/Zelig-weibull-1.png" />
<p class="caption">Zelig-weibull</p>
</div>
</div>
<div class="section" id="id70">
<h3>Model<a class="headerlink" href="#id70" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> be the survival time for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>. This
variable might be censored for some observations at a fixed time
<img class="math" src="_images/math/86ef86313546967dcb442753dcc6f50484dc244b.png" alt="y_c"/> such that the fully observed dependent variable,
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/>, is defined as</p>
<div class="math">
<p><img src="_images/math/744ffb7f8c1f1c604aeed5fe6d625e69fde6a710.png" alt="Y_i = \left\{ \begin{array}{ll}
      Y_i^* &amp; \textrm{if }Y_i^* \leq y_c \\
      y_c &amp; \textrm{if }Y_i^* &gt; y_c
    \end{array} \right."/></p>
</div><ul>
<li><p class="first">The <em>stochastic component</em> is described by the distribution of the
partially observed variable <img class="math" src="_images/math/e4b82f5d0d0b135c68864771e045dc913778a223.png" alt="Y^*"/>. We assume <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/>
follows the Weibull distribution whose density function is given by</p>
<div class="math">
<p><img src="_images/math/7465fec3e361e82de62a1c101669bd8438b47b28.png" alt="f(y_i^*\mid \lambda_i, \alpha) = \frac{\alpha}{\lambda_i^\alpha}
    y_i^{* \alpha-1} \exp \left\{ -\left( \frac{y_i^*}{\lambda_i}
\right)^{\alpha} \right\}"/></p>
</div><p>for <img class="math" src="_images/math/19e86e64ce0de9757bab9d46f47a8550f2992135.png" alt="y_i^* \ge 0"/>, the scale parameter <img class="math" src="_images/math/1edb3935292eba7c0b46ea53b338ca2584694a97.png" alt="\lambda_i &gt; 0"/>,
and the shape parameter <img class="math" src="_images/math/7f2c98bf462cba6083cf18483ba9510e3c2fd3d3.png" alt="\alpha &gt; 0"/>. The mean of this
distribution is :math:<a href="#id71"><span class="problematic" id="id72">`</span></a>lambda_i</p>
<blockquote>
<div><p>Gamma(1 + 1 / alpha)`. When <img class="math" src="_images/math/7eab7af9a483d7c6e4e628ba496fb65c5c56faac.png" alt="\alpha = 1"/>, the distribution</p>
</div></blockquote>
<p>reduces to the exponential distribution (see Section&nbsp;[exp]). (Note
that the output from zelig() parameterizes
scale:math:` = 1 / alpha`.)</p>
<p>In addition, survival models like the Weibull have three additional
properties. The hazard function <img class="math" src="_images/math/8f59c002cacf4a159c9d1b7ccfa141d30d5dc65c.png" alt="h(t)"/> measures the probability
of not surviving past time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> given survival up to <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>.
In general, the hazard function is equal to <img class="math" src="_images/math/89ae40bbb08f26a2d7a02d4d5af4526f1a20c029.png" alt="f(t)/S(t)"/> where
the survival function <img class="math" src="_images/math/cc2e13874b1a2ecf7f79657e6c3a29960829dcc3.png" alt="S(t) = 1 - \int_{0}^t f(s) ds"/>
represents the fraction still surviving at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. The
cumulative hazard function <img class="math" src="_images/math/30319a828ec4029a092aefb503bd693e5776e59b.png" alt="H(t)"/> describes the probability of
dying before time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. In general,
<img class="math" src="_images/math/17b2212e68f15f537abbafa6fa716bb8369f7372.png" alt="H(t)= \int_{0}^{t} h(s) ds = -\log S(t)"/>. In the case of the
Weibull model,</p>
<div class="math">
<p><img src="_images/math/b80098a9b99d9cb7d2800bc57f1d405c54c1623c.png" alt="\begin{aligned}
h(t) &amp;=&amp; \frac{\alpha}{\lambda_i^{\alpha}} t^{\alpha - 1}  \\
S(t) &amp;=&amp;  \exp \left\{ -\left( \frac{t}{\lambda_i} \right)^{\alpha} \right\} \\
H(t) &amp;=&amp; \left( \frac{t}{\lambda_i} \right)^{\alpha}\end{aligned}"/></p>
</div><p>For the Weibull model, the hazard function <img class="math" src="_images/math/8f59c002cacf4a159c9d1b7ccfa141d30d5dc65c.png" alt="h(t)"/> can increase
or decrease monotonically over time.</p>
</li>
<li><p class="first">The <em>systematic component</em> <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/> is modeled as</p>
<div class="math">
<p><img src="_images/math/2ef3e09dc6e0d4710d1d8455992bbb32d072fbb7.png" alt="\lambda_i = \exp(x_i \beta),"/></p>
</div><p>where <img class="math" src="_images/math/33dfc32d00ebd5c5791c824010a155d9e5630b6f.png" alt="x_i"/> is the vector of explanatory variables, and
<img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
</ul>
</div>
<div class="section" id="id73">
<h3>Quantities of Interest<a class="headerlink" href="#id73" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (qi$ev) for the Weibull model are simulations of
the expected duration:</p>
<div class="math">
<p><img src="_images/math/17ade6859efd558a2071e20d2557eda701856a8f.png" alt="E(Y) = \lambda_i \, \Gamma (1 + \alpha^{-1}),"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and <img class="math" src="_images/math/ad59b6e24a4a00ac621801f8d7513d68be654ab5.png" alt="\alpha"/> from their sampling
distributions.</p>
</li>
<li><p class="first">The predicted value (qi$pr) is drawn from a distribution defined by
<img class="math" src="_images/math/b521bd4c806e41fc4d3ebfa9d96ddd5f134846e3.png" alt="(\lambda_i, \alpha)"/>.</p>
</li>
<li><p class="first">The first difference (qi$fd) in expected value is</p>
<div class="math">
<p><img src="_images/math/47e62bb3ada6780d6fd2b247bf221d015d2207d2.png" alt="\textrm{FD} = E(Y \mid x_1) - E(Y \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/f120915a77efb3200aab141647a59f51df9b4611.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      E[Y_i(t_i=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations are due to
uncertainty in simulating <img class="math" src="_images/math/f658a3cf4cf830cd65eab505c530fb4796fea0ed.png" alt="E[Y_i(t_i=0)]"/>, the counterfactual
expected value of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the treatment
group, under the assumption that everything stays the same except
that the treatment indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (att.pr) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/da7e72f6d651b92ad0874d3b40cd633ac38f5663.png" alt="\frac{1}{\sum_{i=1}^n t_i}\sum_{i:t_i=1}^n \left\{ Y_i(t_i=1) -
      \widehat{Y_i(t_i=0)} \right\},"/></p>
</div><p>where <img class="math" src="_images/math/64283d3d8caf93093b91aba9a61dbe26b5c18562.png" alt="t_i"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/cc6ccc003db65691c9d0aebb91b2001e0d752d18.png" alt="t_i=1"/>) and control (<img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>) groups. When
<img class="math" src="_images/math/3574cfcb10f1bd84fea967193a6ae17c6855de01.png" alt="Y_i(t_i=1)"/> is censored rather than observed, we replace it
with a simulation from the model given available knowledge of the
censoring process. Variation in the simulations are due to
uncertainty in simulating <img class="math" src="_images/math/8056bf28681ea8573bf29bfeb3c72dffa40a2fa7.png" alt="\widehat{Y_i(t_i=0)}"/>, the
counterfactual predicted value of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> for observations in the
treatment group, under the assumption that everything stays the same
except that the treatment indicator is switched to <img class="math" src="_images/math/906dfa4d36ff6837226de6ec10a5e18d37c41bc0.png" alt="t_i=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id74">
<h3>Output Values<a class="headerlink" href="#id74" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">weibull,</span> <span class="pre">data)</span></tt>, then you may examine
the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using <tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see
the coefficients by using z.out$coefficients, and a default summary of
information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements available through
the $ operator are listed below.</p>
<ul class="simple">
<li>From the zelig() output object z.out, you may extract:<ul>
<li>coefficients: parameter estimates for the explanatory variables.</li>
<li>icoef: parameter estimates for the intercept and “scale” parameter
<img class="math" src="_images/math/76a126bb9e9fed0f0693260dde6f83a2e4dadc5a.png" alt="1 / \alpha"/>.</li>
<li>var: the variance-covariance matrix.</li>
<li>loglik: a vector containing the log-likelihood for the model and
intercept only (respectively).</li>
<li>linear.predictors: a vector of the <img class="math" src="_images/math/f8237c9c4a91083105a2650415195fa49fe0bafc.png" alt="x_{i}\beta"/>.</li>
<li>df.residual: the residual degrees of freedom.</li>
<li>df.null: the residual degrees of freedom for the null model.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
</ul>
</li>
<li>Most of this may be conveniently summarized using summary(z.out).
From summary(z.out), you may additionally extract:<ul>
<li>table: the parameter estimates with their associated standard
errors, <img class="math" src="_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/>-values, and <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>-statistics.</li>
</ul>
</li>
<li>From the sim() output object s.out, you may extract quantities of
interest arranged as matrices indexed by simulation <img class="math" src="_images/math/c69691d64985442217922c8d34e835a9dea60178.png" alt="\times"/>
x-observation (for more than one x-observation). Available quantities
are:<ul>
<li>qi$ev: the simulated expected values for the specified values of
x.</li>
<li>qi$pr: the simulated predicted values drawn from a distribution
defined by <img class="math" src="_images/math/b521bd4c806e41fc4d3ebfa9d96ddd5f134846e3.png" alt="(\lambda_i, \alpha)"/>.</li>
<li>qi$fd: the simulated first differences between the simulated
expected values for x and x1.</li>
<li>qi$att.ev: the simulated average expected treatment effect for the
treated from conditional prediction models.</li>
<li>qi$att.pr: the simulated average predicted treatment effect for
the treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id75">
<h3>See also<a class="headerlink" href="#id75" title="Permalink to this headline">¶</a></h3>
<p>The Weibull model is part of the survival library by Terry Therneau,
ported to R by Thomas Lumley. Advanced users may wish to refer to
<tt class="docutils literal"><span class="pre">help(survfit)</span></tt> in the survival library.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-factorbayes">
<span id="zfactorbayes"></span><h2>zelig-factorbayes<a class="headerlink" href="#zelig-factorbayes" title="Permalink to this headline">¶</a></h2>
<p>Given some unobserved explanatory variables and observed dependent
variables, the Normal theory factor analysis model estimates the latent
factors. The model is implemented using a Markov Chain Monte Carlo
algorithm (Gibbs sampling with data augmentation). For factor analysis
with ordinal dependent variables, see ordered factor analysis (), and
for a mix of types of dependent variables, see the mixed factor analysis
model ().</p>
<div class="section" id="id76">
<h3>Syntax<a class="headerlink" href="#id76" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zfactorbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>cbind<span class="p">(</span>Y1 <span class="p">,</span>Y2<span class="p">,</span> Y3<span class="p">)</span> <span class="o">~</span> <span class="kc">NULL</span><span class="p">,</span> factors <span class="o">=</span> <span class="m">2</span><span class="p">,</span>
         model <span class="o">=</span> <span class="s">&quot;factor.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>cbind<span class="p">(</span>Y1 <span class="p">,</span>Y2<span class="p">,</span> Y3<span class="p">)</span> <span class="o">~</span> <span class="kc">NULL</span><span class="p">,</span> factors <span class="o">=</span> <span class="m">2</span><span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;factor.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id77">
<h3>Inputs<a class="headerlink" href="#id77" title="Permalink to this headline">¶</a></h3>
<p>zelig() takes the following functions for factor.bayes:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">Y1</span></tt>, <tt class="docutils literal"><span class="pre">Y2</span></tt>, and <tt class="docutils literal"><span class="pre">Y3</span></tt>: variables of interest in factor analysis
(manifest variables), assumed to be normally distributed. The model
requires a minimum of three manifest variables.</li>
<li><tt class="docutils literal"><span class="pre">factors</span></tt>: number of the factors to be fitted (defaults to 2).</li>
</ul>
</div>
<div class="section" id="additional-inputs">
<h3>Additional Inputs<a class="headerlink" href="#additional-inputs" title="Permalink to this headline">¶</a></h3>
<p>In addition, zelig() accepts the following additional arguments for
model specification:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">lambda.constraints</span></tt>: list containing the equality or inequality
constraints on the factor loadings. Choose from one of the following
forms:<ul>
<li>varname = list(): by default, no constraints are imposed.</li>
<li><tt class="docutils literal"><span class="pre">varname</span> <span class="pre">=</span> <span class="pre">list(d,</span> <span class="pre">c)</span></tt>: constrains the <img class="math" src="_images/math/425d86ba2f2979d75b7535c2bcf92c33ed6b285a.png" alt="d"/>th loading for
the variable named <tt class="docutils literal"><span class="pre">varname</span></tt> to be equal to <tt class="docutils literal"><span class="pre">c</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">varname</span> <span class="pre">=</span> <span class="pre">list(d,</span> <span class="pre">+)</span></tt>: constrains the <img class="math" src="_images/math/425d86ba2f2979d75b7535c2bcf92c33ed6b285a.png" alt="d"/>th loading for
the variable named <tt class="docutils literal"><span class="pre">varname</span></tt> to be positive;</li>
<li><tt class="docutils literal"><span class="pre">varname</span> <span class="pre">=</span> <span class="pre">list(d,</span> <span class="pre">-)</span></tt>: constrains the <img class="math" src="_images/math/425d86ba2f2979d75b7535c2bcf92c33ed6b285a.png" alt="d"/>th loading for
the variable named <tt class="docutils literal"><span class="pre">varname</span></tt> to be negative.</li>
</ul>
</li>
<li><tt class="docutils literal"><span class="pre">std.var</span></tt>: defaults to FALSE (manifest variables are rescaled to
zero mean, but retain observed variance). If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the manifest
variables are rescaled to be mean zero and unit variance.</li>
</ul>
<p>In addition, zelig() accepts the following additional inputs for
bayes.factor:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults to
20,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: defaults to FALSE. If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed 12345.</li>
<li><tt class="docutils literal"><span class="pre">Lambda.start</span></tt>: starting values of the factor loading matrix
<img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/>, either a scalar (all unconstrained loadings are set
to that value), or a matrix with compatible dimensions. The default
is <tt class="docutils literal"><span class="pre">NA</span></tt>, where the start value are set to be 0 for unconstrained
factor loadings, and 0.5 or <img class="math" src="_images/math/03e927fcb222d9b3da8313fec8a194aabd267f3c.png" alt="-"/>0.5 for constrained factor
loadings (depending on the nature of the constraints).</li>
<li><tt class="docutils literal"><span class="pre">Psi.start</span></tt>: starting values for the uniquenesses, either a scalar
(the starting values for all diagonal elements of <img class="math" src="_images/math/de6f6ec55da5bff04bdf591d5bb35136eb51b579.png" alt="\Psi"/> are
set to be this value), or a vector with length equal to the number of
manifest variables. In the latter case, the starting values of the
diagonal elements of <img class="math" src="_images/math/de6f6ec55da5bff04bdf591d5bb35136eb51b579.png" alt="\Psi"/> take the values of <tt class="docutils literal"><span class="pre">Psi.start</span></tt>.
The default value is <tt class="docutils literal"><span class="pre">NA</span></tt> where the starting values of the all the
uniquenesses are set to be 0.5.</li>
<li><tt class="docutils literal"><span class="pre">store.lambda</span></tt>: defaults to TRUE, which stores the posterior draws
of the factor loadings.</li>
<li><tt class="docutils literal"><span class="pre">store.scores</span></tt>: defaults to FALSE. If TRUE, stores the posterior
draws of the factor scores. (Storing factor scores may take large
amount of memory for a large number of draws or observations.)</li>
</ul>
<p>The model also accepts the following additional arguments to specify
prior parameters:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">l0</span></tt>: mean of the Normal prior for the factor loadings, either a
scalar or a matrix with the same dimensions as <img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/>. If a
scalar value, that value will be the prior mean for all the factor
loadings. Defaults to 0.</li>
<li><tt class="docutils literal"><span class="pre">L0</span></tt>: precision parameter of the Normal prior for the factor
loadings, either a scalar or a matrix with the same dimensions as
<img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/>. If <tt class="docutils literal"><span class="pre">L0</span></tt> takes a scalar value, then the precision
matrix will be a diagonal matrix with the diagonal elements set to
that value. The default value is 0, which leads to an improper prior.</li>
<li><tt class="docutils literal"><span class="pre">a0</span></tt>: the shape parameter of the Inverse Gamma prior for the
uniquenesses is <tt class="docutils literal"><span class="pre">a0/2</span></tt>. It can take a scalar value or a vector. The
default value is 0.001.</li>
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: the shape parameter of the Inverse Gamma prior for the
uniquenesses is <tt class="docutils literal"><span class="pre">b0/2</span></tt>. It can take a scalar value or a vector. The
default value is 0.001.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCfactanal)</span></tt> for more
information.</p>
</div>
<div class="section" id="id78">
<h3>Example<a class="headerlink" href="#id78" title="Permalink to this headline">¶</a></h3>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>swiss<span class="p">)</span>
names<span class="p">(</span>swiss<span class="p">)</span> <span class="o">&lt;-</span> c<span class="p">(</span><span class="s">&quot;Fert&quot;</span><span class="p">,</span> <span class="s">&quot;Agr&quot;</span><span class="p">,</span> <span class="s">&quot;Exam&quot;</span><span class="p">,</span> <span class="s">&quot;Educ&quot;</span><span class="p">,</span> <span class="s">&quot;Cath&quot;</span><span class="p">,</span> <span class="s">&quot;InfMort&quot;</span><span class="p">)</span>
</pre></div>
</div>
<p>Factor analysis:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>cbind<span class="p">(</span>Agr<span class="p">,</span> Exam<span class="p">,</span> Educ<span class="p">,</span> Cath<span class="p">,</span> InfMort<span class="p">)</span> <span class="o">~</span> <span class="kc">NULL</span><span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;factor.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> swiss<span class="p">,</span> factors <span class="o">=</span> <span class="m">2</span><span class="p">,</span> verbose <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
               a0 <span class="o">=</span> <span class="m">1</span><span class="p">,</span> b0 <span class="o">=</span> <span class="m">0.15</span><span class="p">,</span> burnin <span class="o">=</span> <span class="m">5000</span><span class="p">,</span> mcmc <span class="o">=</span> <span class="m">50000</span><span class="p">)</span>
</pre></div>
</div>
<p>Checking for convergence before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>algor <span class="o">&lt;-</span> try<span class="p">(</span>geweke.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">),</span> silent<span class="o">=</span><span class="k-Variable">T</span><span class="p">)</span>
<span class="kr">if</span> <span class="p">(</span>class<span class="p">(</span>algor<span class="p">)</span> <span class="o">==</span> <span class="s">&quot;try-error&quot;</span><span class="p">)</span>
    print<span class="p">(</span>algor<span class="p">)</span>
</pre></div>
</div>
<p>Since the algorithm did not converge, we now add some constraints on
<img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/> to optimize the algorithm:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>cbind<span class="p">(</span>Agr<span class="p">,</span> Exam<span class="p">,</span> Educ<span class="p">,</span> Cath<span class="p">,</span> InfMort<span class="p">)</span> <span class="o">~</span> <span class="kc">NULL</span><span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;factor.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> swiss<span class="p">,</span> factors <span class="o">=</span> <span class="m">2</span><span class="p">,</span>
               lambda.constraints <span class="o">=</span> list<span class="p">(</span>Exam <span class="o">=</span> list<span class="p">(</span><span class="m">1</span><span class="p">,</span><span class="s">&quot;+&quot;</span><span class="p">),</span>
                   Exam <span class="o">=</span> list<span class="p">(</span><span class="m">2</span><span class="p">,</span><span class="s">&quot;-&quot;</span><span class="p">),</span> Educ <span class="o">=</span> c<span class="p">(</span><span class="m">2</span><span class="p">,</span> <span class="m">0</span><span class="p">),</span>
                   InfMort <span class="o">=</span> c<span class="p">(</span><span class="m">1</span><span class="p">,</span> <span class="m">0</span><span class="p">)),</span>
               verbose <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span> a0 <span class="o">=</span> <span class="m">1</span><span class="p">,</span> b0 <span class="o">=</span> <span class="m">0.15</span><span class="p">,</span>
               burnin <span class="o">=</span> <span class="m">5000</span><span class="p">,</span> mcmc <span class="o">=</span> <span class="m">50000</span><span class="p">)</span>
geweke.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
heidel.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
raftery.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id79">
<h3>Model<a class="headerlink" href="#id79" title="Permalink to this headline">¶</a></h3>
<p>Suppose for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> we observe <img class="math" src="_images/math/28e003020d0ae96250b302d7d779c791f183f707.png" alt="K"/> variables and
hypothesize that there are <img class="math" src="_images/math/425d86ba2f2979d75b7535c2bcf92c33ed6b285a.png" alt="d"/> underlying factors such that:</p>
<div class="math">
<p><img src="_images/math/e2ef04e9babfcd829100c089b8409a81a9095115.png" alt="\begin{aligned}
Y_i = \Lambda \phi_i+\epsilon_i\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> is the vector of <img class="math" src="_images/math/28e003020d0ae96250b302d7d779c791f183f707.png" alt="K"/> manifest variables for
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>. <img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/> is the <img class="math" src="_images/math/990c0ad210f52f294b999caf66e3ff8d0cfaec86.png" alt="K \times d"/> factor
loading matrix and <img class="math" src="_images/math/ca5185b32984bfd8107d41af4a27499db30b7102.png" alt="\phi_i"/> is the <img class="math" src="_images/math/425d86ba2f2979d75b7535c2bcf92c33ed6b285a.png" alt="d"/>-vector of latent
factor scores. Both <img class="math" src="_images/math/7327d24dc2d12421147d5944f0fc23e67753f1b0.png" alt="\Lambda"/> and <img class="math" src="_images/math/10e009bdb83f96c5f47c58b34d5d4b12ef268d5b.png" alt="\phi"/> need to be
estimated.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by:</p>
<div class="math">
<p><img src="_images/math/0baa7603f3ae0325e5f490e904db0fb107e08c5c.png" alt="\begin{aligned}
\epsilon_{i}  \sim \textrm{Normal}(0, \Psi).\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/de6f6ec55da5bff04bdf591d5bb35136eb51b579.png" alt="\Psi"/> is a diagonal, positive definite matrix. The
diagonal elements of <img class="math" src="_images/math/de6f6ec55da5bff04bdf591d5bb35136eb51b579.png" alt="\Psi"/> are referred to as uniquenesses.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/5a20aedd49dd26956cd9f786bd39a83f21cc900c.png" alt="\begin{aligned}
\mu_i = E(Y_i) = \Lambda\phi_i\end{aligned}"/></p>
</div></li>
<li><p class="first">The independent conjugate <em>prior</em> for each <img class="math" src="_images/math/f19197f93cc1d1edf2411522363a0be6feb1651a.png" alt="\Lambda_{ij}"/> is
given by</p>
<div class="math">
<p><img src="_images/math/20bef115cb56894e312db6c08a7101a1288528a0.png" alt="\begin{aligned}
\Lambda_{ij} \sim \textrm{Normal}(l_{0_{ij}}, L_{0_{ij}}^{-1})
\textrm{ for } i=1,\ldots, k; \quad j=1,\ldots, d. \end{aligned}"/></p>
</div></li>
<li><p class="first">The independent conjugate <em>prior</em> for each <img class="math" src="_images/math/e3722bc14e22b59a627887c08fffa161387bef20.png" alt="\Psi_{ii}"/> is given
by</p>
<div class="math">
<p><img src="_images/math/183955a52e4eac8778c685478c004c6d2c001af3.png" alt="\begin{aligned}
\Psi_{ii} \sim \textrm{InverseGamma}(\frac{a_0}{2}, \frac{b_0}{2}), \textrm{ for }
i = 1, \ldots, k.\end{aligned}"/></p>
</div></li>
<li><p class="first">The <em>prior</em> for <img class="math" src="_images/math/ca5185b32984bfd8107d41af4a27499db30b7102.png" alt="\phi_i"/> is</p>
<div class="math">
<p><img src="_images/math/2ef6db0c88292f54d9310202c5304b0bac0707ab.png" alt="\begin{aligned}
\phi_i &amp;\sim&amp; \textrm{Normal}(0, I_d), \textrm{ for } i = 1, \ldots, n.\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/6fcd6923eaac15a18f3b304221cfac4f64034a64.png" alt="I_d"/> is a :math:` dtimes d ` identity matrix.</p>
</li>
</ul>
</div>
<div class="section" id="id80">
<h3>Output Values<a class="headerlink" href="#id80" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>cbind<span class="p">(</span>Y1<span class="p">,</span> Y2<span class="p">,</span> Y3<span class="p">),</span> model <span class="o">=</span> <span class="s">&quot;factor.bayes&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
<ul class="simple">
<li>From the <tt class="docutils literal"><span class="pre">zelig()</span></tt> output object <tt class="docutils literal"><span class="pre">z.out</span></tt>, you may extract:<ul>
<li><tt class="docutils literal"><span class="pre">coefficients</span></tt>: draws from the posterior distributions of the
estimated factor loadings and the uniquenesses. If
<tt class="docutils literal"><span class="pre">store.scores</span> <span class="pre">=</span> <span class="pre">TRUE</span></tt>, the estimated factors scores are also
contained in <tt class="docutils literal"><span class="pre">coefficients</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">data</span></tt>: the name of the input data frame.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: the random seed used in the model.</li>
</ul>
</li>
<li>Since there are no explanatory variables, the <tt class="docutils literal"><span class="pre">sim()</span></tt> procedure is
not applicable for factor analysis models.</li>
</ul>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-mlogitbayes">
<span id="zmlogitbayes"></span><h2>zelig-mlogitbayes<a class="headerlink" href="#zelig-mlogitbayes" title="Permalink to this headline">¶</a></h2>
<p>Use Bayesian multinomial logistic regression to model unordered
categorical variables. The dependent variable may be in the format of
either character strings or integer values. The model is estimated via a
random walk Metropolis algorithm or a slice sampler. See for the
maximum-likelihood estimation of this model.</p>
<div class="section" id="id81">
<h3>Syntax<a class="headerlink" href="#id81" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zmlogitbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;mlogit.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id82">
<h3>Additional Inputs<a class="headerlink" href="#id82" title="Permalink to this headline">¶</a></h3>
<p>zelig() accepts the following arguments for mlogit.bayes:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">baseline</span></tt>: either a character string or numeric value (equal to
one of the observed values in the dependent variable) specifying a
baseline category. The default value is <tt class="docutils literal"><span class="pre">NA</span></tt> which sets the
baseline to the first alphabetical or numerical unique value of the
dependent variable.</li>
</ul>
<p>The model accepts the following additional arguments to monitor the
Markov chains:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults to
10,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">mcmc.method</span></tt>: either “MH” or “slice”, specifying whether to use
Metropolis Algorithm or slice sampler. The default value is <tt class="docutils literal"><span class="pre">MH</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">tune</span></tt>: tuning parameter for the Metropolis-Hasting step, either a
scalar or a numeric vector (for <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> coefficients, enter a
<img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> vector). The tuning parameter should be set such that the
acceptance rate is satisfactory (between 0.2 and 0.5). The default
value is 1.1.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: defaults to <tt class="docutils literal"><span class="pre">FALSE</span></tt>. If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed of 12345.</li>
<li><tt class="docutils literal"><span class="pre">beta.start</span></tt>: starting values for the Markov chain, either a scalar
or a vector (for <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> coefficients, enter a <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> vector).
The default is <tt class="docutils literal"><span class="pre">NA</span></tt> where the maximum likelihood estimates are used
as the starting values.</li>
</ul>
<p>Use the following arguments to specify the priors for the model:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: prior mean for the coefficients, either a scalar or vector.
If a scalar, that value will be the prior mean for all the
coefficients. The default is 0.</li>
<li><tt class="docutils literal"><span class="pre">B0</span></tt>: prior precision parameter for the coefficients, either a
square matrix with the dimensions equal to the number of coefficients
or a scalar. If a scalar, that value times an identity matrix will be
the prior precision parameter. The default is 0 which leads to an
improper prior.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCmnl)</span></tt> for more information.</p>
</div>
<div class="section" id="id83">
<h3>Examples<a class="headerlink" href="#id83" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id84">
<h4>Basic Example<a class="headerlink" href="#id84" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>mexico<span class="p">)</span>
</pre></div>
</div>
<p>Estimating multinomial logistics regression using <tt class="docutils literal"><span class="pre">mlogit.bayes</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>vote88 <span class="o">~</span> pristr <span class="o">+</span> othcok <span class="o">+</span> othsocok<span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;mlogit.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mexico<span class="p">,</span>
               verbose <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Calculating MLEs and large sample var-cov matrix.</span>
<span class="c">## This may take a moment...</span>
<span class="c">## Inverting Hessian to get large sample var-cov matrix.</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Warning in if (mcmc.method == &quot;RWM&quot;) {: the condition has length &gt; 1 and</span>
<span class="c">## only the first element will be used</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Warning in if (mcmc.method == &quot;IndMH&quot;) {: the condition has length &gt; 1 and</span>
<span class="c">## only the first element will be used</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Ben Goodrich, Ying Lu. 2013.</span>
<span class="c">##   mlogitbayes: Bayesian Multinomial Logistic Regression for Dependent Variables with Unordered Categorical Values</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Checking for convergence before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>raftery.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample
averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution
given <tt class="docutils literal"><span class="pre">x.out</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##             mean         sd       50%      2.5%     97.5%</span>
<span class="c">## P(Y=1) 0.5613368 0.01592425 0.5615034 0.5306640 0.5914963</span>
<span class="c">## P(Y=2) 0.2099124 0.01273148 0.2098424 0.1854312 0.2350891</span>
<span class="c">## P(Y=3) 0.2287508 0.01360126 0.2285987 0.2033590 0.2558153</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">##      1      2      3</span>
<span class="c">## 0.5662 0.2013 0.2325</span>
</pre></div>
</div>
</div>
<div class="section" id="id85">
<h4>Simulating First Differences<a class="headerlink" href="#id85" title="Permalink to this headline">¶</a></h4>
<p>Estimating the first difference (and risk ratio) in the
probabilities of voting different candidates when <tt class="docutils literal"><span class="pre">pristr</span></tt> (the
strength of the PRI) is set to be weak (equal to 1) versus strong
(equal to 3) while all the other variables held at their default
values.</p>
<div class="highlight-r"><div class="highlight"><pre>x.weak <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> pristr <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
x.strong <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> pristr <span class="o">=</span> <span class="m">3</span><span class="p">)</span>
s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.strong<span class="p">,</span> x1 <span class="o">=</span> x.weak<span class="p">)</span>
summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##             mean         sd       50%      2.5%     97.5%</span>
<span class="c">## P(Y=1) 0.7156880 0.02127842 0.7158103 0.6725681 0.7561260</span>
<span class="c">## P(Y=2) 0.1270237 0.01458077 0.1265905 0.1000858 0.1562571</span>
<span class="c">## P(Y=3) 0.1572883 0.01646202 0.1568142 0.1260809 0.1909916</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">##      1      2      3</span>
<span class="c">## 0.7152 0.1262 0.1586</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##             mean         sd       50%      2.5%     97.5%</span>
<span class="c">## P(Y=1) 0.4028126 0.02357831 0.4028038 0.3563194 0.4483880</span>
<span class="c">## P(Y=2) 0.3037026 0.02130587 0.3029289 0.2638074 0.3470994</span>
<span class="c">## P(Y=3) 0.2934848 0.02189140 0.2931780 0.2517546 0.3372056</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">##      1      2      3</span>
<span class="c">## 0.3973 0.3027 0.3000</span>
<span class="c">## fd</span>
<span class="c">##              mean         sd        50%        2.5%      97.5%</span>
<span class="c">## P(Y=1) -0.3128754 0.03459857 -0.3128662 -0.38111485 -0.2442630</span>
<span class="c">## P(Y=2)  0.1766789 0.02735176  0.1764581  0.12360341  0.2313796</span>
<span class="c">## P(Y=3)  0.1361965 0.02881430  0.1363242  0.07966018  0.1935930</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id86">
<h3>Model<a class="headerlink" href="#id86" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> be the (unordered) categorical dependent variable for
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> which takes an integer values
<img class="math" src="_images/math/de0b4bef7a011d44e3288965ca51ff93922ab47b.png" alt="j=1, \ldots, J"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by:</p>
<div class="math">
<p><img src="_images/math/906e54ea6929de26cf7f3820c69245f3a00024d1.png" alt="\begin{aligned}
Y_{i} &amp;\sim&amp; \textrm{Multinomial}(Y_i \mid \pi_{ij}).\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/a10535a65002e2c0e4e6e687e44046b32c196ef2.png" alt="\pi_{ij}=\Pr(Y_i=j)"/> for <img class="math" src="_images/math/de0b4bef7a011d44e3288965ca51ff93922ab47b.png" alt="j=1, \ldots, J"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/ab6d29608bd86c84a1d71ca1c9569262e79620a6.png" alt="\begin{aligned}
\pi_{ij}=\frac{\exp(x_i\beta_j)}{\sum_{k=1}^J \exp(x_i\beta_k)},
\textrm{ for } j=1,\ldots, J-1,\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/ef646ee6b727ac98a22d9ae55808ac67708c05fc.png" alt="\beta_j"/> is the vector of
coefficient for category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>. Category <img class="math" src="_images/math/372c2a0702e1e5866656215408b37a2ddb8d2d45.png" alt="J"/> is assumed to
be the baseline category.</p>
</li>
<li><p class="first">The <em>prior</em> for <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is given by</p>
<div class="math">
<p><img src="_images/math/a0d0b43ddf020d952cdb73a3593a5b00365a955e.png" alt="\begin{aligned}
\beta_j \sim \textrm{Normal}_k\left(  b_{0},B_{0}^{-1}\right)
\textrm{ for } j = 1, \ldots, J-1,\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/9d80db7ae48caf02365282c5dff832fcbe0a74d5.png" alt="b_{0}"/> is the vector of means for the <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>
explanatory variables and <img class="math" src="_images/math/6a119b5ae9ee269e28bcaa4d39e47a3d3de4f5e5.png" alt="B_{0}"/> is the <img class="math" src="_images/math/3908ba50237c94909083edd0295cd1c9345b0a8c.png" alt="k \times k"/>
precision matrix (the inverse of a variance-covariance matrix).</p>
</li>
</ul>
</div>
<div class="section" id="id87">
<h3>Quantities of Interest<a class="headerlink" href="#id87" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the multinomial logistics
regression model are the predicted probability of belonging to each
category:</p>
<div class="math">
<p><img src="_images/math/b51bfcb8724675c7143042a5e6a9bd398e57568e.png" alt="\begin{aligned}
\Pr(Y_i=j)=\pi_{ij}=\frac{\exp(x_i \beta_j)}{\sum_{k=1}^J \exp(x_J
\beta_k)}, \quad \textrm{ for } j=1,\ldots, J-1,\end{aligned}"/></p>
</div><p>and</p>
<div class="math">
<p><img src="_images/math/07fa787f6d4b56aa983a189dd5a9e3644447804f.png" alt="\begin{aligned}
\Pr(Y_i=J)=1-\sum_{j=1}^{J-1}\Pr(Y_i=j)\end{aligned}"/></p>
</div><p>given the posterior draws of <img class="math" src="_images/math/ef646ee6b727ac98a22d9ae55808ac67708c05fc.png" alt="\beta_j"/> for all categories from
the MCMC iterations.</p>
</li>
<li><p class="first">The predicted values (<tt class="docutils literal"><span class="pre">qi$pr</span></tt>) are the draws of <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> from a
multinomial distribution whose parameters are the expected
values(<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) computed based on the posterior draws of
<img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from the MCMC iterations.</p>
</li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/> for the
multinomial logistic model is defined as</p>
<div class="math">
<p><img src="_images/math/f9deb29d45320bfa83051c9f7190f07cc6e133dd.png" alt="\begin{aligned}
\text{FD}_j=\Pr(Y_i=j\mid X_{1})-\Pr(Y_i=j\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">The risk ratio (<tt class="docutils literal"><span class="pre">qi$rr</span></tt>) in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/> is defined as</p>
<div class="math">
<p><img src="_images/math/a15aadb2dfb8f2a23131c8cc50962e73bd61c9d5.png" alt="\begin{aligned}
\text{RR}_j=\Pr(Y_i=j\mid X_{1})\ /\ \Pr(Y_i=j\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>
is</p>
<div class="math">
<p><img src="_images/math/532adccbff80a49ed9d9deff459d4b8832ce421a.png" alt="\begin{aligned}
\frac{1}{n_j}\sum_{i:t_{i}=1}^{n_j}[Y_{i}(t_{i}=1)-E[Y_{i}(t_{i}=0)]],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups, and
<img class="math" src="_images/math/f9ecb2e7f732e9a55b306c2cc3774fe8e638e978.png" alt="n_j"/> is the number of treated observations in category
<img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>) for the treatment group in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>
is</p>
<div class="math">
<p><img src="_images/math/3b1b94dffbb72d02de751897cf05c9dfd18a7dcc.png" alt="\begin{aligned}
\frac{1}{n_j}\sum_{i:t_{i}=1}^{n_j}[Y_{i}(t_{i}=1)-\widehat{Y_{i}(t_{i}=0)}],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups, and
<img class="math" src="_images/math/f9ecb2e7f732e9a55b306c2cc3774fe8e638e978.png" alt="n_j"/> is the number of treated observations in category
<img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id88">
<h3>Output Values<a class="headerlink" href="#id88" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;mlogit.bayes&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
</div>
<div class="section" id="id89">
<h3>See also<a class="headerlink" href="#id89" title="Permalink to this headline">¶</a></h3>
<p>Bayesian logistic regression is part of the MCMCpack library by Andrew
D. Martin and Kevin M. Quinn . The convergence diagnostics are part of
the CODA library by Martyn Plummer, Nicky Best, Kate Cowles, Karen Vines, Deepayan Sarkar, Russell Almond.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-oprobitbayes">
<span id="zoprobitbayes"></span><h2>zelig-oprobitbayes<a class="headerlink" href="#zelig-oprobitbayes" title="Permalink to this headline">¶</a></h2>
<p>Use the ordinal probit regression model if your dependent variables are
ordered and categorical. They may take either integer values or
character strings. The model is estimated using a Gibbs sampler with
data augmentation. For a maximum-likelihood implementation of this
models, see <cite>probit</cite>.</p>
<div class="section" id="id90">
<h3>Syntax<a class="headerlink" href="#id90" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zoprobitbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;oprobit.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id91">
<h3>Additional Inputs<a class="headerlink" href="#id91" title="Permalink to this headline">¶</a></h3>
<p>zelig() accepts the following arguments to monitor the Markov chain:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults
10,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">tune</span></tt>: tuning parameter for the Metropolis-Hasting step. The
default value is <tt class="docutils literal"><span class="pre">NA</span></tt> which corresponds to 0.05 divided by the
number of categories in the response variable.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: defaults to FALSE If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed 12345.</li>
<li><tt class="docutils literal"><span class="pre">beta.start</span></tt>: starting values for the Markov chain, either a scalar
or vector with length equal to the number of estimated coefficients.
The default is <tt class="docutils literal"><span class="pre">NA</span></tt>, which uses the maximum likelihood estimates as
the starting values.</li>
</ul>
<p>Use the following parameters to specify the model’s priors:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: prior mean for the coefficients, either a numeric vector or a
scalar. If a scalar value, that value will be the prior mean for all
the coefficients. The default is 0.</li>
<li><tt class="docutils literal"><span class="pre">B0</span></tt>: prior precision parameter for the coefficients, either a
square matrix (with dimensions equal to the number of coefficients)
or a scalar. If a scalar value, that value times an identity matrix
will be the prior precision parameter. The default is 0 which leads
to an improper prior.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCoprobit)</span></tt> for more
information.</p>
</div>
<div class="section" id="id92">
<h3>Examples<a class="headerlink" href="#id92" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id93">
<h4>Basic Example<a class="headerlink" href="#id93" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>sanction<span class="p">)</span>
</pre></div>
</div>
<p>Estimating ordered probit regression using <tt class="docutils literal"><span class="pre">oprobit.bayes</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>ncost <span class="o">~</span> mil <span class="o">+</span> coop<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;oprobit.bayes&quot;</span><span class="p">,</span>
               data <span class="o">=</span> sanction<span class="p">,</span> verbose <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Warning in model.response(mf, &quot;numeric&quot;): using type = &quot;numeric&quot; with a</span>
<span class="c">## factor response will be ignored</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Ben Goodrich, Ying Lu. 2013.</span>
<span class="c">##   oprobitbayes: Bayesian Probit Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Creating an ordered dependent variable:</p>
<div class="highlight-r"><div class="highlight"><pre>sanction<span class="o">$</span>ncost <span class="o">&lt;-</span> factor<span class="p">(</span>sanction <span class="o">~</span> ncost<span class="p">,</span> ordered <span class="o">=</span> <span class="kc">TRUE</span><span class="p">,</span>
                         levels <span class="o">=</span> c<span class="p">(</span><span class="s">&quot;net gain&quot;</span><span class="p">,</span> <span class="s">&quot;little effect&quot;</span><span class="p">,</span> <span class="s">&quot;modest loss&quot;</span><span class="p">,</span>
                                     <span class="s">&quot;major loss&quot;</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Error in as.vector(x, mode): invalid &#39;mode&#39; argument</span>
</pre></div>
</div>
<p>Checking for convergence before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>heidel.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
raftery.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution given: <tt class="docutils literal"><span class="pre">x.out</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##                     mean         sd        50%       2.5%      97.5%</span>
<span class="c">## little effect 0.44981581 0.05601155 0.44883053 0.34151148 0.56103192</span>
<span class="c">## major loss    0.04473004 0.02101827 0.04176495 0.01271310 0.09412959</span>
<span class="c">## modest loss   0.12341501 0.03950140 0.11984735 0.06005967 0.22834556</span>
<span class="c">## net gain      0.38203914 0.05548348 0.38097907 0.27602445 0.49205957</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">## little effect    major loss   modest loss      net gain</span>
<span class="c">##        0.1868        0.2731        0.5213        0.0188</span>
</pre></div>
</div>
</div>
<div class="section" id="id94">
<h4>Simulating First Differences<a class="headerlink" href="#id94" title="Permalink to this headline">¶</a></h4>
<p>Estimating the first difference (and risk ratio) in the probabilities
of incurring different level of cost when there is no military action
versus military action while all the other variables held at their default values.</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> mil <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> mil <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##                     mean         sd        50%       2.5%      97.5%</span>
<span class="c">## little effect 0.43844669 0.05843957 0.43767439 0.32758262 0.55410854</span>
<span class="c">## major loss    0.04458012 0.02095061 0.04165022 0.01271633 0.09387773</span>
<span class="c">## modest loss   0.12377654 0.03963082 0.12024928 0.05993637 0.22927426</span>
<span class="c">## net gain      0.39319665 0.05795514 0.39196529 0.28336346 0.50825527</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">## little effect    major loss   modest loss      net gain</span>
<span class="c">##        0.1491        0.2382        0.5780        0.0347</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##                    mean         sd        50%       2.5%      97.5%</span>
<span class="c">## little effect 0.5464229 0.16109327 0.54796938 0.23474072 0.84451223</span>
<span class="c">## major loss    0.0407613 0.01998891 0.03763614 0.01085385 0.08828575</span>
<span class="c">## modest loss   0.1075956 0.03975880 0.10421634 0.04208043 0.20132712</span>
<span class="c">## net gain      0.3052203 0.14485132 0.29018143 0.07362204 0.62315150</span>
<span class="c">## pv</span>
<span class="c">## qi</span>
<span class="c">## little effect    major loss   modest loss      net gain</span>
<span class="c">##        0.6116        0.0963        0.1862        0.1059</span>
<span class="c">## fd</span>
<span class="c">##                       mean         sd          50%        2.5%       97.5%</span>
<span class="c">## little effect  0.107976200 0.17020693  0.111082214 -0.22740084 0.426282862</span>
<span class="c">## major loss    -0.003818825 0.00665593 -0.001418880 -0.02253975 0.002180789</span>
<span class="c">## modest loss   -0.016180976 0.02172754 -0.008327805 -0.07580491 0.005015329</span>
<span class="c">## net gain      -0.087976398 0.15275950 -0.102164237 -0.34480882 0.241227653</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id95">
<h3>Model<a class="headerlink" href="#id95" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> be the ordered categorical dependent variable for
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> which takes an integer value
<img class="math" src="_images/math/de0b4bef7a011d44e3288965ca51ff93922ab47b.png" alt="j=1, \ldots, J"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is described by an unobserved continuous
variable, <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/>,</p>
<div class="math">
<p><img src="_images/math/eb2c30b74e11eddc163bcdb9c33adac322741da1.png" alt="\begin{aligned}
Y_{i}^*  \sim \textrm{Normal}(\mu_i, 1).\end{aligned}"/></p>
</div><p>Instead of <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/>, we observe categorical variable
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/>,</p>
<div class="math">
<p><img src="_images/math/6250ab78ef29d90efa8637f41b9d9a992aafcde9.png" alt="\begin{aligned}
Y_i = j \quad \textrm{ if } \tau_{j-1} \le Y_i^* \le \tau_j \textrm{
for } j=1,\ldots, J.\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/e7b7b95439d7fb629fa6216dbc718f1c0ab66e28.png" alt="\tau_j"/> for <img class="math" src="_images/math/c8676bf8368b0ab3402a9701d2e91dd5b2a206c0.png" alt="j=0,\ldots, J"/> are the threshold
parameters with the following constraints, <img class="math" src="_images/math/9d6e77a79e113678fa61237c036ca8c526eb1dd6.png" alt="\tau_l &lt; \tau_m"/>
for <img class="math" src="_images/math/72720f025347637106f2eb28797fe1615dd659ee.png" alt="l &lt; m"/>, and <img class="math" src="_images/math/a4db67ac47b297539e14d30e99a8483177a735c9.png" alt="\tau_0=-\infty, \tau_J=\infty"/>.</p>
<p>The probability of observing <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> equal to category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>
is,</p>
<div class="math">
<p><img src="_images/math/512c5c87d774360941b15fffe21cc28dfe755e24.png" alt="\begin{aligned}
\Pr(Y_i=j) &amp;=&amp; \Phi(\tau_j \mid \mu_i)-\Phi(\tau_{j-1} \mid \mu_i)
\textrm{ for } j=1,\ldots, J\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/7fd8c23c16a9f815d734946e572c67d59b81a799.png" alt="\Phi(\cdot \mid \mu_i)"/> is the cumulative distribution
function of the Normal distribution with mean <img class="math" src="_images/math/fe45a709c71e4160e1a09e3b4cc03d2d36d15644.png" alt="\mu_i"/> and
variance 1.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/289d3503884e94b40230aaa0943b7e4c9e10a13f.png" alt="\begin{aligned}
\mu_{i}= x_i \beta,\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
<li><p class="first">The <em>prior</em> for <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is given by</p>
<div class="math">
<p><img src="_images/math/00da19989430e39ccd6410803f142d7652fbcf02.png" alt="\begin{aligned}
\beta \sim \textrm{Normal}_k\left(  b_{0},B_{0}^{-1}\right)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/9d80db7ae48caf02365282c5dff832fcbe0a74d5.png" alt="b_{0}"/> is the vector of means for the <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>
explanatory variables and <img class="math" src="_images/math/6a119b5ae9ee269e28bcaa4d39e47a3d3de4f5e5.png" alt="B_{0}"/> is the <img class="math" src="_images/math/3908ba50237c94909083edd0295cd1c9345b0a8c.png" alt="k \times k"/>
precision matrix (the inverse of a variance-covariance matrix).</p>
</li>
</ul>
</div>
<div class="section" id="id96">
<h3>Quantities of Interest<a class="headerlink" href="#id96" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the ordered probit model are the
predicted probability of belonging to each category:</p>
<div class="math">
<p><img src="_images/math/432a9272c13e9a28264887d70b45726b622d46b5.png" alt="\begin{aligned}
\Pr(Y_i=j)= \Phi(\tau_j \mid x_i \beta)-\Phi(\tau_{j-1} \mid x_i \beta),\end{aligned}"/></p>
</div><p>given the posterior draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and threshold parameters
<img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> from the MCMC iterations.</p>
</li>
<li><p class="first">The predicted values (<tt class="docutils literal"><span class="pre">qi$pr</span></tt>) are the observed values of
<img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> given the observation scheme and the posterior draws of
<img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and cut points <img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/> from the MCMC iterations.</p>
</li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/> for the
ordered probit model is defined as</p>
<div class="math">
<p><img src="_images/math/f9deb29d45320bfa83051c9f7190f07cc6e133dd.png" alt="\begin{aligned}
\text{FD}_j=\Pr(Y_i=j\mid X_{1})-\Pr(Y_i=j\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">The risk ratio (<tt class="docutils literal"><span class="pre">qi$rr</span></tt>) in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/> is defined as</p>
<div class="math">
<p><img src="_images/math/a15aadb2dfb8f2a23131c8cc50962e73bd61c9d5.png" alt="\begin{aligned}
\text{RR}_j=\Pr(Y_i=j\mid X_{1})\ /\ \Pr(Y_i=j\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>
is</p>
<div class="math">
<p><img src="_images/math/85fbdfa785a78d4a9e5525a49939ddda2b50746e.png" alt="\begin{aligned}
\frac{1}{n_j}\sum_{i:t_{i}=1}^{n_j} \{
Y_{i}(t_{i}=1)-E[Y_{i}(t_{i}=0)] \},\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups, and
<img class="math" src="_images/math/f9ecb2e7f732e9a55b306c2cc3774fe8e638e978.png" alt="n_j"/> is the number of observations in the treatment group that
belong to category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>) for the treatment group in category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>
is</p>
<div class="math">
<p><img src="_images/math/3b1b94dffbb72d02de751897cf05c9dfd18a7dcc.png" alt="\begin{aligned}
\frac{1}{n_j}\sum_{i:t_{i}=1}^{n_j}[Y_{i}(t_{i}=1)-\widehat{Y_{i}(t_{i}=0)}],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups, and
<img class="math" src="_images/math/f9ecb2e7f732e9a55b306c2cc3774fe8e638e978.png" alt="n_j"/> is the number of observations in the treatment group that
belong to category <img class="math" src="_images/math/d32c78b759903e3f4bd4fd2ce0b86358f7500c5d.png" alt="j"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id97">
<h3>Output Values<a class="headerlink" href="#id97" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-python"><div class="highlight"><pre>z.out &lt;- zelig(y ~ x, model = &quot;oprobit.bayes&quot;, data)
</pre></div>
</div>
<p>then you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
<ul class="simple">
<li>From the <tt class="docutils literal"><span class="pre">zelig()</span></tt> output object <tt class="docutils literal"><span class="pre">z.out</span></tt>, you may extract:<ul>
<li><tt class="docutils literal"><span class="pre">coefficients</span></tt>: draws from the posterior distributions of the
estimated coefficients <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and threshold parameters
<img class="math" src="_images/math/d9e99c35d3632bb156b7aca0e93cd78a148a10e8.png" alt="\tau"/>. Note, element <img class="math" src="_images/math/7535961332c7b400bb1f935a141711a30c82cd87.png" alt="\tau_1"/> is normalized to 0 and
is not returned in the <tt class="docutils literal"><span class="pre">coefficients</span></tt> object.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: the random seed used in the model.</li>
</ul>
</li>
<li>From the <tt class="docutils literal"><span class="pre">sim()</span></tt> output object <tt class="docutils literal"><span class="pre">s.out</span></tt>:<ul>
<li><tt class="docutils literal"><span class="pre">qi$ev</span></tt>: the simulated expected values (probabilities) of each
of the <img class="math" src="_images/math/372c2a0702e1e5866656215408b37a2ddb8d2d45.png" alt="J"/> categories for the specified values of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$pr</span></tt>: the simulated predicted values (observed values) for
the specified values of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$fd</span></tt>: the simulated first difference in the expected values
of each of the <img class="math" src="_images/math/372c2a0702e1e5866656215408b37a2ddb8d2d45.png" alt="J"/> categories for the values specified in
<tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$rr</span></tt>: the simulated risk ratio for the expected values of
each of the <img class="math" src="_images/math/372c2a0702e1e5866656215408b37a2ddb8d2d45.png" alt="J"/> categories simulated from <tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>: the simulated average expected treatment effect for
the treated from conditional prediction models.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>: the simulated average predicted treatment effect
for the treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id98">
<h3>See also<a class="headerlink" href="#id98" title="Permalink to this headline">¶</a></h3>
<p>Bayesian ordinal probit regression is part of the MCMCpack library by
Andrew D. Martin and Kevin M. Quinn . The convergence diagnostics are
part of the CODA library by Martyn Plummer, Nicky Best, Kate Cowles, and
Karen Vines.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-poissonbayes">
<span id="zpoissonbayes"></span><h2>zelig-poissonbayes<a class="headerlink" href="#zelig-poissonbayes" title="Permalink to this headline">¶</a></h2>
<p>Use the Poisson regression model if the observations of your dependent
variable represents the number of independent events that occur during
a fixed period of time. The model is fit using a random walk
Metropolis algorithm.  For a maximum-likelihood estimation of this
model see <cite>poisson</cite>.</p>
<div class="section" id="id99">
<h3>Syntax<a class="headerlink" href="#id99" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zpoissonbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id100">
<h3>Additional Inputs<a class="headerlink" href="#id100" title="Permalink to this headline">¶</a></h3>
<p>Use the following argument to monitor the Markov chain:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults to
10,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">tune</span></tt>: Metropolis tuning parameter, either a positive scalar or a
vector of length <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>, where <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> is the number of
coefficients. The tuning parameter should be set such that the
acceptance rate of the Metropolis algorithm is satisfactory
(typically between 0.20 and 0.5). The default value is 1.1.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: default to FALSE. If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed of 12345.</li>
<li><tt class="docutils literal"><span class="pre">beta.start</span></tt>: starting values for the Markov chain, either a scalar
or vector with length equal to the number of estimated coefficients.
The default is <tt class="docutils literal"><span class="pre">NA</span></tt>, such that the maximum likelihood estimates are
used as the starting values.</li>
</ul>
<p>Use the following parameters to specify the model’s priors:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: prior mean for the coefficients, either a numeric vector or a
scalar. If a scalar, that value will be the prior mean for all the
coefficients. The default is 0.</li>
<li><tt class="docutils literal"><span class="pre">B0</span></tt>: prior precision parameter for the coefficients, either a
square matrix (with the dimensions equal to the number of the
coefficients) or a scalar. If a scalar, that value times an identity
matrix will be the prior precision parameter. The default is 0, which
leads to an improper prior.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCpoisson)</span></tt> for more
information.</p>
</div>
<div class="section" id="id101">
<h3>Examples<a class="headerlink" href="#id101" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id102">
<h4>Basic Example<a class="headerlink" href="#id102" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>sanction<span class="p">)</span>
</pre></div>
</div>
<p>Estimating the Poisson regression using <tt class="docutils literal"><span class="pre">poisson.bayes</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>num <span class="o">~</span> target <span class="o">+</span> coop<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.bayes&quot;</span><span class="p">,</span>
               data <span class="o">=</span> sanction<span class="p">,</span> verbose <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Ben Goodrich, Ying Lu. 2013.</span>
<span class="c">##   poissonbayes: Bayesian Poisson Regression</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Checking convergence diagnostics before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>geweke.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>heidel.diag<span class="p">(</span>z.out`\ coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>raftery.diag<span class="p">(</span>z.out..coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution given <tt class="docutils literal"><span class="pre">x.out</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id103">
<h4>Simulating First Differences<a class="headerlink" href="#id103" title="Permalink to this headline">¶</a></h4>
<p>Estimating the first difference in the number of countries imposing
sanctions when the number of targets is set to be its maximum versus its minimum :</p>
<div class="highlight-r"><div class="highlight"><pre>x.max <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> target <span class="o">=</span> max<span class="p">(</span>sanction<span class="o">$</span>target<span class="p">))</span>
x.min <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> target <span class="o">=</span> min<span class="p">(</span>sanction<span class="o">$</span>target<span class="p">))</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.max<span class="p">,</span> x1 <span class="o">=</span> x.min<span class="p">)</span>
summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 3.191614 0.2936585 3.183013 2.642371 3.803733</span>
<span class="c">## pv</span>
<span class="c">##        mean       sd 50% 2.5% 97.5%</span>
<span class="c">## [1,] 3.2099 1.823177   3    0     7</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 3.306252 0.3059862 3.300095 2.729466 3.944022</span>
<span class="c">## pv</span>
<span class="c">##        mean       sd 50% 2.5% 97.5%</span>
<span class="c">## [1,] 3.3174 1.853322   3    0     7</span>
<span class="c">## fd</span>
<span class="c">##           mean        sd       50%       2.5%     97.5%</span>
<span class="c">## [1,] 0.1146376 0.3671544 0.1265036 -0.6072035 0.8342282</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id104">
<h3>Model<a class="headerlink" href="#id104" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> be the number of independent events that occur during
a fixed time period.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/ea903a844966d327c5804197b944c5b57f6f2e4b.png" alt="\begin{aligned}
Y_{i}  &amp;  \sim &amp; \textrm{Poisson}(\lambda_i)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/> is the mean and variance parameter.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/f17f0771bc7a8cd4c9fe0c301f3cf813fad32a13.png" alt="\begin{aligned}
\lambda_{i}= \exp(x_{i} \beta)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
<li><p class="first">The <em>prior</em> for <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is given by</p>
<div class="math">
<p><img src="_images/math/b786073674269995bb95620d4df7511b8cd7e452.png" alt="\begin{aligned}
\beta \sim \textrm{Normal}_k \left(  b_{0},B_{0}^{-1}\right)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/9d80db7ae48caf02365282c5dff832fcbe0a74d5.png" alt="b_{0}"/> is the vector of means for the <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>
explanatory variables and <img class="math" src="_images/math/6a119b5ae9ee269e28bcaa4d39e47a3d3de4f5e5.png" alt="B_{0}"/> is the <img class="math" src="_images/math/3908ba50237c94909083edd0295cd1c9345b0a8c.png" alt="k \times k"/>
precision matrix (the inverse of a variance-covariance matrix).</p>
</li>
</ul>
</div>
<div class="section" id="id105">
<h3>Quantities of Interest<a class="headerlink" href="#id105" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the Poisson model are calculated
as following:</p>
<div class="math">
<p><img src="_images/math/2668f5aa2c8db1b92dc68b5950271ab71880e0f4.png" alt="\begin{aligned}
E(Y\mid X) = \lambda_i = \exp(x_i \beta),\end{aligned}"/></p>
</div><p>given the posterior draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> based on the MCMC
iterations.</p>
</li>
<li><p class="first">The predicted values (<tt class="docutils literal"><span class="pre">qi$pr</span></tt>) are draws from the Poisson
distribution with parameter <img class="math" src="_images/math/3b765ea6939f1eba2541e91f870cb3c078aa29be.png" alt="\lambda_i"/>.</p>
</li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) for the Poisson model is defined as</p>
<div class="math">
<p><img src="_images/math/e32bcefcd3232180d0737b016162063cf960ee3d.png" alt="\begin{aligned}
\text{FD}=E(Y\mid X_{1})-E(Y\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/155601c69591d66ffefcd58f0d7909591a67f704.png" alt="\begin{aligned}
\frac{1}{\sum_{i=1}^n t_{i}}\sum_{i:t_{i}=1}\{Y_{i}(t_{i}=1)-E[Y_{i}(t_{i}=0)]\},\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/50406baf96b768004de658ce511942b4ae34b395.png" alt="\begin{aligned}
\frac{1}{\sum_{i=1}^n t_{i}}\sum_{i:t_{i}=1}[Y_{i}(t_{i}=1)-\widehat{Y_{i}(t_{i}=0)}],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
</ul>
</div>
<div class="section" id="id106">
<h3>Output Values<a class="headerlink" href="#id106" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.bayes&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
<ul class="simple">
<li>From the <tt class="docutils literal"><span class="pre">zelig()</span></tt> output object <tt class="docutils literal"><span class="pre">z.out</span></tt>, you may extract:<ul>
<li><tt class="docutils literal"><span class="pre">coefficients</span></tt>: draws from the posterior distributions of the
estimated parameters.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: the random seed used in the model.</li>
</ul>
</li>
<li>From the <tt class="docutils literal"><span class="pre">sim()</span></tt> output object <tt class="docutils literal"><span class="pre">s.out</span></tt>:<ul>
<li><tt class="docutils literal"><span class="pre">qi$ev</span></tt>: the simulated expected values for the specified values
of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$pr</span></tt>: the simulated predicted values for the specified values
of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$fd</span></tt>: the simulated first difference in the expected values
for the values specified in <tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>: the simulated average expected treatment effect for
the treated from conditional prediction models.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>: the simulated average predicted treatment effect
for the treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id107">
<h3>See also<a class="headerlink" href="#id107" title="Permalink to this headline">¶</a></h3>
<p>Bayesian poisson regression is part of the MCMCpack library by Andrew D.
Martin and Kevin M. Quinn . The convergence diagnostics are part of the
CODA library by Martyn Plummer, Nicky Best, Kate Cowles, and Karen Vines.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-probitbayes">
<span id="zprobitbayes"></span><h2>zelig-probitbayes<a class="headerlink" href="#zelig-probitbayes" title="Permalink to this headline">¶</a></h2>
<p>Use the probit regression model for model binary dependent variables
specified as a function of a set of explanatory variables. The model is
estimated using a Gibbs sampler. For other models suitable for binary
response variables, see Bayesian logistic regression, maximum
likelihood logit regression, and maximum likelihood probit regression.</p>
<div class="section" id="id108">
<h3>Syntax<a class="headerlink" href="#id108" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zprobitbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id109">
<h3>Additional Inputs<a class="headerlink" href="#id109" title="Permalink to this headline">¶</a></h3>
<p>Using the following arguments to monitor the Markov chains:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults to
10,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: defaults to FALSE. If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed of 12345.</li>
<li><tt class="docutils literal"><span class="pre">beta.start</span></tt>: starting values for the Markov chain, either a scalar
or vector with length equal to the number of estimated coefficients.
The default is <tt class="docutils literal"><span class="pre">NA</span></tt>, such that the maximum likelihood estimates are
used as the starting values.</li>
</ul>
<p>Use the following parameters to specify the model’s priors:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: prior mean for the coefficients, either a numeric vector or a
scalar. If a scalar value, that value will be the prior mean for all
the coefficients. The default is 0.</li>
<li><tt class="docutils literal"><span class="pre">B0</span></tt>: prior precision parameter for the coefficients, either a
square matrix (with the dimensions equal to the number of the
coefficients) or a scalar. If a scalar value, that value times an
identity matrix will be the prior precision parameter. The default is
0, which leads to an improper prior.</li>
</ul>
<p>Use the following arguments to specify optional output for the model:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">bayes.resid</span></tt>: defaults to FALSE. If TRUE, the latent Bayesian
residuals for all observations are returned. Alternatively, users can
specify a vector of observations for which the latent residuals
should be returned.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCprobit)</span></tt> for more
information.</p>
</div>
<div class="section" id="id110">
<h3>Examples<a class="headerlink" href="#id110" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id111">
<h4>Basic Example<a class="headerlink" href="#id111" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>turnout<span class="p">)</span>
</pre></div>
</div>
<p>Estimating the probit regression using <tt class="docutils literal"><span class="pre">probit.bayes</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> race <span class="o">+</span> educate<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.bayes&quot;</span><span class="p">,</span>
               data <span class="o">=</span> turnout<span class="p">,</span> verbose <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Ben Goodrich, Ying Lu. 2013.</span>
<span class="c">##   probitbayes: Bayesian Probit Regression for Dichotomous Dependent Variables</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Checking for convergence before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>geweke.diag<span class="p">(</span>z.out coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>heidel.diag<span class="p">(</span>z.out coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>raftery.diag<span class="p">(</span>z.out coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution given: <tt class="docutils literal"><span class="pre">x.out</span></tt></p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id112">
<h4>Simulating First Differences<a class="headerlink" href="#id112" title="Permalink to this headline">¶</a></h4>
<p>Estimating the first difference (and risk ratio) in individual’s
probability of voting when education is set to be low (25th
percentile) versus high (75th percentile) while all the other variables are held at their default values:</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.75</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.25</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id113">
<h3>Model<a class="headerlink" href="#id113" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> be the binary dependent variable for observation
<img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> which takes the value of either 0 or 1.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/b8dda738cea0ccca73ff135a88c134d9e9288e53.png" alt="\begin{aligned}
Y_{i}  &amp;  \sim &amp; \textrm{Bernoulli}(\pi_{i})\\
&amp;  = &amp; \pi_{i}^{Y_{i}}(1-\pi_{i})^{1-Y_{i}},\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/119dd61e8bc116cc69c29362754356787f09e5be.png" alt="\pi_{i}=\Pr(Y_{i}=1)"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/05e662995b85cc81eb1fb362e134a98b0be57fac.png" alt="\begin{aligned}
\pi_{i}= \Phi(x_i \beta),\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/262565daee431ce18fed8cbe6ff35db082bb6a9b.png" alt="\Phi(\cdot)"/> is the cumulative density function of the
standard Normal distribution with mean 0 and variance 1,
<img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables for
observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>, and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
<li><p class="first">The <em>prior</em> for <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is given by</p>
<div class="math">
<p><img src="_images/math/7607fa8c02af8d4dfc7f426eb5732a4bb9069b42.png" alt="\begin{aligned}
\beta \sim \textrm{Normal}_k \left(  b_{0}, B_{0}^{-1} \right)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/9d80db7ae48caf02365282c5dff832fcbe0a74d5.png" alt="b_{0}"/> is the vector of means for the <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>
explanatory variables and <img class="math" src="_images/math/6a119b5ae9ee269e28bcaa4d39e47a3d3de4f5e5.png" alt="B_{0}"/> is the <img class="math" src="_images/math/3908ba50237c94909083edd0295cd1c9345b0a8c.png" alt="k \times k"/>
precision matrix (the inverse of a variance-covariance matrix).</p>
</li>
</ul>
</div>
<div class="section" id="id114">
<h3>Quantities of Interest<a class="headerlink" href="#id114" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the probit model are the
predicted probability of a success:</p>
<div class="math">
<p><img src="_images/math/251a0c2a46caf0d86aab6add5e2c4f5960c88b2a.png" alt="\begin{aligned}
E(Y \mid X) = \pi_{i}= \Phi(x_i \beta),\end{aligned}"/></p>
</div><p>given the posterior draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from the MCMC iterations.</p>
</li>
<li><p class="first">The predicted values (<tt class="docutils literal"><span class="pre">qi$pr</span></tt>) are draws from the Bernoulli
distribution with mean equal to the simulated expected value
<img class="math" src="_images/math/56d13901556c9d691c7d1893b7b9329d60794bd6.png" alt="\pi_{i}"/>.</p>
</li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) for the probit model is defined as</p>
<div class="math">
<p><img src="_images/math/b8f51d813d7d58f8d1db9ff6d4999f95081705c9.png" alt="\begin{aligned}
\text{FD}=\Pr(Y=1\mid X_{1})-\Pr(Y=1\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">The risk ratio (<tt class="docutils literal"><span class="pre">qi$rr</span></tt>)is defined as</p>
<div class="math">
<p><img src="_images/math/d2bdd6f85746d5731bd9679e294d3e22b805d218.png" alt="\begin{aligned}
\text{RR}=\Pr(Y=1\mid X_{1})\ /\ \Pr(Y=1\mid X).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/72b57ba1ea5551a32c3fa473511823ae37508662.png" alt="\begin{aligned}
\frac{1}{\sum t_{i}}\sum_{i:t_{i}=1}[Y_{i}(t_{i}=1)-E[Y_{i}(t_{i}=0)]],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
<li><p class="first">In conditional prediction models, the average predicted treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/9009cc654392b7154f901adf17e5170c105149a0.png" alt="\begin{aligned}
\frac{1}{\sum t_{i}}\sum_{i:t_{i}=1}[Y_{i}(t_{i}=1)-\widehat{Y_{i}(t_{i}=0)}],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
</ul>
</div>
<div class="section" id="id115">
<h3>Output Values<a class="headerlink" href="#id115" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.bayes&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
<ul class="simple">
<li>From the <tt class="docutils literal"><span class="pre">zelig()</span></tt> output object <tt class="docutils literal"><span class="pre">z.out</span></tt>, you may extract:<ul>
<li><tt class="docutils literal"><span class="pre">coefficients</span></tt>: draws from the posterior distributions of the
estimated parameters.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
<li><tt class="docutils literal"><span class="pre">bayes.residuals</span></tt>: When <tt class="docutils literal"><span class="pre">bayes.residual</span></tt> is <tt class="docutils literal"><span class="pre">TRUE</span></tt> or a set
of observation numbers is given, this object contains the
posterior draws of the latent Bayesian residuals of all the
observations or the observations specified by the user.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: the random seed used in the model.</li>
</ul>
</li>
<li>From the <tt class="docutils literal"><span class="pre">sim()</span></tt> output object <tt class="docutils literal"><span class="pre">s.out</span></tt>:<ul>
<li><tt class="docutils literal"><span class="pre">qi$ev</span></tt>: the simulated expected values (probabilities) for the
specified values of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$pr</span></tt>: the simulated predicted values for the specified values
of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$fd</span></tt>: the simulated first difference in the expected values
for the values specified in <tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$rr</span></tt>: the simulated risk ratio for the expected values
simulated from <tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>: the simulated average expected treatment effect for
the treated from conditional prediction models.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.pr</span></tt>: the simulated average predicted treatment effect
for the treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id116">
<h3>See also<a class="headerlink" href="#id116" title="Permalink to this headline">¶</a></h3>
<p>Bayesian probit regression is part of the MCMCpack library by Andrew D.
Martin and Kevin M. Quinn . The convergence diagnostics are part of the
CODA library by Martyn Plummer, Nicky Best, Kate Cowles, and Karen Vines.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-tobitbayes">
<span id="ztobitbayes"></span><h2>zelig-tobitbayes<a class="headerlink" href="#zelig-tobitbayes" title="Permalink to this headline">¶</a></h2>
<p>Bayesian tobit regression estimates a linear regression model with a
censored dependent variable using a Gibbs sampler. The dependent
variable may be censored from below and/or from above. For other linear
regression models with fully observed dependent variables, see Bayesian
regression, maximum likelihood normal regression, or least squares.</p>
<div class="section" id="id117">
<h3>Syntax<a class="headerlink" href="#id117" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zprobitbayes<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">((</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> below <span class="o">=</span> <span class="m">0</span><span class="p">,</span> above <span class="o">=</span> <span class="kc">Inf</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> below <span class="o">=</span> <span class="m">0</span><span class="p">,</span> above <span class="o">=</span> <span class="kc">Inf</span><span class="p">,</span>
               model <span class="o">=</span> <span class="s">&quot;tobit.bayes&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id118">
<h3>Inputs<a class="headerlink" href="#id118" title="Permalink to this headline">¶</a></h3>
<p>zelig() accepts the following arguments to specify how the dependent
variable is censored.</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">below</span></tt>: point at which the dependent variable is censored from
below. If the dependent variable is only censored from above, set
<tt class="docutils literal"><span class="pre">below</span> <span class="pre">=</span> <span class="pre">-Inf</span></tt>. The default value is 0.</li>
<li><tt class="docutils literal"><span class="pre">above</span></tt>: point at which the dependent variable is censored from
above. If the dependent variable is only censored from below, set
<tt class="docutils literal"><span class="pre">above</span> <span class="pre">=</span> <span class="pre">Inf</span></tt>. The default value is <tt class="docutils literal"><span class="pre">Inf</span></tt>.</li>
</ul>
</div>
<div class="section" id="id119">
<h3>Additional Inputs<a class="headerlink" href="#id119" title="Permalink to this headline">¶</a></h3>
<p>Use the following arguments to monitor the convergence of the Markov
chain:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">burnin</span></tt>: number of the initial MCMC iterations to be discarded
(defaults to 1,000).</li>
<li><tt class="docutils literal"><span class="pre">mcmc</span></tt>: number of the MCMC iterations after burnin (defaults to
10,000).</li>
<li><tt class="docutils literal"><span class="pre">thin</span></tt>: thinning interval for the Markov chain. Only every
<tt class="docutils literal"><span class="pre">thin</span></tt>-th draw from the Markov chain is kept. The value of <tt class="docutils literal"><span class="pre">mcmc</span></tt>
must be divisible by this value. The default value is 1.</li>
<li><tt class="docutils literal"><span class="pre">verbose</span></tt>: defaults to FALSE. If <tt class="docutils literal"><span class="pre">TRUE</span></tt>, the progress of the
sampler (every <img class="math" src="_images/math/edce6cd55e38e73a0341fc79fb72bb04c90eeca4.png" alt="10\%"/>) is printed to the screen.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: seed for the random number generator. The default is <tt class="docutils literal"><span class="pre">NA</span></tt>
which corresponds to a random seed of 12345.</li>
<li><tt class="docutils literal"><span class="pre">beta.start</span></tt>: starting values for the Markov chain, either a scalar
or vector with length equal to the number of estimated coefficients.
The default is <tt class="docutils literal"><span class="pre">NA</span></tt>, such that the least squares estimates are used
as the starting values.</li>
</ul>
<p>Use the following parameters to specify the model’s priors:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">b0</span></tt>: prior mean for the coefficients, either a numeric vector or a
scalar. If a scalar, that value will be the prior mean for all
coefficients. The default is 0.</li>
<li><tt class="docutils literal"><span class="pre">B0</span></tt>: prior precision parameter for the coefficients, either a
square matrix (with the dimensions equal to the number of the
coefficients) or a scalar. If a scalar, that value times an identity
matrix will be the prior precision parameter. The default is 0, which
leads to an improper prior.</li>
<li><tt class="docutils literal"><span class="pre">c0</span></tt>: <tt class="docutils literal"><span class="pre">c0/2</span></tt> is the shape parameter for the Inverse Gamma prior
on the variance of the disturbance terms.</li>
<li><tt class="docutils literal"><span class="pre">d0</span></tt>: <tt class="docutils literal"><span class="pre">d0/2</span></tt> is the scale parameter for the Inverse Gamma prior
on the variance of the disturbance terms.</li>
</ul>
<p>Zelig users may wish to refer to <tt class="docutils literal"><span class="pre">help(MCMCtobit)</span></tt> for more
information.</p>
</div>
<div class="section" id="id120">
<h3>Examples<a class="headerlink" href="#id120" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id121">
<h4>Basic Example<a class="headerlink" href="#id121" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>tobin<span class="p">)</span>
</pre></div>
</div>
<p>Estimating linear regression using <tt class="docutils literal"><span class="pre">tobit.bayes</span></tt>:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>durable <span class="o">~</span> age <span class="o">+</span> quant<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;tobit.bayes&quot;</span><span class="p">,</span>
               data <span class="o">=</span> tobin<span class="p">,</span> verbose <span class="o">=</span> <span class="kc">FALSE</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Ben Goodrich, Ying Lu. 2013.</span>
<span class="c">##   tobitbayes: Bayesian Tobit Regression for a Censored Dependent Variable</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Checking for convergence before summarizing the estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>geweke.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>heidel.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>raftery.diag<span class="p">(</span>z.out<span class="o">$</span>coefficients<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their sample averages:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest from the posterior distribution given <tt class="docutils literal"><span class="pre">x.out</span></tt>.</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
</div>
<div class="section" id="id122">
<h4>Simulating First Differences<a class="headerlink" href="#id122" title="Permalink to this headline">¶</a></h4>
<p>Set explanatory variables to their default(mean/mode) values, with
high (80th percentile) and low (20th percentile) liquidity ratio (<tt class="docutils literal"><span class="pre">quant</span></tt>):</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> quant <span class="o">=</span> quantile<span class="p">(</span>tobin<span class="o">$</span>quant<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.8</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> quant <span class="o">=</span> quantile<span class="p">(</span>tobin<span class="o">$</span>quant<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.2</span><span class="p">))</span>
</pre></div>
</div>
<p>Estimating the first difference for the effect of high versus low liquidity ratio on duration( <tt class="docutils literal"><span class="pre">durable</span></tt>):</p>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id123">
<h3>Model<a class="headerlink" href="#id123" title="Permalink to this headline">¶</a></h3>
<p>Let <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> be the dependent variable which is not directly
observed. Instead, we observe <img class="math" src="_images/math/bc99eef60e4cfed714571a15a3192837beb4bd4c.png" alt="Y_i"/> which is defined as following:</p>
<div class="math">
<p><img src="_images/math/6177e9ea11a1ce79d4a22769db58be80cbcefb65.png" alt="Y_i = \left\{
\begin{array}{lcl}
Y_i^*  &amp;\textrm{if} &amp; c_1&lt;Y_i^*&lt;c_2 \\
c_1    &amp;\textrm{if} &amp; c_1 \ge Y_i^* \\
c_2    &amp;\textrm{if} &amp; c_2 \le Y_i^*
\end{array}\right."/></p>
</div><p>where <img class="math" src="_images/math/3b3656365ff2748ee55de15a942b63693cb47dae.png" alt="c_1"/> is the lower bound below which <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/> is
censored, and <img class="math" src="_images/math/5081ba4156b5c83e0ad11d0dbf1ddd83fa988929.png" alt="c_2"/> is the upper bound above which <img class="math" src="_images/math/d618fcc344538f93b4596930823c957af4a11865.png" alt="Y_i^*"/>
is censored.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/bba99ec8514413a99fe9eeac87dc26cb92200c20.png" alt="\begin{aligned}
\epsilon_{i}  &amp;  \sim &amp; \textrm{Normal}(0, \sigma^2)\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/a2e934f8fc24c921e07ec0ba27d0c89922ad8fbc.png" alt="\epsilon_{i}=Y^*_i-\mu_i"/>.</p>
</li>
<li><p class="first">The <em>systematic component</em> is given by</p>
<div class="math">
<p><img src="_images/math/30262eeff311dcffec8a37eeee77beb1e7de5ba8.png" alt="\begin{aligned}
\mu_{i}= x_{i} \beta,\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/c223a5391d7bfd7e7c68274c104fc7418944376e.png" alt="x_{i}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for observation <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of
coefficients.</p>
</li>
<li><p class="first">The <em>semi-conjugate priors</em> for <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and <img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/>
are given by</p>
<div class="math">
<p><img src="_images/math/e490be64f5b3ec90fae48cb38d9e07d87810802f.png" alt="\begin{aligned}
\beta &amp; \sim &amp; \textrm{Normal}_k \left( b_{0},B_{0}^{-1}\right) \\
\sigma^{2} &amp; \sim &amp; \textrm{InverseGamma} \left( \frac{c_0}{2}, \frac{d_0}{2}
\right) \end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/9d80db7ae48caf02365282c5dff832fcbe0a74d5.png" alt="b_{0}"/> is the vector of means for the <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/>
explanatory variables, <img class="math" src="_images/math/6a119b5ae9ee269e28bcaa4d39e47a3d3de4f5e5.png" alt="B_{0}"/> is the <img class="math" src="_images/math/f7ad0969df4d6e13c90ede9c19428da3d9d72b51.png" alt="k\times k"/>
precision matrix (the inverse of a variance-covariance matrix), and
<img class="math" src="_images/math/b3e82fc83be5a12833b456068201e39c2c8a32aa.png" alt="c_0/2"/> and <img class="math" src="_images/math/5e09d331f43f4162daeb55e031911436ab1f687a.png" alt="d_0/2"/> are the shape and scale parameters
for <img class="math" src="_images/math/c257a14ece85ea075f075f381c2729c3d3da0100.png" alt="\sigma^{2}"/>. Note that <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> and <img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/>
are assumed <em>a priori</em> independent.</p>
</li>
</ul>
</div>
<div class="section" id="id124">
<h3>Quantities of Interest<a class="headerlink" href="#id124" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">The expected values (<tt class="docutils literal"><span class="pre">qi$ev</span></tt>) for the tobit regression model is
calculated as following. Let</p>
<div class="math">
<p><img src="_images/math/fb8797d633023f188277ec0f3d76220a12ebcc64.png" alt="\begin{aligned}
\Phi_1 &amp;=&amp; \Phi\left(\frac{(c_1 - x \beta)}{\sigma}\right) \\
\Phi_2 &amp;=&amp; \Phi\left(\frac{(c_2 - x \beta)}{\sigma}\right) \\
\phi_1 &amp;=&amp; \phi\left(\frac{(c_1 - x \beta)}{\sigma}\right) \\
\phi_2 &amp;=&amp; \phi\left(\frac{(c_2 - x \beta)}{\sigma}\right) \end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/262565daee431ce18fed8cbe6ff35db082bb6a9b.png" alt="\Phi(\cdot)"/> is the (cumulative) Normal density function
and <img class="math" src="_images/math/c1174cd218c6279eac6e39968dd22da553ac30a6.png" alt="\phi(\cdot)"/> is the Normal probability density function of
the standard normal distribution. Then the expected values are</p>
<div class="math">
<p><img src="_images/math/b4248a90240916af1a721025820ad7c1eba9b532.png" alt="\begin{aligned}
E(Y|x) &amp;=&amp; P(Y^* \le c_1|x) c_1+P(c_1&lt;Y^*&lt;c_2|x) E(Y^* \mid c_1&lt;Y^*&lt;c_2, x)+P(Y^* \ge c_2) c_2 \\
   &amp;=&amp; \Phi_{1}c_1 + x \beta(\Phi_{2}-\Phi_{1}) + \sigma (\phi_1 -\phi_2) + (1-\Phi_2) c_2,\end{aligned}"/></p>
</div></li>
<li><p class="first">The first difference (<tt class="docutils literal"><span class="pre">qi$fd</span></tt>) for the tobit regression model is
defined as</p>
<div class="math">
<p><img src="_images/math/6af4ba00e718a455472875d35f977fd00b79e943.png" alt="\begin{aligned}
\text{FD}=E(Y\mid x_{1})-E(Y\mid x).\end{aligned}"/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (<tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/72b57ba1ea5551a32c3fa473511823ae37508662.png" alt="\begin{aligned}
\frac{1}{\sum t_{i}}\sum_{i:t_{i}=1}[Y_{i}(t_{i}=1)-E[Y_{i}(t_{i}=0)]],\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/10ea017369414222fb61b769804fb55f01f1f886.png" alt="t_{i}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/4313486a2444e99c09afd625a9f811c03d22538f.png" alt="t_{i}=1"/>) and control (<img class="math" src="_images/math/254a135214133c3926c66cf9682fd612cae45ce4.png" alt="t_{i}=0"/>) groups.</p>
</li>
</ul>
</div>
<div class="section" id="id125">
<h3>Output Values<a class="headerlink" href="#id125" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;tobit.bayes&quot;</span><span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the draws from the posterior distribution of the
<tt class="docutils literal"><span class="pre">coefficients</span></tt> by using <tt class="docutils literal"><span class="pre">z.out$coefficients</span></tt>, and view a default
summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other elements
available through the <tt class="docutils literal"><span class="pre">$</span></tt> operator are listed below.</p>
<ul class="simple">
<li>From the <tt class="docutils literal"><span class="pre">zelig()</span></tt> output object <tt class="docutils literal"><span class="pre">z.out</span></tt>, you may extract:<ul>
<li><tt class="docutils literal"><span class="pre">coefficients</span></tt>: draws from the posterior distributions of the
estimated parameters. The first <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> columns contain the
posterior draws of the coefficients <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/>, and the last
column contains the posterior draws of the variance
<img class="math" src="_images/math/93c29bfedc993de115781b4fbb7e3a8b9e8d6b51.png" alt="\sigma^2"/>.</li>
<li>zelig.data: the input data frame if save.data = TRUE.</li>
<li><tt class="docutils literal"><span class="pre">seed</span></tt>: the random seed used in the model.</li>
</ul>
</li>
<li>From the <tt class="docutils literal"><span class="pre">sim()</span></tt> output object <tt class="docutils literal"><span class="pre">s.out</span></tt>:<ul>
<li><tt class="docutils literal"><span class="pre">qi$ev</span></tt>: the simulated expected value for the specified values
of <tt class="docutils literal"><span class="pre">x</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$fd</span></tt>: the simulated first difference in the expected values
given the values specified in <tt class="docutils literal"><span class="pre">x</span></tt> and <tt class="docutils literal"><span class="pre">x1</span></tt>.</li>
<li><tt class="docutils literal"><span class="pre">qi$att.ev</span></tt>: the simulated average expected treatment effect for
the treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id126">
<h3>See also<a class="headerlink" href="#id126" title="Permalink to this headline">¶</a></h3>
<p>Bayesian tobit regression is part of the MCMCpack library by Andrew D.
Martin and Kevin M. Quinn . The convergence diagnostics are part of the
CODA library by Martyn Plummer, Nicky Best, Kate Cowles, and Karen Vines.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-gammagee">
<span id="zgammagee"></span><h2>zelig-gammagee<a class="headerlink" href="#zelig-gammagee" title="Permalink to this headline">¶</a></h2>
<p>The GEE gamma is similar to standard gamma regression (appropriate when
you have an uncensored, positive-valued, continuous dependent variable
such as the time until a parliamentary cabinet falls). Unlike in gamma
regression, GEE gamma allows for dependence within clusters, such as in
longitudinal data, although its use is not limited to just panel data.
GEE models make no distributional assumptions but require three
specifications: a mean function, a variance function, and a “working”
correlation matrix for the clusters, which models the dependence of each
observation with other observations in the same cluster. The “working”
correlation matrix is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/> matrix of correlations, where
<img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/> is the size of the largest cluster and the elements of the
matrix are correlations between within-cluster observations. The appeal
of GEE models is that it gives consistent estimates of the parameters
and consistent estimates of the standard errors can be obtained using a
robust “sandwich” estimator even if the “working” correlation matrix is
incorrectly specified. If the “working” correlation matrix is correctly
specified, GEE models will give more efficient estimates of the
parameters. GEE models measure population-averaged effects as opposed to
cluster-specific effects.</p>
<div class="section" id="id127">
<h3>Syntax<a class="headerlink" href="#id127" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zgammagee<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma.gee&quot;</span><span class="p">,</span>
         id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma.gee&quot;</span><span class="p">,</span>
               id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>where <tt class="docutils literal"><span class="pre">id</span></tt> is a variable which identifies the clusters. The data
should be sorted by <tt class="docutils literal"><span class="pre">id</span></tt> and should be ordered within each cluster
when appropriate.</p>
</div>
<div class="section" id="id128">
<h3>Additional Inputs<a class="headerlink" href="#id128" title="Permalink to this headline">¶</a></h3>
<p>Use the following arguments to specify the structure of the “working”
correlations within clusters:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">corstr</span></tt>: character string specifying the correlation structure:
&#8220;independence&#8221;, &#8220;exchangeable&#8221;, &#8220;ar1&#8221;, &#8220;unstructured&#8221; and
&#8220;userdefined&#8221;</li>
<li>See <tt class="docutils literal"><span class="pre">geeglm</span></tt> in package <tt class="docutils literal"><span class="pre">geepack</span></tt> for other function arguments.</li>
</ul>
</div>
<div class="section" id="id129">
<h3>Examples<a class="headerlink" href="#id129" title="Permalink to this headline">¶</a></h3>
<div class="section" id="example-with-exchangeable-dependence">
<h4>Example with Exchangeable Dependence<a class="headerlink" href="#example-with-exchangeable-dependence" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample turnout dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>coalition<span class="p">)</span>
</pre></div>
</div>
<p>Sorted variable identifying clusters</p>
<div class="highlight-r"><div class="highlight"><pre>coalition<span class="o">$</span>cluster <span class="o">&lt;-</span> c<span class="p">(</span>rep<span class="p">(</span>c<span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">62</span><span class="p">),</span> <span class="m">5</span><span class="p">),</span>rep<span class="p">(</span>c<span class="p">(</span><span class="m">63</span><span class="p">),</span> <span class="m">4</span><span class="p">))</span>
sorted.coalition <span class="o">&lt;-</span> coalition<span class="p">[</span>order<span class="p">(</span>coalition<span class="o">$</span>cluster<span class="p">),</span> <span class="p">]</span>
</pre></div>
</div>
<p>Estimating model and presenting summary:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>duration <span class="o">~</span> fract <span class="o">+</span> numst2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma.gee&quot;</span><span class="p">,</span>
               id <span class="o">=</span> <span class="s">&quot;cluster&quot;</span><span class="p">,</span> data <span class="o">=</span> sorted.coalition<span class="p">,</span>
               corstr <span class="o">=</span> <span class="s">&quot;exchangeable&quot;</span><span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Patrick Lam. 2011.</span>
<span class="c">##   zgammagee: General Estimating Equation for Gamma Regression</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## geepack::geeglm(formula = duration ~ fract + numst2, family = Gamma(&quot;inverse&quot;),</span>
<span class="c">##     data = ., id = c(1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3,</span>
<span class="c">##     3, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 6, 6, 6, 6, 6, 7, 7, 7,</span>
<span class="c">##     7, 7, 8, 8, 8, 8, 8, 9, 9, 9, 9, 9, 10, 10, 10, 10, 10, 11,</span>
<span class="c">##     11, 11, 11, 11, 12, 12, 12, 12, 12, 13, 13, 13, 13, 13, 14,</span>
<span class="c">##     14, 14, 14, 14, 15, 15, 15, 15, 15, 16, 16, 16, 16, 16, 17,</span>
<span class="c">##     17, 17, 17, 17, 18, 18, 18, 18, 18, 19, 19, 19, 19, 19, 20,</span>
<span class="c">##     20, 20, 20, 20, 21, 21, 21, 21, 21, 22, 22, 22, 22, 22, 23,</span>
<span class="c">##     23, 23, 23, 23, 24, 24, 24, 24, 24, 25, 25, 25, 25, 25, 26,</span>
<span class="c">##     26, 26, 26, 26, 27, 27, 27, 27, 27, 28, 28, 28, 28, 28, 29,</span>
<span class="c">##     29, 29, 29, 29, 30, 30, 30, 30, 30, 31, 31, 31, 31, 31, 32,</span>
<span class="c">##     32, 32, 32, 32, 33, 33, 33, 33, 33, 34, 34, 34, 34, 34, 35,</span>
<span class="c">##     35, 35, 35, 35, 36, 36, 36, 36, 36, 37, 37, 37, 37, 37, 38,</span>
<span class="c">##     38, 38, 38, 38, 39, 39, 39, 39, 39, 40, 40, 40, 40, 40, 41,</span>
<span class="c">##     41, 41, 41, 41, 42, 42, 42, 42, 42, 43, 43, 43, 43, 43, 44,</span>
<span class="c">##     44, 44, 44, 44, 45, 45, 45, 45, 45, 46, 46, 46, 46, 46, 47,</span>
<span class="c">##     47, 47, 47, 47, 48, 48, 48, 48, 48, 49, 49, 49, 49, 49, 50,</span>
<span class="c">##     50, 50, 50, 50, 51, 51, 51, 51, 51, 52, 52, 52, 52, 52, 53,</span>
<span class="c">##     53, 53, 53, 53, 54, 54, 54, 54, 54, 55, 55, 55, 55, 55, 56,</span>
<span class="c">##     56, 56, 56, 56, 57, 57, 57, 57, 57, 58, 58, 58, 58, 58, 59,</span>
<span class="c">##     59, 59, 59, 59, 60, 60, 60, 60, 60, 61, 61, 61, 61, 61, 62,</span>
<span class="c">##     62, 62, 62, 62, 63, 63, 63, 63), corstr = &quot;exchangeable&quot;)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">##   (Intercept)         fract        numst2</span>
<span class="c">## -0.0129634262  0.0001149139 -0.0174009664</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 314 Total (i.e. Null);  311 Residual</span>
<span class="c">##</span>
<span class="c">## Scale Link:                   identity</span>
<span class="c">## Estimated Scale Parameters:  [1] 0.6231419</span>
<span class="c">##</span>
<span class="c">## Correlation:  Structure = exchangeable    Link = identity</span>
<span class="c">## Estimated Correlation Parameters:</span>
<span class="c">##        alpha</span>
<span class="c">## -0.008086333</span>
<span class="c">##</span>
<span class="c">## Number of clusters:   63   Maximum cluster size: 5</span>
<span class="c">##</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Setting the explanatory variables at their default values (mode for factor variables and mean for non-factor variables), with numst2 set to the vector 0 = no crisis, 1 = crisis.</p>
<div class="highlight-r"><div class="highlight"><pre>x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">0</span><span class="p">)</span>
x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">,</span> numst2 <span class="o">=</span> <span class="m">1</span><span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.low<span class="p">,</span> x1 <span class="o">=</span> x.high<span class="p">)</span>
summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean       sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 14.44855 1.106705 14.34894 12.45345 16.79042</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%       2.5%    97.5%</span>
<span class="c">## [1,] 14.58999 18.61796 7.400069 0.04056687 66.71991</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean       sd     50%    2.5%    97.5%</span>
<span class="c">## [1,] 19.14387 1.074071 19.1044 17.1316 21.58457</span>
<span class="c">## pv</span>
<span class="c">##          mean       sd      50%      2.5%    97.5%</span>
<span class="c">## [1,] 19.69058 25.09517 10.78035 0.0497614 85.65704</span>
<span class="c">## fd</span>
<span class="c">##          mean       sd     50%     2.5%   97.5%</span>
<span class="c">## [1,] 4.695313 1.529355 4.70746 1.523622 7.75042</span>
</pre></div>
</div>
<p>Generate a plot of quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-gammagee" src="_images/Zelig-exp-1.png" />
<p class="caption">Zelig-gammagee</p>
</div>
</div>
</div>
<div class="section" id="the-model">
<h3>The Model<a class="headerlink" href="#the-model" title="Permalink to this headline">¶</a></h3>
<p>Suppose we have a panel dataset, with <img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> denoting the
positive-valued, continuous dependent variable for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> at
time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> is a vector or cluster of correlated data
where <img class="math" src="_images/math/b840d9057a2e3bad080c274b8f1d9303520f9052.png" alt="y_{it}"/> is correlated with <img class="math" src="_images/math/da57192d38416f50ec88d145367cc7d5be0c06c5.png" alt="y_{it^\prime}"/> for some
or all <img class="math" src="_images/math/2d3acc5e57adc9f94593e8bb1b3b28df87caf045.png" alt="t, t^\prime"/>. Note that the model assumes correlations
within <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> but independence across <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by the joint and marginal
distributions</p>
<div class="math">
<p><img src="_images/math/07e7595e1d9c1222ab0eab8fa6aa39fe54251868.png" alt="\begin{aligned}
Y_{i} &amp;\sim&amp; f(y_{i} \mid \lambda_{i})\\
Y_{it} &amp;\sim&amp; g(y_{it} \mid \lambda_{it})\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/0001d02b63ede2fe3219e05a7cd09c82ae6298b6.png" alt="f"/> and <img class="math" src="_images/math/67f4714f065d485540ad40829e0717bf75e9dd85.png" alt="g"/> are unspecified distributions with
means <img class="math" src="_images/math/3d9a1fde3ad183d72319e293ff637c7528c831a4.png" alt="\lambda_{i}"/> and <img class="math" src="_images/math/5fc838daf306e42980d81b1d6b359f4a47fbd630.png" alt="\lambda_{it}"/>. GEE models make
no distributional assumptions and only require three specifications:
a mean function, a variance function, and a correlation structure.</p>
</li>
<li><p class="first">The <em>systematic component</em> is the <em>mean function</em>, given by:</p>
<div class="math">
<p><img src="_images/math/1bf427decffdf24a129a20e250662475e0b743fa.png" alt="\lambda_{it} = \frac{1}{x_{it} \beta}"/></p>
</div><p>where <img class="math" src="_images/math/d43ef14e28002ff45394741fc187dced216e80f0.png" alt="x_{it}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector
of coefficients.</p>
</li>
<li><p class="first">The <em>variance function</em> is given by:</p>
<div class="math">
<p><img src="_images/math/1dcc2e3f0a1cde0a5a95bc9337eee01383107fb9.png" alt="V_{it} = \lambda_{it}^2 = \frac{1}{(x_{it} \beta)^2}"/></p>
</div></li>
<li><p class="first">The <em>correlation structure</em> is defined by a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/>
“working” correlation matrix, where <img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/> is the size of the
largest cluster. Users must specify the structure of the “working”
correlation matrix <em>a priori</em>. The “working” correlation matrix then
enters the variance term for each <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>, given by:</p>
<div class="math">
<p><img src="_images/math/16993e098fe3b3316a2033c341dedf1a4a5722f0.png" alt="V_{i} = \phi \, A_{i}^{\frac{1}{2}} R_{i}(\alpha) A_{i}^{\frac{1}{2}}"/></p>
</div><p>where <img class="math" src="_images/math/7f4051f79dc41b4b9b5b105d110e19898642c02d.png" alt="A_{i}"/> is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/> diagonal matrix with the
variance function <img class="math" src="_images/math/ce380154fb239a28baa4d6a9b375ac5d8346653a.png" alt="V_{it} = \lambda_{it}^2"/> as the
<img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>th diagonal element, <img class="math" src="_images/math/165cf808f99660333864da96d49d02862c5804ba.png" alt="R_{i}(\alpha)"/> is the
“working” correlation matrix, and <img class="math" src="_images/math/10e009bdb83f96c5f47c58b34d5d4b12ef268d5b.png" alt="\phi"/> is a scale parameter.
The parameters are then estimated via a quasi-likelihood approach.</p>
</li>
<li><p class="first">In GEE models, if the mean is correctly specified, but the variance
and correlation structure are incorrectly specified, then GEE models
provide consistent estimates of the parameters and thus the mean
function as well, while consistent estimates of the standard errors
can be obtained via a robust “sandwich” estimator. Similarly, if the
mean and variance are correctly specified but the correlation
structure is incorrectly specified, the parameters can be estimated
consistently and the standard errors can be estimated consistently
with the sandwich estimator. If all three are specified correctly,
then the estimates of the parameters are more efficient.</p>
</li>
</ul>
</div>
<div class="section" id="id130">
<h3>Quantities of Interest<a class="headerlink" href="#id130" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">All quantities of interest are for marginal means rather than joint
means.</p>
</li>
<li><p class="first">The method of bootstrapping generally should not be used in GEE
models. If you must bootstrap, bootstrapping should be done within
clusters, which is not currently supported in Zelig. For conditional
prediction models, data should be matched within clusters.</p>
</li>
<li><p class="first">The expected values (qi$ev) for the GEE gamma model is the mean:</p>
<div class="math">
<p><img src="_images/math/34774f15177e0863ed0b736b9ab797a97e4dc19d.png" alt="E(Y) =
  \lambda_{c} = \frac{1}{x_c \beta},"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution, where
<img class="math" src="_images/math/5502b77b431ce3a22763c40fede2822fe42a5aa2.png" alt="x_{c}"/> is a vector of values, one for each independent
variable, chosen by the user.</p>
</li>
<li><p class="first">The first difference (qi$fd) for the GEE gamma model is defined as</p>
<div class="math">
<p><img src="_images/math/8f669920018c72d7f23a4ba843925381c4703f17.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1) - \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/40ac87a54bf074ab721d0606edc354d20bfe2890.png" alt="\frac{1}{\sum_{i=1}^n \sum_{t=1}^T tr_{it}}\sum_{i:tr_{it}=1}^n \sum_{t:tr_{it}=1}^T \left\{ Y_{it}(tr_{it}=1) -
      E[Y_{it}(tr_{it}=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/e6f125c03f6d788e356d369529940f022a7f2758.png" alt="tr_{it}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/6456da07c94c2f41fd2db133459327f848c0f4bd.png" alt="tr_{it}=1"/>) and control (<img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/55447d1ae15443b039fe51a56aee680735c5eff7.png" alt="E[Y_{it}(tr_{it}=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id131">
<h3>Output Values<a class="headerlink" href="#id131" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>y <span class="o">~</span> x<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;gamma.gee&quot;</span><span class="p">,</span> id<span class="p">,</span> data<span class="p">)</span>
</pre></div>
</div>
<p>then you may see a default summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other
elements available through the $ operator are listed below.</p>
<ul class="simple">
<li>From the zelig() output object z.out, you may extract:<ul>
<li>coefficients: parameter estimates for the explanatory variables.</li>
<li>residuals: the working residuals in the final iteration of the
fit.</li>
<li>fitted.values: the vector of fitted values for the systemic
component.</li>
<li>linear.predictors: the vector of <img class="math" src="_images/math/62d42b965bd7739c65b214098af7195a9940d3fc.png" alt="x_{it}\beta"/></li>
<li>max.id: the size of the largest cluster.</li>
</ul>
</li>
<li>From summary(z.out), you may extract:<ul>
<li>coefficients: the parameter estimates with their associated
standard errors, <img class="math" src="_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/>-values, and <img class="math" src="_images/math/84d7271dd9e78c1e05d6c3c6ecb60309ef7dfc73.png" alt="z"/>-statistics.</li>
<li>working.correlation: the “working” correlation matrix</li>
</ul>
</li>
<li>From the sim() output object s.out, you may extract quantities of
interest arranged as matrices indexed by simulation <img class="math" src="_images/math/c69691d64985442217922c8d34e835a9dea60178.png" alt="\times"/>
x-observation (for more than one x-observation). Available quantities
are:<ul>
<li>qi$ev: the simulated expected values for the specified values of
x.</li>
<li>qi$fd: the simulated first difference in the expected
probabilities for the values specified in x and x1.</li>
<li>qi$att.ev: the simulated average expected treatment effect for the
treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id132">
<h3>See also<a class="headerlink" href="#id132" title="Permalink to this headline">¶</a></h3>
<p>The geeglm function is part of the geepack package by Søren Højsgaard,
Ulrich Halekoh and Jun Yan. Advanced users may wish to refer
to <tt class="docutils literal"><span class="pre">help(geepack)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-probitgee">
<span id="zprobitgee"></span><h2>zelig-probitgee<a class="headerlink" href="#zelig-probitgee" title="Permalink to this headline">¶</a></h2>
<p>The GEE probit estimates the same model as the standard probit
regression (appropriate when you have a dichotomous dependent variable
and a set of explanatory variables). Unlike in probit regression, GEE
probit allows for dependence within clusters, such as in longitudinal
data, although its use is not limited to just panel data. The user must
first specify a “working” correlation matrix for the clusters, which
models the dependence of each observation with other observations in the
same cluster. The “working” correlation matrix is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/>
matrix of correlations, where <img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/> is the size of the largest
cluster and the elements of the matrix are correlations between
within-cluster observations. The appeal of GEE models is that it gives
consistent estimates of the parameters and consistent estimates of the
standard errors can be obtained using a robust “sandwich” estimator even
if the “working” correlation matrix is incorrectly specified. If the
“working” correlation matrix is correctly specified, GEE models will
give more efficient estimates of the parameters. GEE models measure
population-averaged effects as opposed to cluster-specific effects.</p>
<div class="section" id="id133">
<h3>Syntax<a class="headerlink" href="#id133" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zgammagee<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.gee&quot;</span><span class="p">,</span>
         id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.gee&quot;</span><span class="p">,</span>
               id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>where <tt class="docutils literal"><span class="pre">id</span></tt> is a variable which identifies the clusters. The data
should be sorted by <tt class="docutils literal"><span class="pre">id</span></tt> and should be ordered within each cluster
when appropriate.</p>
</div>
<div class="section" id="id134">
<h3>Additional Inputs<a class="headerlink" href="#id134" title="Permalink to this headline">¶</a></h3>
<p>Use the following arguments to specify the structure of the “working”
correlations within clusters:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">corstr</span></tt>: character string specifying the correlation structure:
&#8220;independence&#8221;, &#8220;exchangeable&#8221;, &#8220;ar1&#8221;, &#8220;unstructured&#8221; and
&#8220;userdefined&#8221;</li>
<li>See <tt class="docutils literal"><span class="pre">geeglm</span></tt> in package <tt class="docutils literal"><span class="pre">geepack</span></tt> for other function arguments.</li>
</ul>
</div>
<div class="section" id="id135">
<h3>Examples<a class="headerlink" href="#id135" title="Permalink to this headline">¶</a></h3>
<div class="section" id="example-with-stationary-3-dependence">
<h4>Example with Stationary 3 Dependence<a class="headerlink" href="#example-with-stationary-3-dependence" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample turnout dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>turnout<span class="p">)</span>
</pre></div>
</div>
<p>Variable identifying clusters</p>
<div class="highlight-r"><div class="highlight"><pre>turnout<span class="o">$</span>cluster <span class="o">&lt;-</span> rep<span class="p">(</span>c<span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">200</span><span class="p">),</span> <span class="m">10</span><span class="p">)</span>
sorted.turnout <span class="o">&lt;-</span> turnout<span class="p">[</span>order<span class="p">(</span>turnout<span class="o">$</span>cluster<span class="p">),</span> <span class="p">]</span>
</pre></div>
</div>
<p>Estimating parameter values:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out1 <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> race <span class="o">+</span> educate<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.gee&quot;</span><span class="p">,</span>
                id <span class="o">=</span> <span class="s">&quot;cluster&quot;</span><span class="p">,</span> data <span class="o">=</span> sorted.turnout<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Patrick Lam. 2011.</span>
<span class="c">##   probitgee: General Estimating Equation for Probit Regression</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Setting values for the explanatory variables to their default values:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out1 <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">)</span>
</pre></div>
</div>
<p>Simulating quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>s.out1 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##           mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.7710248 0.01148642 0.7715162 0.7476183 0.7937804</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.229 0.771</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out1<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-probitgee1" src="_images/Zelig-probitgee1-1.png" />
<p class="caption">Zelig-probitgee1</p>
</div>
</div>
<div class="section" id="id136">
<h4>Simulating First Differences<a class="headerlink" href="#id136" title="Permalink to this headline">¶</a></h4>
<p>Estimating the risk difference (and risk ratio) between low education
(25th percentile) and high education (75th percentile) while all the other variables held at their default values.</p>
<div class="highlight-r"><div class="highlight"><pre>x.high <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.75</span><span class="p">))</span>
x.low <span class="o">&lt;-</span> setx<span class="p">(</span>z.out1<span class="p">,</span> educate <span class="o">=</span> quantile<span class="p">(</span>turnout<span class="o">$</span>educate<span class="p">,</span> prob <span class="o">=</span> <span class="m">0.25</span><span class="p">))</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>s.out2 <span class="o">&lt;-</span> sim<span class="p">(</span>z.out1<span class="p">,</span> x <span class="o">=</span> x.high<span class="p">,</span> x1 <span class="o">=</span> x.low<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##           mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.8243189 0.01064132 0.8251121 0.8034013 0.8442554</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.181 0.819</span>
<span class="c">##</span>
<span class="c">##  sim x1 :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##           mean         sd       50%      2.5%     97.5%</span>
<span class="c">## [1,] 0.7064974 0.01460634 0.7059701 0.6776687 0.7355476</span>
<span class="c">## pv</span>
<span class="c">##          0     1</span>
<span class="c">## [1,] 0.292 0.708</span>
<span class="c">## fd</span>
<span class="c">##            mean         sd        50%       2.5%       97.5%</span>
<span class="c">## [1,] -0.1178215 0.01203205 -0.1175201 -0.1422335 -0.09466796</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out2<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-probitgee2" src="_images/Zelig-probitgee2-1.png" />
<p class="caption">Zelig-probitgee2</p>
</div>
</div>
<div class="section" id="example-with-fixed-correlation-structure">
<h4>Example with Fixed Correlation Structure<a class="headerlink" href="#example-with-fixed-correlation-structure" title="Permalink to this headline">¶</a></h4>
<p>User-defined correlation structure</p>
<div class="highlight-r"><div class="highlight"><pre>corr.mat <span class="o">&lt;-</span> matrix<span class="p">(</span>rep<span class="p">(</span><span class="m">0.5</span><span class="p">,</span> <span class="m">100</span><span class="p">),</span> nrow <span class="o">=</span> <span class="m">10</span><span class="p">,</span> ncol <span class="o">=</span> <span class="m">10</span><span class="p">)</span>
diag<span class="p">(</span>corr.mat<span class="p">)</span> <span class="o">&lt;-</span> <span class="m">1</span>
corr.mat <span class="o">&lt;-</span> fixed2Zcor<span class="p">(</span>corr.mat<span class="p">,</span> id<span class="o">=</span>sorted.turnout<span class="o">$</span>cluster<span class="p">,</span> waves<span class="o">=</span>sorted.turnout<span class="o">$</span>race<span class="p">)</span>
</pre></div>
</div>
<p>Generating empirical estimates:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out2 <span class="o">&lt;-</span> zelig<span class="p">(</span>vote <span class="o">~</span> race <span class="o">+</span> educate<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;probit.gee&quot;</span><span class="p">,</span>
                id <span class="o">=</span> <span class="s">&quot;cluster&quot;</span><span class="p">,</span> data <span class="o">=</span> sorted.turnout<span class="p">,</span>
                corstr <span class="o">=</span> <span class="s">&quot;fixed&quot;</span><span class="p">,</span> zcor <span class="o">=</span> corr.mat<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Patrick Lam. 2011.</span>
<span class="c">##   probitgee: General Estimating Equation for Probit Regression</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<p>Viewing the regression output:</p>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out2<span class="p">)</span>
</pre></div>
</div>
</div>
</div>
<div class="section" id="id137">
<h3>The Model<a class="headerlink" href="#id137" title="Permalink to this headline">¶</a></h3>
<p>Suppose we have a panel dataset, with <img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> denoting the binary
dependent variable for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/>
is a vector or cluster of correlated data where <img class="math" src="_images/math/b840d9057a2e3bad080c274b8f1d9303520f9052.png" alt="y_{it}"/> is
correlated with <img class="math" src="_images/math/da57192d38416f50ec88d145367cc7d5be0c06c5.png" alt="y_{it^\prime}"/> for some or all
<img class="math" src="_images/math/2d3acc5e57adc9f94593e8bb1b3b28df87caf045.png" alt="t, t^\prime"/>. Note that the model assumes correlations within
<img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> but independence across <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by the joint and marginal
distributions</p>
<div class="math">
<p><img src="_images/math/99e52afc172ca96e215469c88f249c909a028bb5.png" alt="\begin{aligned}
Y_{i} &amp;\sim&amp; f(y_{i} \mid \pi_{i})\\
Y_{it} &amp;\sim&amp; g(y_{it} \mid \pi_{it})\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/0001d02b63ede2fe3219e05a7cd09c82ae6298b6.png" alt="f"/> and <img class="math" src="_images/math/67f4714f065d485540ad40829e0717bf75e9dd85.png" alt="g"/> are unspecified distributions with
means <img class="math" src="_images/math/56d13901556c9d691c7d1893b7b9329d60794bd6.png" alt="\pi_{i}"/> and <img class="math" src="_images/math/241cd55f42e0d99b5eaf7d34334bf6f3d2ca4c29.png" alt="\pi_{it}"/>. GEE models make no
distributional assumptions and only require three specifications: a
mean function, a variance function, and a correlation structure.</p>
</li>
<li><p class="first">The <em>systematic component</em> is the <em>mean function</em>, given by:</p>
<div class="math">
<p><img src="_images/math/565417850a92690cacd5341abe762a0a1f00425b.png" alt="\pi_{it} = \Phi(x_{it} \beta)"/></p>
</div><p>where <img class="math" src="_images/math/e4f08b55c6f99b472e0a295b96ff6487d545dfd0.png" alt="\Phi(\mu)"/> is the cumulative distribution function of
the Normal distribution with mean 0 and unit variance, <img class="math" src="_images/math/d43ef14e28002ff45394741fc187dced216e80f0.png" alt="x_{it}"/>
is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>
at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector of coefficients.</p>
</li>
<li><p class="first">The <em>variance function</em> is given by:</p>
<div class="math">
<p><img src="_images/math/0366b791aadbdf9d17328c80ea70cebf89e3b9df.png" alt="V_{it} = \pi_{it} (1-\pi_{it})"/></p>
</div></li>
<li><p class="first">The <em>correlation structure</em> is defined by a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/>
“working” correlation matrix, where <img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/> is the size of the
largest cluster. Users must specify the structure of the “working”
correlation matrix <em>a priori</em>. The “working” correlation matrix then
enters the variance term for each <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>, given by:</p>
<div class="math">
<p><img src="_images/math/16993e098fe3b3316a2033c341dedf1a4a5722f0.png" alt="V_{i} = \phi \, A_{i}^{\frac{1}{2}} R_{i}(\alpha) A_{i}^{\frac{1}{2}}"/></p>
</div><p>where <img class="math" src="_images/math/7f4051f79dc41b4b9b5b105d110e19898642c02d.png" alt="A_{i}"/> is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/> diagonal matrix with the
variance function <img class="math" src="_images/math/41fbbeaa3c8f0735faaab30c5bcc4432123fc864.png" alt="V_{it} = \pi_{it} (1-\pi_{it})"/> as the
<img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>th diagonal element, <img class="math" src="_images/math/165cf808f99660333864da96d49d02862c5804ba.png" alt="R_{i}(\alpha)"/> is the
“working” correlation matrix, and <img class="math" src="_images/math/10e009bdb83f96c5f47c58b34d5d4b12ef268d5b.png" alt="\phi"/> is a scale parameter.
The parameters are then estimated via a quasi-likelihood approach.</p>
</li>
<li><p class="first">In GEE models, if the mean is correctly specified, but the variance
and correlation structure are incorrectly specified, then GEE models
provide consistent estimates of the parameters and thus the mean
function as well, while consistent estimates of the standard errors
can be obtained via a robust “sandwich” estimator. Similarly, if the
mean and variance are correctly specified but the correlation
structure is incorrectly specified, the parameters can be estimated
consistently and the standard errors can be estimated consistently
with the sandwich estimator. If all three are specified correctly,
then the estimates of the parameters are more efficient.</p>
</li>
<li><p class="first">The robust “sandwich” estimator gives consistent estimates of the
standard errors when the correlations are specified incorrectly only
if the number of units <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> is relatively large and the number
of repeated periods <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> is relatively small. Otherwise, one
should use the “naïve” model-based standard errors, which assume that
the specified correlations are close approximations to the true
underlying correlations. See for more details.</p>
</li>
</ul>
</div>
<div class="section" id="id138">
<h3>Quantities of Interest<a class="headerlink" href="#id138" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">All quantities of interest are for marginal means rather than joint
means.</p>
</li>
<li><p class="first">The method of bootstrapping generally should not be used in GEE
models. If you must bootstrap, bootstrapping should be done within
clusters, which is not currently supported in Zelig. For conditional
prediction models, data should be matched within clusters.</p>
</li>
<li><p class="first">The expected values (qi$ev) for the GEE probit model are simulations
of the predicted probability of a success:</p>
<div class="math">
<p><img src="_images/math/a46d02dd9a03502f3cc57aaa0cd73a509c5814e2.png" alt="E(Y) =
  \pi_{c}= \Phi(x_{c} \beta),"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution, where
<img class="math" src="_images/math/5502b77b431ce3a22763c40fede2822fe42a5aa2.png" alt="x_{c}"/> is a vector of values, one for each independent
variable, chosen by the user.</p>
</li>
<li><p class="first">The first difference (qi$fd) for the GEE probit model is defined as</p>
<div class="math">
<p><img src="_images/math/8f669920018c72d7f23a4ba843925381c4703f17.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1) - \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">The risk ratio (qi$rr) is defined as</p>
<div class="math">
<p><img src="_images/math/8540cd3e413420e1eceaa11af2b8fc70ef717f37.png" alt="\textrm{RR} = \Pr(Y = 1 \mid x_1) \ / \ \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/40ac87a54bf074ab721d0606edc354d20bfe2890.png" alt="\frac{1}{\sum_{i=1}^n \sum_{t=1}^T tr_{it}}\sum_{i:tr_{it}=1}^n \sum_{t:tr_{it}=1}^T \left\{ Y_{it}(tr_{it}=1) -
      E[Y_{it}(tr_{it}=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/e6f125c03f6d788e356d369529940f022a7f2758.png" alt="tr_{it}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/6456da07c94c2f41fd2db133459327f848c0f4bd.png" alt="tr_{it}=1"/>) and control (<img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/55447d1ae15443b039fe51a56aee680735c5eff7.png" alt="E[Y_{it}(tr_{it}=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id139">
<h3>Output Values<a class="headerlink" href="#id139" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For examle, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">probit.gee,</span> <span class="pre">id,</span> <span class="pre">data)</span></tt>, then you may
examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the coefficients by using z.out$coefficients, and
a default summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other
elements available through the $ operator are listed below.</p>
<ul class="simple">
<li>From the zelig() output object z.out, you may extract:<ul>
<li>coefficients: parameter estimates for the explanatory variables.</li>
<li>residuals: the working residuals in the final iteration of the
fit.</li>
<li>fitted.values: the vector of fitted values for the systemic
component, <img class="math" src="_images/math/241cd55f42e0d99b5eaf7d34334bf6f3d2ca4c29.png" alt="\pi_{it}"/>.</li>
<li>linear.predictors: the vector of <img class="math" src="_images/math/62d42b965bd7739c65b214098af7195a9940d3fc.png" alt="x_{it}\beta"/></li>
<li>max.id: the size of the largest cluster.</li>
</ul>
</li>
<li>From summary(z.out), you may extract:<ul>
<li>coefficients: the parameter estimates with their associated
standard errors, <img class="math" src="_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/>-values, and <img class="math" src="_images/math/84d7271dd9e78c1e05d6c3c6ecb60309ef7dfc73.png" alt="z"/>-statistics.</li>
<li>working.correlation: the “working” correlation matrix</li>
</ul>
</li>
<li>From the sim() output object s.out, you may extract quantities of
interest arranged as matrices indexed by simulation <img class="math" src="_images/math/c69691d64985442217922c8d34e835a9dea60178.png" alt="\times"/>
x-observation (for more than one x-observation). Available quantities
are:<ul>
<li>qi$ev: the simulated expected probabilities for the specified
values of x.</li>
<li>qi$fd: the simulated first difference in the expected
probabilities for the values specified in x and x1.</li>
<li>qi$rr: the simulated risk ratio for the expected probabilities
simulated from x and x1.</li>
<li>qi$att.ev: the simulated average expected treatment effect for the
treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id140">
<h3>See also<a class="headerlink" href="#id140" title="Permalink to this headline">¶</a></h3>
<p>The geeglm function is part of the geepack package by Søren Højsgaard,
Ulrich Halekoh and Jun Yan. Advanced users may wish to refer
to <tt class="docutils literal"><span class="pre">help(geepack)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
</div>
</div>
<div class="section" id="zelig-poissongee">
<span id="zpoissongee"></span><h2>zelig-poissongee<a class="headerlink" href="#zelig-poissongee" title="Permalink to this headline">¶</a></h2>
<p>The GEE poisson estimates the same model as the standard poisson
regression (appropriate when your dependent variable represents the
number of independent events that occur during a fixed period of time).
Unlike in poisson regression, GEE poisson allows for dependence within
clusters, such as in longitudinal data, although its use is not limited
to just panel data. The user must first specify a “working” correlation
matrix for the clusters, which models the dependence of each observation
with other observations in the same cluster. The “working” correlation
matrix is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/> matrix of correlations, where <img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/>
is the size of the largest cluster and the elements of the matrix are
correlations between within-cluster observations. The appeal of GEE
models is that it gives consistent estimates of the parameters and
consistent estimates of the standard errors can be obtained using a
robust “sandwich” estimator even if the “working” correlation matrix is
incorrectly specified. If the “working” correlation matrix is correctly
specified, GEE models will give more efficient estimates of the
parameters. GEE models measure population-averaged effects as opposed to
cluster-specific effects.</p>
<div class="section" id="id141">
<h3>Syntax<a class="headerlink" href="#id141" title="Permalink to this headline">¶</a></h3>
<p>With reference classes:</p>
<div class="highlight-r"><div class="highlight"><pre>z5 <span class="o">&lt;-</span> zgammagee<span class="o">$</span>new<span class="p">()</span>
z5<span class="o">$</span>zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.gee&quot;</span><span class="p">,</span>
         id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
z5<span class="o">$</span>setx<span class="p">()</span>
z5<span class="o">$</span>sim<span class="p">()</span>
</pre></div>
</div>
<p>With the Zelig 4 compatibility wrappers:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>Y <span class="o">~</span> X1 <span class="o">+</span> X2<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.gee&quot;</span><span class="p">,</span>
               id <span class="o">=</span> <span class="s">&quot;X3&quot;</span><span class="p">,</span> data <span class="o">=</span> mydata<span class="p">)</span>
x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
</pre></div>
</div>
<p>where <tt class="docutils literal"><span class="pre">id</span></tt> is a variable which identifies the clusters. The data
should be sorted by <tt class="docutils literal"><span class="pre">id</span></tt> and should be ordered within each cluster
when appropriate.</p>
</div>
<div class="section" id="id142">
<h3>Additional Inputs<a class="headerlink" href="#id142" title="Permalink to this headline">¶</a></h3>
<p>Use the following arguments to specify the structure of the “working”
correlations within clusters:</p>
<ul class="simple">
<li><tt class="docutils literal"><span class="pre">corstr</span></tt>: character string specifying the correlation structure:
&#8220;independence&#8221;, &#8220;exchangeable&#8221;, &#8220;ar1&#8221;, &#8220;unstructured&#8221; and
&#8220;userdefined&#8221;</li>
<li>See <tt class="docutils literal"><span class="pre">geeglm</span></tt> in package <tt class="docutils literal"><span class="pre">geepack</span></tt> for other function arguments.</li>
</ul>
</div>
<div class="section" id="id143">
<h3>Examples<a class="headerlink" href="#id143" title="Permalink to this headline">¶</a></h3>
<div class="section" id="id144">
<h4>Example with Exchangeable Dependence<a class="headerlink" href="#id144" title="Permalink to this headline">¶</a></h4>
<p>Attaching the sample turnout dataset:</p>
<div class="highlight-r"><div class="highlight"><pre>data<span class="p">(</span>sanction<span class="p">)</span>
</pre></div>
</div>
<p>Variable identifying clusters</p>
<div class="highlight-r"><div class="highlight"><pre>sanction<span class="o">$</span>cluster <span class="o">&lt;-</span> c<span class="p">(</span>rep<span class="p">(</span>c<span class="p">(</span><span class="m">1</span><span class="o">:</span><span class="m">15</span><span class="p">),</span> <span class="m">5</span><span class="p">),</span> rep<span class="p">(</span>c<span class="p">(</span><span class="m">16</span><span class="p">),</span> <span class="m">3</span><span class="p">))</span>
</pre></div>
</div>
<p>Sorting by cluster</p>
<div class="highlight-r"><div class="highlight"><pre>sorted.sanction <span class="o">&lt;-</span> sanction<span class="p">[</span>order<span class="p">(</span>sanction<span class="o">$</span>cluster<span class="p">),</span> <span class="p">]</span>
</pre></div>
</div>
<p>Estimating model and presenting summary:</p>
<div class="highlight-r"><div class="highlight"><pre>z.out <span class="o">&lt;-</span> zelig<span class="p">(</span>num <span class="o">~</span> target <span class="o">+</span> coop<span class="p">,</span> model <span class="o">=</span> <span class="s">&quot;poisson.gee&quot;</span><span class="p">,</span>
               id <span class="o">=</span> <span class="s">&quot;cluster&quot;</span><span class="p">,</span> data <span class="o">=</span> sorted.sanction<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## How to cite this model in Zelig:</span>
<span class="c">##   Patrick Lam. 2011.</span>
<span class="c">##   poissongee: General Estimating Equation for Poisson Regression</span>
<span class="c">##   in Kosuke Imai, Gary King, and Olivia Lau, &quot;Zelig: Everyone&#39;s Statistical Software,&quot;</span>
<span class="c">##   http://zeligproject.org/</span>
</pre></div>
</div>
<div class="highlight-r"><div class="highlight"><pre>summary<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">## Model:</span>
<span class="c">## $by</span>
<span class="c">## [1] 1</span>
<span class="c">##</span>
<span class="c">##</span>
<span class="c">## Call:</span>
<span class="c">## geepack::geeglm(formula = num ~ target + coop, family = poisson(&quot;log&quot;),</span>
<span class="c">##     data = ., id = c(1, 1, 1, 1, 1, 2, 2, 2, 2, 2, 3, 3, 3, 3,</span>
<span class="c">##     3, 4, 4, 4, 4, 4, 5, 5, 5, 5, 5, 6, 6, 6, 6, 6, 7, 7, 7,</span>
<span class="c">##     7, 7, 8, 8, 8, 8, 8, 9, 9, 9, 9, 9, 10, 10, 10, 10, 10, 11,</span>
<span class="c">##     11, 11, 11, 11, 12, 12, 12, 12, 12, 13, 13, 13, 13, 13, 14,</span>
<span class="c">##     14, 14, 14, 14, 15, 15, 15, 15, 15, 16, 16, 16), corstr = &quot;independence&quot;)</span>
<span class="c">##</span>
<span class="c">## Coefficients:</span>
<span class="c">## (Intercept)      target        coop</span>
<span class="c">## -0.96771994 -0.02102351  1.21081908</span>
<span class="c">##</span>
<span class="c">## Degrees of Freedom: 78 Total (i.e. Null);  75 Residual</span>
<span class="c">##</span>
<span class="c">## Scale Link:                   identity</span>
<span class="c">## Estimated Scale Parameters:  [1] 16.4486</span>
<span class="c">##</span>
<span class="c">## Correlation:  Structure = independence</span>
<span class="c">## Number of clusters:   16   Maximum cluster size: 5</span>
<span class="c">##</span>
<span class="c">## Next step: Use &#39;setx&#39; method</span>
</pre></div>
</div>
<p>Set explanatory variables to their default values:</p>
<div class="highlight-r"><div class="highlight"><pre>x.out <span class="o">&lt;-</span> setx<span class="p">(</span>z.out<span class="p">)</span>
</pre></div>
</div>
<p>Simulate quantities of interest</p>
<div class="highlight-r"><div class="highlight"><pre>s.out <span class="o">&lt;-</span> sim<span class="p">(</span>z.out<span class="p">,</span> x <span class="o">=</span> x.out<span class="p">)</span>
summary<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="highlight-python"><div class="highlight"><pre><span class="c">##</span>
<span class="c">##  sim x :</span>
<span class="c">##  -----</span>
<span class="c">## ev</span>
<span class="c">##          mean        sd      50%     2.5%    97.5%</span>
<span class="c">## [1,] 3.314683 0.8607266 3.188464 1.927721 5.355897</span>
<span class="c">## pv</span>
<span class="c">##       mean      sd 50% 2.5% 97.5%</span>
<span class="c">## [1,] 3.318 2.07103   3    0 8.025</span>
</pre></div>
</div>
<p>Generate a plot of quantities of interest:</p>
<div class="highlight-r"><div class="highlight"><pre>plot<span class="p">(</span>s.out<span class="p">)</span>
</pre></div>
</div>
<div class="figure">
<img alt="Zelig-poisson" src="_images/Zelig-poissongee-1.png" />
<p class="caption">Zelig-poisson</p>
</div>
</div>
</div>
<div class="section" id="id145">
<h3>The Model<a class="headerlink" href="#id145" title="Permalink to this headline">¶</a></h3>
<p>Suppose we have a panel dataset, with <img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> denoting the
dependent variable of the number of independent events for a fixed
period of time for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>. <img class="math" src="_images/math/0e1e069d7b701cb98cfecc70660e4e31b786db9a.png" alt="Y_{i}"/> is a
vector or cluster of correlated data where <img class="math" src="_images/math/b840d9057a2e3bad080c274b8f1d9303520f9052.png" alt="y_{it}"/> is correlated
with <img class="math" src="_images/math/da57192d38416f50ec88d145367cc7d5be0c06c5.png" alt="y_{it^\prime}"/> for some or all <img class="math" src="_images/math/2d3acc5e57adc9f94593e8bb1b3b28df87caf045.png" alt="t, t^\prime"/>. Note
that the model assumes correlations within <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> but independence
across <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>.</p>
<ul>
<li><p class="first">The <em>stochastic component</em> is given by the joint and marginal
distributions</p>
<div class="math">
<p><img src="_images/math/07e7595e1d9c1222ab0eab8fa6aa39fe54251868.png" alt="\begin{aligned}
Y_{i} &amp;\sim&amp; f(y_{i} \mid \lambda_{i})\\
Y_{it} &amp;\sim&amp; g(y_{it} \mid \lambda_{it})\end{aligned}"/></p>
</div><p>where <img class="math" src="_images/math/0001d02b63ede2fe3219e05a7cd09c82ae6298b6.png" alt="f"/> and <img class="math" src="_images/math/67f4714f065d485540ad40829e0717bf75e9dd85.png" alt="g"/> are unspecified distributions with
means <img class="math" src="_images/math/3d9a1fde3ad183d72319e293ff637c7528c831a4.png" alt="\lambda_{i}"/> and <img class="math" src="_images/math/5fc838daf306e42980d81b1d6b359f4a47fbd630.png" alt="\lambda_{it}"/>. GEE models make
no distributional assumptions and only require three specifications:
a mean function, a variance function, and a correlation structure.</p>
</li>
<li><p class="first">The <em>systematic component</em> is the <em>mean function</em>, given by:</p>
<div class="math">
<p><img src="_images/math/20ee5577bd63708f61b808d3433488f940221b46.png" alt="\lambda_{it}  = \mathrm{exp}(x_{it} \beta)"/></p>
</div><p>where <img class="math" src="_images/math/d43ef14e28002ff45394741fc187dced216e80f0.png" alt="x_{it}"/> is the vector of <img class="math" src="_images/math/e9203da50e1059455123460d4e716c9c7f440cc3.png" alt="k"/> explanatory variables
for unit <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> at time <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> and <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> is the vector
of coefficients.</p>
</li>
<li><p class="first">The <em>variance function</em> is given by:</p>
<div class="math">
<p><img src="_images/math/662af28b641c55a6dde1cfe560e751b26ee2666b.png" alt="V_{it} = \lambda_{it}"/></p>
</div></li>
<li><p class="first">The <em>correlation structure</em> is defined by a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/>
“working” correlation matrix, where <img class="math" src="_images/math/6d42c88506b8da39a2a23653aecbfb7c29728063.png" alt="T"/> is the size of the
largest cluster. Users must specify the structure of the “working”
correlation matrix <em>a priori</em>. The “working” correlation matrix then
enters the variance term for each <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/>, given by:</p>
<div class="math">
<p><img src="_images/math/16993e098fe3b3316a2033c341dedf1a4a5722f0.png" alt="V_{i} = \phi \, A_{i}^{\frac{1}{2}} R_{i}(\alpha) A_{i}^{\frac{1}{2}}"/></p>
</div><p>where <img class="math" src="_images/math/7f4051f79dc41b4b9b5b105d110e19898642c02d.png" alt="A_{i}"/> is a <img class="math" src="_images/math/88cefb40c3b35d50f8819fd81d3fd0694f0c9df4.png" alt="T \times T"/> diagonal matrix with the
variance function <img class="math" src="_images/math/7010e7d09e4adb6f88eade64a20026b8a6418def.png" alt="V_{it} = \lambda_{it}"/> as the <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/>th
diagonal element, <img class="math" src="_images/math/165cf808f99660333864da96d49d02862c5804ba.png" alt="R_{i}(\alpha)"/> is the “working” correlation
matrix, and <img class="math" src="_images/math/10e009bdb83f96c5f47c58b34d5d4b12ef268d5b.png" alt="\phi"/> is a scale parameter. The parameters are
then estimated via a quasi-likelihood approach.</p>
</li>
<li><p class="first">In GEE models, if the mean is correctly specified, but the variance
and correlation structure are incorrectly specified, then GEE models
provide consistent estimates of the parameters and thus the mean
function as well, while consistent estimates of the standard errors
can be obtained via a robust “sandwich” estimator. Similarly, if the
mean and variance are correctly specified but the correlation
structure is incorrectly specified, the parameters can be estimated
consistently and the standard errors can be estimated consistently
with the sandwich estimator. If all three are specified correctly,
then the estimates of the parameters are more efficient.</p>
</li>
<li><p class="first">The robust “sandwich” estimator gives consistent estimates of the
standard errors when the correlations are specified incorrectly only
if the number of units <img class="math" src="_images/math/a581f053bbfa5115f42c13094857cdd12a37ec49.png" alt="i"/> is relatively large and the number
of repeated periods <img class="math" src="_images/math/ef9270877405055756d345facd044e4ab297f858.png" alt="t"/> is relatively small. Otherwise, one
should use the “naïve” model-based standard errors, which assume that
the specified correlations are close approximations to the true
underlying correlations. See for more details.</p>
</li>
</ul>
</div>
<div class="section" id="id146">
<h3>Quantities of Interest<a class="headerlink" href="#id146" title="Permalink to this headline">¶</a></h3>
<ul>
<li><p class="first">All quantities of interest are for marginal means rather than joint
means.</p>
</li>
<li><p class="first">The method of bootstrapping generally should not be used in GEE
models. If you must bootstrap, bootstrapping should be done within
clusters, which is not currently supported in Zelig. For conditional
prediction models, data should be matched within clusters.</p>
</li>
<li><p class="first">The expected values (qi$ev) for the GEE poisson model is the mean of
simulations from the stochastic component:</p>
<div class="math">
<p><img src="_images/math/63bc144cca597bb88033d1558cd78b4257b46d3d.png" alt="E(Y) =
  \lambda_{c} = \mathrm{exp}(x_{c} \beta),"/></p>
</div><p>given draws of <img class="math" src="_images/math/8ce03f78ed945f2ef3dac87c8799b55b393527e7.png" alt="\beta"/> from its sampling distribution, where
<img class="math" src="_images/math/5502b77b431ce3a22763c40fede2822fe42a5aa2.png" alt="x_{c}"/> is a vector of values, one for each independent
variable, chosen by the user.</p>
</li>
<li><p class="first">The first difference (qi$fd) for the GEE poisson model is defined as</p>
<div class="math">
<p><img src="_images/math/8f669920018c72d7f23a4ba843925381c4703f17.png" alt="\textrm{FD} = \Pr(Y = 1 \mid x_1) - \Pr(Y = 1 \mid x)."/></p>
</div></li>
<li><p class="first">In conditional prediction models, the average expected treatment
effect (att.ev) for the treatment group is</p>
<div class="math">
<p><img src="_images/math/40ac87a54bf074ab721d0606edc354d20bfe2890.png" alt="\frac{1}{\sum_{i=1}^n \sum_{t=1}^T tr_{it}}\sum_{i:tr_{it}=1}^n \sum_{t:tr_{it}=1}^T \left\{ Y_{it}(tr_{it}=1) -
      E[Y_{it}(tr_{it}=0)] \right\},"/></p>
</div><p>where <img class="math" src="_images/math/e6f125c03f6d788e356d369529940f022a7f2758.png" alt="tr_{it}"/> is a binary explanatory variable defining the
treatment (<img class="math" src="_images/math/6456da07c94c2f41fd2db133459327f848c0f4bd.png" alt="tr_{it}=1"/>) and control (<img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>) groups.
Variation in the simulations are due to uncertainty in simulating
<img class="math" src="_images/math/55447d1ae15443b039fe51a56aee680735c5eff7.png" alt="E[Y_{it}(tr_{it}=0)]"/>, the counterfactual expected value of
<img class="math" src="_images/math/30f89e733d3b75f1db48233adcde1aa3ef74a897.png" alt="Y_{it}"/> for observations in the treatment group, under the
assumption that everything stays the same except that the treatment
indicator is switched to <img class="math" src="_images/math/cacdbdc3e92ce5d077bfe72555b39b0aa02b8a35.png" alt="tr_{it}=0"/>.</p>
</li>
</ul>
</div>
<div class="section" id="id147">
<h3>Output Values<a class="headerlink" href="#id147" title="Permalink to this headline">¶</a></h3>
<p>The output of each Zelig command contains useful information which you
may view. For example, if you run
<tt class="docutils literal"><span class="pre">z.out</span> <span class="pre">&lt;-</span> <span class="pre">zelig(y</span> <span class="pre">~</span> <span class="pre">x,</span> <span class="pre">model</span> <span class="pre">=</span> <span class="pre">poisson.gee,</span> <span class="pre">id,</span> <span class="pre">data)</span></tt>, then you may
examine the available information in <tt class="docutils literal"><span class="pre">z.out</span></tt> by using
<tt class="docutils literal"><span class="pre">names(z.out)</span></tt>, see the coefficients by using z.out$coefficients, and
a default summary of information through <tt class="docutils literal"><span class="pre">summary(z.out)</span></tt>. Other
elements available through the $ operator are listed below.</p>
<ul class="simple">
<li>From the zelig() output object z.out, you may extract:<ul>
<li>coefficients: parameter estimates for the explanatory variables.</li>
<li>residuals: the working residuals in the final iteration of the
fit.</li>
<li>fitted.values: the vector of fitted values for the systemic
component, <img class="math" src="_images/math/5fc838daf306e42980d81b1d6b359f4a47fbd630.png" alt="\lambda_{it}"/>.</li>
<li>linear.predictors: the vector of <img class="math" src="_images/math/62d42b965bd7739c65b214098af7195a9940d3fc.png" alt="x_{it}\beta"/></li>
<li>max.id: the size of the largest cluster.</li>
</ul>
</li>
<li>From summary(z.out), you may extract:<ul>
<li>coefficients: the parameter estimates with their associated
standard errors, <img class="math" src="_images/math/3eca8557203e86160952e1c0f735f7417f3285b1.png" alt="p"/>-values, and <img class="math" src="_images/math/84d7271dd9e78c1e05d6c3c6ecb60309ef7dfc73.png" alt="z"/>-statistics.</li>
<li>working.correlation: the “working” correlation matrix</li>
</ul>
</li>
<li>From the sim() output object s.out, you may extract quantities of
interest arranged as matrices indexed by simulation <img class="math" src="_images/math/c69691d64985442217922c8d34e835a9dea60178.png" alt="\times"/>
x-observation (for more than one x-observation). Available quantities
are:<ul>
<li>qi$ev: the simulated expected values for the specified values of
x.</li>
<li>qi$fd: the simulated first difference in the expected
probabilities for the values specified in x and x1.</li>
<li>qi$att.ev: the simulated average expected treatment effect for the
treated from conditional prediction models.</li>
</ul>
</li>
</ul>
</div>
<div class="section" id="id148">
<h3>See also<a class="headerlink" href="#id148" title="Permalink to this headline">¶</a></h3>
<p>The geeglm function is part of the geepack package by Søren Højsgaard,
Ulrich Halekoh and Jun Yan. Advanced users may wish to refer
to <tt class="docutils literal"><span class="pre">help(geepack)</span></tt> and <tt class="docutils literal"><span class="pre">help(family)</span></tt>.</p>
<hr class="docutils" />
<div class="toctree-wrapper compound">
</div>
</div>
</div>
</div>


    </div>
      
  </div>
</div>

<footer class="footer">
  <div class="container">
    <p class="pull-right">
      <a href="#">Back to top</a>
      
        <br/>
        
<div id="sourcelink">
  <a href="_sources/vignette.txt"
     rel="nofollow">Source</a>
</div>
      
    </p>
    <p>
        &copy; Copyright 2014, The Zelig Team.<br/>
      Created using <a href="http://sphinx.pocoo.org/">Sphinx</a> 1.2.2.<br/>
    </p>
  </div>
</footer>
<script>
  (function(i,s,o,g,r,a,m){i['GoogleAnalyticsObject']=r;i[r]=i[r]||function(){
  (i[r].q=i[r].q||[]).push(arguments)},i[r].l=1*new Date();a=s.createElement(o),
  m=s.getElementsByTagName(o)[0];a.async=1;a.src=g;m.parentNode.insertBefore(a,m)
  })(window,document,'script','//www.google-analytics.com/analytics.js','ga');

  ga('create', 'UA-27761864-7', 'http://zeligproject.org/');
  ga('send', 'pageview');

</script>



  </body>
</html>